diff -Naur gnome-vfs-2.24.1.old/schemas/desktop_gnome_url_handlers.schemas.in gnome-vfs-2.24.1/schemas/desktop_gnome_url_handlers.schemas.in
--- gnome-vfs-2.24.1.old/schemas/desktop_gnome_url_handlers.schemas.in	2009-03-17 13:48:19.000000000 +0000
+++ gnome-vfs-2.24.1/schemas/desktop_gnome_url_handlers.schemas.in	2009-07-28 18:52:05.000000000 +0100
@@ -137,6 +137,40 @@
       </locale>
     </schema>
 
+  <schema>
+      <key>/schemas/desktop/gnome/url-handlers/ftp/command</key>
+      <applyto>/desktop/gnome/url-handlers/ftp/command</applyto>
+      <owner>gnome</owner>
+      <type>bool</type>
+      <default>true</default>
+      <locale name="C">
+        <short>Whether the specified command should handle "ftp" URLs</short>
+        <long>True if the command specified in the "command" key should handle "ftp" URLs.</long>
+      </locale>
+    </schema>
+    <schema>
+      <key>/schemas/desktop/gnome/url-handlers/ftp/command</key>
+      <applyto>/desktop/gnome/url-handlers/ftp/command</applyto>
+      <owner>gnome</owner>
+      <type>string</type>
+      <default>nautilus %s</default>
+      <locale name="C">
+        <short>The handler for "ftp" URLs</short>
+        <long>The command used to handle "ftp" URLs, if enabled.</long>
+      </locale>
+    </schema>
+    <schema>
+      <key>/schemas/desktop/gnome/url-handlers/ftp/needs_terminal</key>
+      <applyto>/desktop/gnome/url-handlers/ftp/needs_terminal</applyto>
+      <owner>gnome</owner>
+      <type>bool</type>
+      <default>false</default>
+      <locale name="C">
+        <short>Run the command in a terminal</short>
+        <long>True if the command used to handle this type of URL should be run in a terminal.</long>
+      </locale>
+    </schema>
+
     <schema>
       <key>/schemas/desktop/gnome/url-handlers/http/enabled</key>
       <applyto>/desktop/gnome/url-handlers/http/enabled</applyto>
@@ -289,7 +323,7 @@
       <applyto>/desktop/gnome/url-handlers/callto/command</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>gnomemeeting -c %s</default>
+      <default>ekiga -c %s</default>
       <locale name="C">
         <short>The handler for "callto" URLs</short>
         <long>The command used to handle "callto" URLs, if enabled.</long>
@@ -323,7 +357,7 @@
       <applyto>/desktop/gnome/url-handlers/h323/command</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>gnomemeeting -c %s</default>
+      <default>ekiga -c %s</default>
       <locale name="C">
         <short>The handler for "h323" URLs</short>
         <long>The command used to handle "h323" URLs, if enabled.</long>
