diff -ur gdm-3.6.2/daemon/gdm.in gdm-3.6.2.new/daemon/gdm.in
--- gdm-3.6.2/daemon/gdm.in	2012-07-21 03:00:19.000000000 +0800
+++ gdm-3.6.2.new/daemon/gdm.in	2013-01-23 15:42:08.792520233 +0800
@@ -8,11 +8,11 @@
 
 # Make sure LANG is set
 #
-if [ -z "$LANG" ]
+if [ -z "$LANG" ] || [ "x$LANG" = "xC" ]
 then
-  if [ -f /etc/sysconfig/language ]
+  if [ -f /etc/default/locale ]
   then
-    LANG=`. /etc/sysconfig/language; echo $RC_LANG`
+    LANG=`. /etc/default/locale; echo $LANG`
     export LANG
   fi
 fi
