--- k3b-2.0.2/CMakeLists.txt	2011-01-16 04:52:01.000000000 +0800
+++ k3b-2.0.2n/CMakeLists.txt	2011-12-21 15:40:00.544640713 +0800
@@ -88,11 +88,6 @@
     FALSE "1.1" "")
 endif(K3B_ENABLE_MUSICBRAINZ)
 
-find_package(Kcddb)
-macro_log_feature( KCDDB_FOUND "KCddb"
-  "KCddb is used to retrieve audio CD meta data from the internet."
-  "http://multimedia.kde.org"
-  TRUE "" "")
 
 if(K3B_BUILD_FFMPEG_DECODER_PLUGIN)
   find_package(FFmpeg)
