diff -ur systemd-44/config.h.in systemd-44.new//config.h.in
--- systemd-44/config.h.in	2012-03-16 08:58:04.000000000 +0800
+++ systemd-44.new//config.h.in	2012-04-06 10:54:27.643777067 +0800
@@ -139,6 +139,9 @@
 /* Target is Fedora/RHEL */
 #undef TARGET_FEDORA
 
+/* Target is StartOS */
+#undef TARGET_STARTOS
+
 /* Target is Frugalware */
 #undef TARGET_FRUGALWARE
 
diff -ur systemd-44/configure systemd-44.new//configure
--- systemd-44/configure	2012-03-16 08:58:03.000000000 +0800
+++ systemd-44.new//configure	2012-04-06 10:54:34.953990293 +0800
@@ -717,6 +717,8 @@
 TARGET_SUSE_TRUE
 TARGET_FEDORA_FALSE
 TARGET_FEDORA_TRUE
+TARGET_STARTOS_FALSE
+TARGET_STARTOS_TRUE
 M4_DEFINES
 SYSTEM_SYSVRCND_PATH
 SYSTEM_SYSVINIT_PATH
@@ -1677,7 +1679,7 @@
                         (or the compiler's sysroot if not specified).
   --with-distro=DISTRO    Specify the distribution to target: One of fedora,
                           suse, debian, ubuntu, arch, gentoo, slackware,
-                          altlinux, mandriva, meego, mageia, angstrom or other
+                          altlinux, mandriva, meego, mageia, angstrom, startos or other
   --with-sysvinit-path=PATH
                           Specify the path to where the SysV init scripts are
                           located [default=based on distro]
@@ -15854,6 +15856,18 @@
                 M4_DEFINES=-DTARGET_FEDORA=1
                 have_plymouth=yes
                 ;;
+
+	startos)
+                SYSTEM_SYSVINIT_PATH=/etc/init.d
+		SYSTEM_SYSVRCND_PATH=/etc
+
+$as_echo "#define TARGET_STARTOS /**/" >>confdefs.h
+
+                M4_DEFINES=-DTARGET_STARTOS=1
+                have_plymouth=yes
+                ;;
+
+
         opensuse|suse)
                 SYSTEM_SYSVRCND_PATH=/etc/init.d
 
@@ -16011,6 +16025,14 @@
   TARGET_FEDORA_FALSE=
 fi
 
+ if test x"$with_distro" = xstartos; then
+  TARGET_STARTOS_TRUE=
+  TARGET_STARTOS_FALSE='#'
+else
+  TARGET_STARTOS_TRUE='#'
+  TARGET_STARTOS_FALSE=
+fi
+
  if test x"$with_distro" = xsuse; then
   TARGET_SUSE_TRUE=
   TARGET_SUSE_FALSE='#'
@@ -16457,6 +16479,10 @@
   as_fn_error $? "conditional \"TARGET_FEDORA\" was never defined.
 Usually this means the macro was only invoked conditionally." "$LINENO" 5
 fi
+if test -z "${TARGET_STARTOS_TRUE}" && test -z "${TARGET_STARTOS_FALSE}"; then
+  as_fn_error $? "conditional \"TARGET_STARTOS\" was never defined.
+Usually this means the macro was only invoked conditionally." "$LINENO" 5
+fi
 if test -z "${TARGET_SUSE_TRUE}" && test -z "${TARGET_SUSE_FALSE}"; then
   as_fn_error $? "conditional \"TARGET_SUSE\" was never defined.
 Usually this means the macro was only invoked conditionally." "$LINENO" 5
diff -ur systemd-44/configure.ac systemd-44.new//configure.ac
--- systemd-44/configure.ac	2012-03-16 08:46:57.000000000 +0800
+++ systemd-44.new//configure.ac	2012-04-06 10:53:17.355086122 +0800
@@ -396,7 +396,7 @@
 
 AC_PATH_PROG([M4], [m4])
 
-AC_ARG_WITH(distro, AS_HELP_STRING([--with-distro=DISTRO],[Specify the distribution to target: One of fedora, suse, debian, ubuntu, arch, gentoo, slackware, altlinux, mandriva, meego, mageia, angstrom or other]))
+AC_ARG_WITH(distro, AS_HELP_STRING([--with-distro=DISTRO],[Specify the distribution to target: One of fedora, suse, debian, ubuntu, arch, gentoo, slackware, altlinux, mandriva, meego, mageia, angstrom, startos or other]))
 if test "z$with_distro" = "z"; then
         if test "$cross_compiling" = yes; then
                 AC_MSG_WARN([Target distribution cannot be reliably detected when cross-compiling. You should specify it with --with-distro (see $0 --help for recognized distros)])
@@ -424,6 +424,13 @@
                 M4_DEFINES=-DTARGET_FEDORA=1
                 have_plymouth=yes
                 ;;
+        startos)
+                SYSTEM_SYSVINIT_PATH=/etc/init.d
+		SYSTEM_SYSVRCND_PATH=/etc
+                AC_DEFINE(TARGET_STARTOS, [], [Target is startos])
+                M4_DEFINES=-DTARGET_STARTOS=1
+                have_plymouth=yes
+                ;;
         opensuse|suse)
                 SYSTEM_SYSVRCND_PATH=/etc/init.d
                 AC_DEFINE(TARGET_SUSE, [], [Target is openSUSE/SLE])
@@ -537,6 +544,7 @@
 fi
 
 AM_CONDITIONAL(TARGET_FEDORA, test x"$with_distro" = xfedora)
+AM_CONDITIONAL(TARGET_STARTOS, test x"$with_distro" = xstartos)
 AM_CONDITIONAL(TARGET_SUSE, test x"$with_distro" = xsuse)
 AM_CONDITIONAL(TARGET_DEBIAN, test x"$with_distro" = xdebian)
 AM_CONDITIONAL(TARGET_UBUNTU, test x"$with_distro" = xubuntu)
