diff -ur systemd-44/Makefile.am systemd-44.new//Makefile.am
--- systemd-44/Makefile.am	2012-03-16 08:46:51.000000000 +0800
+++ systemd-44.new//Makefile.am	2012-04-06 14:12:16.337689273 +0800
@@ -295,6 +295,7 @@
 	units/sys-fs-fuse-connections.mount \
 	units/var-run.mount \
 	units/media.mount \
+	units/tmp.mount \
 	units/remount-rootfs.service \
 	units/printer.target \
 	units/sound.target \
@@ -2307,12 +2308,14 @@
 			fsck-root.service \
 			remount-rootfs.service \
 			var-run.mount \
+			tmp.mount \
 			media.mount && \
 		$(LN_S) ../systemd-remount-api-vfs.service systemd-remount-api-vfs.service && \
 		$(LN_S) ../fsck-root.service fsck-root.service && \
 		$(LN_S) ../remount-rootfs.service remount-rootfs.service && \
 		$(LN_S) ../var-run.mount var-run.mount && \
-		$(LN_S) ../media.mount media.mount )
+		$(LN_S) ../media.mount media.mount && \
+		$(LN_S) ../tmp.mount tmp.mount )
 	( cd $(DESTDIR)$(userunitdir) && \
 		rm -f shutdown.target sockets.target bluetooth.target printer.target sound.target && \
 		$(LN_S) $(systemunitdir)/shutdown.target shutdown.target && \
diff -ur systemd-44/Makefile.in systemd-44.new//Makefile.in
--- systemd-44/Makefile.in	2012-03-16 08:58:05.000000000 +0800
+++ systemd-44.new//Makefile.in	2012-04-06 14:13:55.115273290 +0800
@@ -1486,7 +1486,7 @@
 	units/dev-mqueue.mount units/sys-kernel-config.mount \
 	units/sys-kernel-debug.mount units/sys-kernel-security.mount \
 	units/sys-fs-fuse-connections.mount units/var-run.mount \
-	units/media.mount units/remount-rootfs.service \
+	units/media.mount units/tmp.mount units/remount-rootfs.service \
 	units/printer.target units/sound.target units/bluetooth.target \
 	units/smartcard.target units/systemd-tmpfiles-clean.timer \
 	units/quotaon.service units/systemd-ask-password-wall.path \
@@ -1947,7 +1947,7 @@
 	units/dev-mqueue.mount units/sys-kernel-config.mount \
 	units/sys-kernel-debug.mount units/sys-kernel-security.mount \
 	units/sys-fs-fuse-connections.mount units/var-run.mount \
-	units/media.mount units/remount-rootfs.service \
+	units/media.mount units/tmp.mount units/remount-rootfs.service \
 	units/printer.target units/sound.target units/bluetooth.target \
 	units/smartcard.target units/systemd-tmpfiles-clean.timer \
 	units/quotaon.service units/systemd-ask-password-wall.path \
@@ -8291,12 +8291,14 @@
 			fsck-root.service \
 			remount-rootfs.service \
 			var-run.mount \
+			tmp.mount \
 			media.mount && \
 		$(LN_S) ../systemd-remount-api-vfs.service systemd-remount-api-vfs.service && \
 		$(LN_S) ../fsck-root.service fsck-root.service && \
 		$(LN_S) ../remount-rootfs.service remount-rootfs.service && \
 		$(LN_S) ../var-run.mount var-run.mount && \
-		$(LN_S) ../media.mount media.mount )
+		$(LN_S) ../media.mount media.mount && \
+		$(LN_S) ../tmp.mount tmp.mount )
 	( cd $(DESTDIR)$(userunitdir) && \
 		rm -f shutdown.target sockets.target bluetooth.target printer.target sound.target && \
 		$(LN_S) $(systemunitdir)/shutdown.target shutdown.target && \
--- /dev/null	                   2012-03-16 08:58:05.000000000 +0800
+++ systemd-44.new/units/tmp.mount 2012-04-06 14:13:55.115273290 +0800
@@ -0,0 +1,17 @@
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=Tmp Directory
+Before=local-fs.target
+
+[Mount]
+What=tmpfs
+Where=/tmp
+Type=tmpfs
+Options=mode=1777,nosuid,nodev
+
