diff -ur avahi-0.6.30/configure avahi-0.6.30.new//configure
--- avahi-0.6.30/configure	2011-04-04 06:10:41.000000000 +0800
+++ avahi-0.6.30.new//configure	2011-12-13 11:27:37.118030802 +0800
@@ -1763,8 +1759,8 @@
   --with-sysroot=DIR Search for dependent libraries within DIR
                         (or the compiler's sysroot if not specified).
   --with-distro=DISTRO    Specify the distribution to target: One of lfs,
-                          debian, gentoo, archlinux, fedora, mandriva, darwin,
-                          netbsd, freebsd, slackware or none
+                          ylmfos, debian, gentoo, archlinux, fedora, mandriva,
+                          darwin, netbsd, freebsd, slackware or none
   --with-dbus-sys=<dir>   Path to D-Bus system.d directory
   --with-dbus-system-address=<address>
                           Path to the D-Bus system socket, you probably want
@@ -17279,6 +17275,25 @@
   with_distro="debian"
 fi
 
+        { $as_echo "$as_me:${as_lineno-$LINENO}: checking for /etc/ylmfos-release" >&5
+$as_echo_n "checking for /etc/ylmfos-release... " >&6; }
+if ${ac_cv_file__etc_ylmfos_release+:} false; then :
+  $as_echo_n "(cached) " >&6
+else
+  test "$cross_compiling" = yes &&
+  as_fn_error $? "cannot check for file existence when cross compiling" "$LINENO" 5
+if test -r "/etc/ylmfos-release"; then
+  ac_cv_file__etc_ylmfos_release=yes
+else
+  ac_cv_file__etc_ylmfos_release=no
+fi
+fi
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_file__etc_ylmfos_release" >&5
+$as_echo "$ac_cv_file__etc_ylmfos_release" >&6; }
+if test "x$ac_cv_file__etc_ylmfos_release" = xyes; then :
+  with_distro="ylmfos"
+fi
+
         { $as_echo "$as_me:${as_lineno-$LINENO}: checking for /etc/redhat-release" >&5
 $as_echo_n "checking for /etc/redhat-release... " >&6; }
 if ${ac_cv_file__etc_redhat_release+:} false; then :
@@ -17344,7 +17359,7 @@
 with_distro=`echo ${with_distro} | tr '[:upper:]' '[:lower:]' `
 
 case $with_distro in
-    lfs|debian|gentoo|archlinux|suse|fedora|mandriva|darwin|freebsd|slackware|none)
+    lfs|ylmfos|debian|gentoo|archlinux|suse|fedora|mandriva|darwin|freebsd|slackware|none)
      ;;
     netbsd)
      { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Your distribution (${with_distro}) is supported but no init script exist yet! (patches welcome)" >&5
@@ -17390,6 +17405,14 @@
   TARGET_DEBIAN_FALSE=
 fi
 
+ if test x"$with_distro" = xylmfos; then
+  TARGET_DEBIAN_TRUE=
+  TARGET_DEBIAN_FALSE='#'
+else
+  TARGET_DEBIAN_TRUE='#'
+  TARGET_DEBIAN_FALSE=
+fi
+
  if test x"$with_distro" = xarchlinux; then
   TARGET_ARCHLINUX_TRUE=
   TARGET_ARCHLINUX_FALSE='#'
diff -ur avahi-0.6.30/configure.ac avahi-0.6.30.new//configure.ac
--- avahi-0.6.30/configure.ac	2011-04-04 06:08:03.000000000 +0800
+++ avahi-0.6.30.new//configure.ac	2011-12-13 11:14:35.197192591 +0800
@@ -250,7 +250,7 @@
 #
 # Detecting the linux distribution for specific things like init scripts.
 #
-AC_ARG_WITH(distro, AS_HELP_STRING([--with-distro=DISTRO],[Specify the distribution to target: One of lfs, debian, gentoo, archlinux, fedora, mandriva, darwin, netbsd, freebsd, slackware or none]))
+AC_ARG_WITH(distro, AS_HELP_STRING([--with-distro=DISTRO],[Specify the distribution to target: One of lfs, ylmfos, debian, gentoo, archlinux, fedora, mandriva, darwin, netbsd, freebsd, slackware or none]))
 if test "z$with_distro" = "z"; then
     if test "$cross_compiling" = yes; then
         AC_MSG_WARN([Target distribution cannot be reliably detected when cross-compiling. You should specify it with --with-distro (see $0 --help for recognized distros)])
@@ -260,6 +260,7 @@
         AC_CHECK_FILE(/etc/gentoo-release,with_distro="gentoo")
         AC_CHECK_FILE(/etc/arch-release,with_distro="archlinux")
         AC_CHECK_FILE(/etc/debian_version,with_distro="debian")
+        AC_CHECK_FILE(/etc/ylmfos-release,with_distro="ylmfos")
         AC_CHECK_FILE(/etc/redhat-release,with_distro="fedora")
         AC_CHECK_FILE(/etc/mandriva-release,with_distro="mandriva")
         AC_CHECK_FILE(/etc/slackware-version,with_distro="slackware")
@@ -271,7 +272,7 @@
 with_distro=`echo ${with_distro} | tr '[[:upper:]]' '[[:lower:]]' `
 
 case $with_distro in
-    lfs|debian|gentoo|archlinux|suse|fedora|mandriva|darwin|freebsd|slackware|none)
+    lfs|ylmfos|debian|gentoo|archlinux|suse|fedora|mandriva|darwin|freebsd|slackware|none)
      ;;
     netbsd)
      AC_MSG_WARN([Your distribution (${with_distro}) is supported but no init script exist yet! (patches welcome)])
@@ -288,6 +289,7 @@
 AM_CONDITIONAL(TARGET_SUSE, test x"$with_distro" = xsuse)
 AM_CONDITIONAL(TARGET_GENTOO, test x"$with_distro" = xgentoo)
 AM_CONDITIONAL(TARGET_DEBIAN, test x"$with_distro" = xdebian)
+AM_CONDITIONAL(TARGET_DEBIAN, test x"$with_distro" = xylmfos)
 AM_CONDITIONAL(TARGET_ARCHLINUX, test x"$with_distro" = xarchlinux)
 AM_CONDITIONAL(TARGET_FEDORA, test x"$with_distro" = xfedora)
 AM_CONDITIONAL(TARGET_MANDRIVA, test x"$with_distro" = xmandriva)
