diff -ruN /tmp/gnome-session-2.32.1/gnome-session/gs-idle-monitor.c gnome-session-2.32.1/gnome-session/gs-idle-monitor.c
--- /tmp/gnome-session-2.32.1/gnome-session/gs-idle-monitor.c	2010-11-17 11:17:04.000000000 +0000
+++ gnome-session-2.32.1/gnome-session/gs-idle-monitor.c	2011-09-29 13:48:26.732529728 +0000
@@ -459,6 +459,8 @@
                 watch->xalarm_positive = XSyncCreateAlarm (GDK_DISPLAY (), flags, &attr);
         }
 
+	/* fixed: https://bugzilla.gnome.org/show_bug.cgi?id=627903 */
+	attr.trigger.wait_value = _int64_to_xsyncvalue ( _xsyncvalue_to_int64 (watch->interval) - 1);
         attr.trigger.test_type = XSyncNegativeTransition;
         if (watch->xalarm_negative != None) {
                 g_debug ("GSIdleMonitor: updating alarm for negative transition wait=%lld",
