diff -ur lightdm-1.3.1/data/lightdm.conf lightdm-1.3.1.new//data/lightdm.conf
--- lightdm-1.3.1/data/lightdm.conf	2012-05-28 23:27:22.000000000 +0000
+++ lightdm-1.3.1.new//data/lightdm.conf	2012-06-26 13:55:10.342348284 +0000
@@ -64,11 +64,11 @@
 #xdmcp-manager=
 #xdmcp-port=177
 #xdmcp-key=
-#greeter-session=example-gtk-gnome
+greeter-session=lightdm-ylmfos-greeter
 #greeter-hide-users=false
 #greeter-allow-guest=true
 #greeter-show-manual-login=false
-#user-session=default
+user-session=gnome
 #allow-guest=true
 #guest-session=UNIMPLEMENTED
 #session-wrapper=lightdm-session
diff -ur lightdm-1.3.1/data/users.conf lightdm-1.3.1.new//data/users.conf
--- lightdm-1.3.1/data/users.conf	2011-12-09 01:51:23.000000000 +0000
+++ lightdm-1.3.1.new//data/users.conf	2012-06-26 13:53:41.428468556 +0000
@@ -9,6 +9,6 @@
 # hidden-shells = Shells that indicate a user cannot login
 #
 [UserAccounts]
-minimum-uid=500
+minimum-uid=1000
 hidden-users=nobody nobody4 noaccess
 hidden-shells=/bin/false /usr/sbin/nologin
