diff -ur lightdm-1.3.1-r4/utils/gdmflexiserver lightdm-1.3.1-r4.new//utils/gdmflexiserver
--- lightdm-1.3.1-r4/utils/gdmflexiserver	2011-12-09 01:51:23.000000000 +0000
+++ lightdm-1.3.1-r4.new//utils/gdmflexiserver	2012-07-11 09:59:01.246409201 +0000
@@ -9,9 +9,42 @@
 #
 # See http://www.gnu.org/copyleft/gpl.html for the full text of the license.
 
+export LIGHTDMRUN=1
+export GDMRUN=1
+
+if [ -e "/var/run/lightdm.pid" ]; then
+	LPIDF=$(cat /var/run/lightdm.pid) # $(pidof lightdm) will match many more processes .
+	/bin/ps ax | grep -qi ^$LPIDF || LIGHTDMRUN=0
+fi
+
+if [ -e "/var/run/gdm.pid" ]; then
+	GPIDF=$(cat /var/run/gdm.pid)
+	GPIDP=$(pidof gdm-binary)
+	if [[ $GPIDF != $GPIDP ]]; then
+		GDMRUN=0
+	fi
+else
+	GDMRUN=0
+fi
+
 if [ -z "$XDG_SEAT_PATH" ]; then
-  # something went wrong
-  exit 1
+	# something went wrong
+	# exit 1
+	LIGHTDMRUN=0
+fi
+
+if ! [[ $LIGHTDMRUN || $GDMRUN ]]; then
+	exit 1
+fi
+
+if [[ $LIGHTDMRUN -eq 1 ]]; then
+	echo "lightdm"
+	dbus-send --system --type=method_call --print-reply --dest=org.freedesktop.DisplayManager $XDG_SEAT_PATH org.freedesktop.DisplayManager.Seat.SwitchToGreeter
+elif [[ $GDMRUN -eq 1 ]]; then
+	echo "gdm"
+	dbus-send --system --type=method_call --print-reply --dest=org.gnome.DisplayManager /org/gnome/DisplayManager/LocalDisplayFactory  org.gnome.DisplayManager.LocalDisplayFactory.CreateTransientDisplay
+else
+    exit 2
 fi
 
-dbus-send --system --type=method_call --print-reply --dest=org.freedesktop.DisplayManager $XDG_SEAT_PATH org.freedesktop.DisplayManager.Seat.SwitchToGreeter
+exit 0
