diff -ur pygobject-2.21.3/gobject/pygobject.c pygobject-2.21.3.new//gobject/pygobject.c
--- pygobject-2.21.3/gobject/pygobject.c	2010-06-21 23:36:32.000000000 +0800
+++ pygobject-2.21.3.new//gobject/pygobject.c	2011-06-28 15:18:02.413026730 +0800
@@ -176,8 +176,8 @@
 {
     SinkFunc sf;
 
-    g_message ("pygobject_register_sinkfunc is deprecated (%s)",
-               g_type_name(type));
+    //g_message ("pygobject_register_sinkfunc is deprecated (%s)",
+    //           g_type_name(type));
 
 #if 0
     g_return_if_fail(G_TYPE_IS_OBJECT(type));
diff -ur pygobject-2.21.3/gobject/pygobject.h pygobject-2.21.3.new//gobject/pygobject.h
--- pygobject-2.21.3/gobject/pygobject.h	2010-06-21 23:36:32.000000000 +0800
+++ pygobject-2.21.3.new//gobject/pygobject.h	2011-06-28 15:15:14.333028956 +0800
@@ -211,7 +211,7 @@
 #define pygobject_register_class    (_PyGObject_API->register_class)
 #define pygobject_register_wrapper  (_PyGObject_API->register_wrapper)
 /* This is deprecated, sinkfuncs are not needed anymore */
-#define pygobject_register_sinkfunc (_PyGObject_API->register_sinkfunc)
+#define pygobject_register_sinkfunc (_PyGObject_API->register_sinkfunc) 
 #define pygobject_lookup_class      (_PyGObject_API->lookup_class)
 #define pygobject_new               (_PyGObject_API->newgobj)
 #define pyg_closure_new             (_PyGObject_API->closure_new)
