diff -ur pygobject-2.26.0/gobject/pygobject.c pygobject-2.26.0.new//gobject/pygobject.c
--- pygobject-2.26.0/gobject/pygobject.c	2010-09-17 23:00:22.000000000 +0800
+++ pygobject-2.26.0.new//gobject/pygobject.c	2011-09-15 14:23:15.191470155 +0800
@@ -176,8 +176,8 @@
 {
     SinkFunc sf;
 
-    g_message ("pygobject_register_sinkfunc is deprecated (%s)",
-               g_type_name(type));
+    //g_message ("pygobject_register_sinkfunc is deprecated (%s)",
+    //           g_type_name(type));
 
 #if 0
     g_return_if_fail(G_TYPE_IS_OBJECT(type));
