diff -ur linux-kernel-3.0.20-ylmf1/fs/file_table.c linux-kernel-3.0.20-ylmf1.new//fs/file_table.c
--- linux-kernel-3.0.20-ylmf1/fs/file_table.c	2012-02-07 01:31:45.000000000 +0800
+++ linux-kernel-3.0.20-ylmf1.new//fs/file_table.c	2012-04-25 14:19:23.182234005 +0800
@@ -443,6 +443,7 @@
 	}
 }
 
+EXPORT_SYMBOL(file_sb_list_del);
 #ifdef CONFIG_SMP
 
 /*
diff -ur linux-kernel-3.0.20-ylmf1/fs/inode.c linux-kernel-3.0.20-ylmf1.new//fs/inode.c
--- linux-kernel-3.0.20-ylmf1/fs/inode.c	2012-02-07 01:31:45.000000000 +0800
+++ linux-kernel-3.0.20-ylmf1.new//fs/inode.c	2012-04-25 14:20:03.199725010 +0800
@@ -69,6 +69,7 @@
 
 __cacheline_aligned_in_smp DEFINE_SPINLOCK(inode_sb_list_lock);
 __cacheline_aligned_in_smp DEFINE_SPINLOCK(inode_wb_list_lock);
+EXPORT_SYMBOL(inode_sb_list_lock);
 
 /*
  * iprune_sem provides exclusion between the icache shrinking and the
diff -ur linux-kernel-3.0.20-ylmf1/fs/namei.c linux-kernel-3.0.20-ylmf1.new//fs/namei.c
--- linux-kernel-3.0.20-ylmf1/fs/namei.c	2012-04-25 14:17:47.013592272 +0800
+++ linux-kernel-3.0.20-ylmf1.new//fs/namei.c	2012-04-25 14:20:47.983982579 +0800
@@ -368,6 +368,8 @@
 	return 0;
 }
 
+EXPORT_SYMBOL(deny_write_access);
+
 /**
  * path_get - get a reference to a path
  * @path: path to get the reference to
@@ -1710,6 +1712,8 @@
 	return __lookup_hash(&nd->last, nd->path.dentry, nd);
 }
 
+EXPORT_SYMBOL(lookup_hash);
+
 /**
  * lookup_one_len - filesystem helper to lookup single pathname component
  * @name:	pathname component to lookup
diff -ur linux-kernel-3.0.20-ylmf1/fs/namespace.c linux-kernel-3.0.20-ylmf1.new//fs/namespace.c
--- linux-kernel-3.0.20-ylmf1/fs/namespace.c	2012-02-07 01:31:45.000000000 +0800
+++ linux-kernel-3.0.20-ylmf1.new//fs/namespace.c	2012-04-25 14:21:19.007957218 +0800
@@ -1507,6 +1507,8 @@
 	return 0;
 }
 
+EXPORT_SYMBOL(iterate_mounts);
+
 static void cleanup_group_ids(struct vfsmount *mnt, struct vfsmount *end)
 {
 	struct vfsmount *p;
diff -ur linux-kernel-3.0.20-ylmf1/fs/notify/group.c linux-kernel-3.0.20-ylmf1.new//fs/notify/group.c
--- linux-kernel-3.0.20-ylmf1/fs/notify/group.c	2012-02-07 01:31:45.000000000 +0800
+++ linux-kernel-3.0.20-ylmf1.new//fs/notify/group.c	2012-04-25 14:22:46.386434633 +0800
@@ -22,6 +22,7 @@
 #include <linux/srcu.h>
 #include <linux/rculist.h>
 #include <linux/wait.h>
+#include <linux/module.h>
 
 #include <linux/fsnotify_backend.h>
 #include "fsnotify.h"
@@ -70,6 +71,7 @@
 	if (atomic_dec_and_test(&group->refcnt))
 		fsnotify_destroy_group(group);
 }
+EXPORT_SYMBOL(fsnotify_put_group);
 
 /*
  * Create a new fsnotify_group and hold a reference for the group returned.
@@ -102,3 +104,5 @@
 
 	return group;
 }
+EXPORT_SYMBOL(fsnotify_alloc_group);
+
diff -ur linux-kernel-3.0.20-ylmf1/fs/notify/mark.c linux-kernel-3.0.20-ylmf1.new//fs/notify/mark.c
--- linux-kernel-3.0.20-ylmf1/fs/notify/mark.c	2012-02-07 01:31:45.000000000 +0800
+++ linux-kernel-3.0.20-ylmf1.new//fs/notify/mark.c	2012-04-25 14:24:34.338680808 +0800
@@ -112,6 +112,7 @@
 	if (atomic_dec_and_test(&mark->refcnt))
 		mark->free_mark(mark);
 }
+EXPORT_SYMBOL(fsnotify_put_mark);
 
 /*
  * Any time a mark is getting freed we end up here.
@@ -191,6 +192,7 @@
 	if (unlikely(atomic_dec_and_test(&group->num_marks)))
 		fsnotify_final_destroy_group(group);
 }
+EXPORT_SYMBOL(fsnotify_destroy_mark);
 
 void fsnotify_set_mark_mask_locked(struct fsnotify_mark *mark, __u32 mask)
 {
@@ -278,6 +280,7 @@
 
 	return ret;
 }
+EXPORT_SYMBOL(fsnotify_add_mark);
 
 /*
  * clear any marks in a group in which mark->flags & flags is true
@@ -333,6 +336,7 @@
 	atomic_set(&mark->refcnt, 1);
 	mark->free_mark = free_mark;
 }
+EXPORT_SYMBOL(fsnotify_init_mark);
 
 static int fsnotify_mark_destroy(void *ignored)
 {
diff -ur linux-kernel-3.0.20-ylmf1/fs/open.c linux-kernel-3.0.20-ylmf1.new//fs/open.c
--- linux-kernel-3.0.20-ylmf1/fs/open.c	2012-02-07 01:31:45.000000000 +0800
+++ linux-kernel-3.0.20-ylmf1.new//fs/open.c	2012-04-25 14:25:05.926006578 +0800
@@ -60,6 +60,7 @@
 	mutex_unlock(&dentry->d_inode->i_mutex);
 	return ret;
 }
+EXPORT_SYMBOL(do_truncate);
 
 static long do_sys_truncate(const char __user *pathname, loff_t length)
 {
diff -ur linux-kernel-3.0.20-ylmf1/fs/splice.c linux-kernel-3.0.20-ylmf1.new//fs/splice.c
--- linux-kernel-3.0.20-ylmf1/fs/splice.c	2012-04-25 14:17:47.016925674 +0800
+++ linux-kernel-3.0.20-ylmf1.new//fs/splice.c	2012-04-25 14:25:47.413540190 +0800
@@ -1109,6 +1109,7 @@
 
 	return splice_write(pipe, out, ppos, len, flags);
 }
+EXPORT_SYMBOL(do_splice_from);
 
 /*
  * Attempt to initiate a splice from a file to a pipe.
@@ -1135,6 +1136,7 @@
 
 	return splice_read(in, ppos, pipe, len, flags);
 }
+EXPORT_SYMBOL(do_splice_to);
 
 /**
  * splice_direct_to_actor - splices data directly between two non-pipes
diff -ur linux-kernel-3.0.20-ylmf1/security/commoncap.c linux-kernel-3.0.20-ylmf1.new//security/commoncap.c
--- linux-kernel-3.0.20-ylmf1/security/commoncap.c	2012-02-07 01:31:45.000000000 +0800
+++ linux-kernel-3.0.20-ylmf1.new//security/commoncap.c	2012-04-25 14:26:20.327562319 +0800
@@ -971,3 +971,6 @@
 	}
 	return ret;
 }
+EXPORT_SYMBOL(cap_file_mmap);
+
+
diff -ur linux-kernel-3.0.20-ylmf1/security/device_cgroup.c linux-kernel-3.0.20-ylmf1.new//security/device_cgroup.c
--- linux-kernel-3.0.20-ylmf1/security/device_cgroup.c	2012-02-07 01:31:45.000000000 +0800
+++ linux-kernel-3.0.20-ylmf1.new//security/device_cgroup.c	2012-04-25 14:26:56.278315677 +0800
@@ -508,6 +508,7 @@
 
 	return -EPERM;
 }
+EXPORT_SYMBOL(__devcgroup_inode_permission);
 
 int devcgroup_inode_mknod(int mode, dev_t dev)
 {
diff -ur linux-kernel-3.0.20-ylmf1/security/security.c linux-kernel-3.0.20-ylmf1.new//security/security.c
--- linux-kernel-3.0.20-ylmf1/security/security.c	2012-02-07 01:31:45.000000000 +0800
+++ linux-kernel-3.0.20-ylmf1.new//security/security.c	2012-04-25 14:31:05.566901283 +0800
@@ -373,6 +373,7 @@
 		return 0;
 	return security_ops->path_rmdir(dir, dentry);
 }
+EXPORT_SYMBOL(security_path_rmdir);
 
 int security_path_unlink(struct path *dir, struct dentry *dentry)
 {
@@ -389,6 +390,7 @@
 		return 0;
 	return security_ops->path_symlink(dir, dentry, old_name);
 }
+EXPORT_SYMBOL(security_path_symlink);
 
 int security_path_link(struct dentry *old_dentry, struct path *new_dir,
 		       struct dentry *new_dentry)
@@ -397,6 +399,7 @@
 		return 0;
 	return security_ops->path_link(old_dentry, new_dir, new_dentry);
 }
+EXPORT_SYMBOL(security_path_link);
 
 int security_path_rename(struct path *old_dir, struct dentry *old_dentry,
 			 struct path *new_dir, struct dentry *new_dentry)
@@ -415,6 +418,7 @@
 		return 0;
 	return security_ops->path_truncate(path);
 }
+EXPORT_SYMBOL(security_path_truncate);
 
 int security_path_chmod(struct dentry *dentry, struct vfsmount *mnt,
 			mode_t mode)
@@ -423,6 +427,7 @@
 		return 0;
 	return security_ops->path_chmod(dentry, mnt, mode);
 }
+EXPORT_SYMBOL(security_path_chmod);
 
 int security_path_chown(struct path *path, uid_t uid, gid_t gid)
 {
@@ -430,6 +435,7 @@
 		return 0;
 	return security_ops->path_chown(path, uid, gid);
 }
+EXPORT_SYMBOL(security_path_chown);
 
 int security_path_chroot(struct path *path)
 {
@@ -506,6 +512,7 @@
 		return 0;
 	return security_ops->inode_readlink(dentry);
 }
+EXPORT_SYMBOL(security_inode_readlink);
 
 int security_inode_follow_link(struct dentry *dentry, struct nameidata *nd)
 {
@@ -520,6 +527,7 @@
 		return 0;
 	return security_ops->inode_permission(inode, mask, 0);
 }
+EXPORT_SYMBOL(security_inode_permission);
 
 int security_inode_exec_permission(struct inode *inode, unsigned int flags)
 {
@@ -626,6 +634,7 @@
 
 	return fsnotify_perm(file, mask);
 }
+EXPORT_SYMBOL(security_file_permission);
 
 int security_file_alloc(struct file *file)
 {
@@ -653,6 +662,7 @@
 		return ret;
 	return ima_file_mmap(file, prot);
 }
+EXPORT_SYMBOL(security_file_mmap);
 
 int security_file_mprotect(struct vm_area_struct *vma, unsigned long reqprot,
 			    unsigned long prot)
