diff -ur fcitx-configtool-0.4.4/CMakeLists.txt fcitx-configtool-0.4.4.new//CMakeLists.txt
--- fcitx-configtool-0.4.4/CMakeLists.txt	2012-06-02 16:14:45.000000000 +0000
+++ fcitx-configtool-0.4.4.new//CMakeLists.txt	2012-06-18 12:27:26.180745961 +0000
@@ -2,8 +2,8 @@
 
 project(fcitx-configtool)
 
-option(ENABLE_GTK2 "Enable GTK2 Version" Off)
-option(ENABLE_GTK3 "Enable GTK3 Version" On)
+option(ENABLE_GTK2 "Enable GTK2 Version" On)
+option(ENABLE_GTK3 "Enable GTK3 Version" Off)
 
 find_package(PkgConfig REQUIRED)
 
