diff -ur deepinwine-qq2012-0.0.5/data/usr/bin/wine-qq2012 deepinwine-qq2012-0.0.5.new/data/usr/bin/wine-qq2012
--- deepinwine-qq2012-0.0.5/data/usr/bin/wine-qq2012	2012-07-24 10:39:20.000000000 +0800
+++ deepinwine-qq2012-0.0.5.new/data/usr/bin/wine-qq2012	2012-12-18 13:53:46.149575401 +0800
@@ -35,6 +35,9 @@
 	else
 		mkdir -p $WINEPREFIX
 		tar -xzf $APPDIR/$APPTAR -C $WINEPREFIX
+		# regedit import winebrowser.reg
+		env WINEPREFIX=$WINEPREFIX $WINEDIR/bin/regedit $APPDIR/winebrowser.reg
+		#
 		env WINEPREFIX=$WINEPREFIX $WINEDIR/bin/wine "$EXECUTE"
 	fi
 }
diff -ur deepinwine-qq2012-0.0.5/data/usr/share/deepinwine/qq2012/wine-qq2012 deepinwine-qq2012-0.0.5.new/data/usr/share/deepinwine/qq2012/wine-qq2012
--- deepinwine-qq2012-0.0.5/data/usr/share/deepinwine/qq2012/wine-qq2012	2012-07-24 10:39:20.000000000 +0800
+++ deepinwine-qq2012-0.0.5.new/data/usr/share/deepinwine/qq2012/wine-qq2012	2012-12-18 13:53:46.149575401 +0800
@@ -35,6 +35,9 @@
 	else
 		mkdir -p $WINEPREFIX
 		tar -xzf $APPDIR/$APPTAR -C $WINEPREFIX
+		# regedit import winebrowser.reg
+		env WINEPREFIX=$WINEPREFIX $WINEDIR/bin/regedit $APPDIR/winebrowser.reg
+		#
 		env WINEPREFIX=$WINEPREFIX $WINEDIR/bin/wine "$EXECUTE"
 	fi
 }
