--- nautilus-gksu/libnautilus-gksu.c	2009-02-23 10:03:37.000000000 +0800
+++ nautilus-gksu/libnautilus-gksu.c.new	2011-03-08 15:55:21.113239002 +0800
@@ -240,6 +240,13 @@
   if (tmp)
     *tmp = '\0';
 
+   /* 
+    * Sometimes 'cmd' contains extra blank space at the end which leads
+    * to problem with some programs(e.g. file-roller opens extra window
+    * and steals focus).
+    */
+   cmd = g_strstrip(cmd);
+
   if (is_desktop)
     full_cmd = cmd;
   else
