diff -ur lovewallpaper-1.5.2-r1/data/usr/share/pyshared/lovewallpaper/Plugin/Gnome.py lovewallpaper-1.5.2-r1.new//data/usr/share/pyshared/lovewallpaper/Plugin/Gnome.py
--- lovewallpaper-1.5.2-r1/data/usr/share/pyshared/lovewallpaper/Plugin/Gnome.py	2012-09-23 12:58:23.000000000 +0000
+++ lovewallpaper-1.5.2-r1.new//data/usr/share/pyshared/lovewallpaper/Plugin/Gnome.py	2012-10-08 04:55:15.819723687 +0000
@@ -16,10 +16,12 @@
         pass
 
     def setWallpaper(self, path, key):
-        os.system( "gsettings set org.gnome.desktop.background picture-uri \"file://%s/%s.jpg\""%(path, key))
+    	#os.system( "gsettings set org.gnome.desktop.background picture-uri \"file://%s/%s.jpg\""%(path, key))
+	os.system( "gconftool-2 --type=string --set /desktop/gnome/background/picture_filename %s/%s.jpg" %(path, key))
     
     def puresetWallpaper(self, url):
-        os.system( "gsettings set org.gnome.desktop.background picture-uri \"file://%s\""%(url))
+        #os.system( "gsettings set org.gnome.desktop.background picture-uri \"file://%s\""%(url))
+	os.system( "gconftool-2 --type=string --set /desktop/gnome/background/picture_filename %s" %(url))
 
 class AutoSlide:
     """docstring for AutoSlide"""
