--- nautilus-gksu/libnautilus-gksu.c	2011-04-12 20:48:40.826184002 +0800
+++ nautilus-gksu/libnautilus-gksu.c.new	2011-04-12 20:50:19.358183983 +0800
@@ -189,6 +189,7 @@
   uri = nautilus_file_info_get_uri (file);
   mime_type = nautilus_file_info_get_mime_type (file);
 
+  gchar *filename = g_filename_from_uri (uri, NULL, NULL);
   if (!strcmp (mime_type, "application/x-desktop"))
     { /* we're handling a .desktop file */
       GKeyFile *key_file = g_key_file_new ();
@@ -251,7 +252,8 @@
     full_cmd = cmd;
   else
     {
-      full_cmd = g_strdup_printf ("%s '%s'", cmd, uri);
+      //full_cmd = g_strdup_printf ("%s '%s'", cmd, uri);
+      full_cmd = g_strdup_printf ("%s '%s'", cmd, filename);
       g_free (cmd);
     }
 
