diff -ur w3m-0.5.3/istream.c w3m-0.5.3.new//istream.c
--- w3m-0.5.3/istream.c	2011-01-04 17:22:22.000000000 +0800
+++ w3m-0.5.3.new//istream.c	2011-11-17 10:41:40.087116262 +0800
@@ -40,7 +40,7 @@
 {
     int len;
     base->stream.cur = base->stream.next = 0;
-    len = base->read(base->handle, base->stream.buf, base->stream.size);
+    len = (base->read) (base->handle, base->stream.buf, base->stream.size);
     if (len <= 0)
 	base->iseos = TRUE;
     else
@@ -323,7 +323,7 @@
     len = buffer_read(&base->stream, buf->ptr, count);
     rest = count - len;
     if (MUST_BE_UPDATED(base)) {
-	len = base->read(base->handle, &buf->ptr[len], rest);
+    	len = (base->read) (base->handle, &buf->ptr[len], rest);
 	if (len <= 0) {
 	    base->iseos = TRUE;
 	    len = 0;
