diff -ur omnibook-20110718/Makefile omnibook-20110718.new//Makefile
--- omnibook-20110718/Makefile	2011-07-18 14:23:12.444408289 +0800
+++ omnibook-20110718.new//Makefile	2011-11-03 14:03:14.498174378 +0800
@@ -48,8 +48,7 @@
 KVER	= $(KVERS)
 VMODDIR = $(MODDIR)/$(KVER)
 INSTDIR	= extra
-#KSRC	= /usr/src/linux
-KSRC	= $(VMODDIR)/build
+KSRC	= /usr/src/linux
 KMODDIR	= $(KSRC)/drivers/misc/omnibook
 KDOCDIR	= $(KSRC)/Documentation/omnibook
 PWD	= $(shell pwd)
