diff -ur rt5370sta-2.5.0.2//Makefile w311m_linux//Makefile
--- rt5370sta-2.5.0.2//Makefile	2011-04-07 09:46:49.000000000 +0800
+++ w311m_linux//Makefile	2011-10-31 15:14:47.803716106 +0800
@@ -1,3 +1,4 @@
+MODE = STA
 ifeq ($(WIFI_MODE),)
 RT28xx_MODE = STA
 else
@@ -447,6 +448,7 @@
 	$(MAKE) -C $(RT28xx_DIR)/os/linux -f Makefile.4 install
 else
 	$(MAKE) -C $(RT28xx_DIR)/os/linux -f Makefile.6 install
+	cp common/rt2870.bin /lib/firmware/
 endif
 endif
 
diff -ur rt5370sta-2.5.0.2//os/linux/config.mk w311m_linux//os/linux/config.mk
--- rt5370sta-2.5.0.2//os/linux/config.mk	2011-04-07 09:46:49.000000000 +0800
+++ w311m_linux//os/linux/config.mk	2011-10-31 13:25:50.000000000 +0800
@@ -54,11 +54,11 @@
 HAS_APCLI=n
 
 # Support Wpa_Supplicant
-HAS_WPA_SUPPLICANT=n
+HAS_WPA_SUPPLICANT=y
 
 
 # Support Native WpaSupplicant for Network Maganger
-HAS_NATIVE_WPA_SUPPLICANT_SUPPORT=n
+HAS_NATIVE_WPA_SUPPLICANT_SUPPORT=y
 
 #Support Net interface block while Tx-Sw queue full
 HAS_BLOCK_NET_IF=n
diff -ur rt5370sta-2.5.0.2//os/linux/usb_main_dev.c w311m_linux//os/linux/usb_main_dev.c
--- rt5370sta-2.5.0.2//os/linux/usb_main_dev.c	2011-04-07 09:46:49.000000000 +0800
+++ w311m_linux//os/linux/usb_main_dev.c	2011-10-31 14:33:31.000000000 +0800
@@ -38,6 +38,7 @@
 /* Otherwise post to forum in ralinktech's web site(www.ralinktech.com) and let all users help you. *** */
 MODULE_AUTHOR("Paul Lin <paul_lin@ralinktech.com>");
 MODULE_DESCRIPTION("RT2870 Wireless Lan Linux Driver");
+MODULE_LICENSE("GPL");
 #ifdef CONFIG_STA_SUPPORT
 #ifdef MODULE_VERSION
 MODULE_VERSION(STA_DRIVER_VERSION);
Binary files rt5370sta-2.5.0.2//tools/bin2h and w311m_linux//tools/bin2h differ
