--- src/ui/classic/fcitx-classic-ui.desc	2012-07-17 20:21:04.000000000 +0000
+++ src/ui/classic/fcitx-classic-ui.desc.new	2012-08-02 13:14:28.212393408 +0000
@@ -1,12 +1,12 @@
 [ClassicUI/MainWindowOffsetX]
 Description=Main Window Position X
 Type=Integer
-DefaultValue=500
+DefaultValue=1000
 
 [ClassicUI/MainWindowOffsetY]
 Description=Main Window Position Y
 Type=Integer
-DefaultValue=40
+DefaultValue=700
 
 [ClassicUI/Font]
 Type=Font
@@ -44,8 +44,8 @@
 
 [ClassicUI/SkinType]
 Type=String
-DefaultValue=default
-Description=Skin Name
+DefaultValue=startos
+Description=StartOS skin for fcitx
 
 [DescriptionFile]
 LocaleDomain=fcitx
