--- a/gui/simple-greeter/Makefile.am	2010-04-27 01:08:42.000000000 +0800
+++ b/gui/simple-greeter/Makefile.am.new	2010-11-24 15:28:21.377000053 +0800
@@ -10,6 +10,7 @@
 	-DDMCONFDIR=\""$(dmconfdir)"\"			\
 	-DGDMCONFDIR=\"$(gdmconfdir)\"                  \
 	-DDATADIR=\""$(datadir)"\"		 	\
+	-DCACHEDIR=\""$(GDM_WORKING_DIR)/.cache"\"      \
 	-DSYSCONFDIR=\""$(sysconfdir)"\"		\
 	-DLIBLOCALEDIR=\""$(prefix)/lib/locale"\"	\
 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\" 	\
