diff -ruN gdm-2.32.1-ylmf5/utils/gdmflexiserver.c gdm-2.32.1-ylmf5-edit/utils/gdmflexiserver.c
--- gdm-2.32.1-ylmf5/utils/gdmflexiserver.c	2011-03-29 00:48:50.000000000 +0800
+++ gdm-2.32.1-ylmf5-edit/utils/gdmflexiserver.c	2012-07-10 14:54:43.219907145 +0800
@@ -157,20 +157,23 @@
                           GError        **error)
 {
         DBusError       local_error;
-        DBusMessage    *message;
+        DBusMessage    *message, *message_bak;
         DBusMessage    *reply;
         gboolean        ret;
         DBusMessageIter iter;
         const char     *value;
 
-        ret = FALSE;
-        reply = NULL;
+        ret     = FALSE;
+        reply   = NULL;
+        message     = NULL;
+        message_bak = NULL;
 
         dbus_error_init (&local_error);
         message = dbus_message_new_method_call (GDM_DBUS_NAME,
                                                 GDM_DBUS_LOCAL_DISPLAY_FACTORY_PATH,
                                                 GDM_DBUS_LOCAL_DISPLAY_FACTORY_INTERFACE,
                                                 "CreateTransientDisplay");
+
         if (message == NULL) {
                 g_set_error (error, GDM_FLEXISERVER_ERROR, 0, "Out of memory.");
                 goto out;
@@ -186,7 +189,7 @@
                         g_warning ("Unable to create transient display: %s", local_error.message);
                         g_set_error (error, GDM_FLEXISERVER_ERROR, 0, "%s", local_error.message);
                         dbus_error_free (&local_error);
-                        goto out;
+                        goto bak_switch;
                 }
         }
 
@@ -195,6 +198,24 @@
         g_debug ("Started %s", value);
 
         ret = TRUE;
+        goto out;
+
+ bak_switch:
+        /*
+         * This app belong to gdm, We patch it just for LightDM can switch user with it,
+         * AND We re-implement lightdmflexiserver in lightdm, SO here all error messages
+         * we will ignore.
+         */
+        message_bak = dbus_message_new_method_call ("org.freedesktop.DisplayManager",
+                                                "/org/freedesktop/DisplayManager/Seat0",
+                                                "org.freedesktop.DisplayManager.Seat",
+                                                "SwitchToGreeter");
+        if (message_bak == NULL) goto out;
+        if (!dbus_connection_send(connection, message_bak, NULL))
+            goto out;
+        g_error_free (*error);
+
+        ret = TRUE;
  out:
         if (message != NULL) {
                 dbus_message_unref (message);
@@ -202,7 +223,9 @@
         if (reply != NULL) {
                 dbus_message_unref (reply);
         }
-
+        if (message_bak != NULL) {
+                dbus_message_unref (message_bak);
+        }
         return ret;
 }
 
