diff -ur gdm-2.32.1-ylmf1/data/gdm gdm-2.32.1-ylmf1.new/data/gdm
--- gdm-2.32.1-ylmf1/data/gdm	2011-03-08 01:40:37.000000000 +0800
+++ gdm-2.32.1-ylmf1.new/data/gdm	2011-10-27 17:06:36.537381487 +0800
@@ -1,5 +1,7 @@
 #%PAM-1.0
 auth       required    pam_env.so
+auth       required    pam_unix.so nullok 
+auth       optional    pam_gnome_keyring.so try_first_pass
 auth       required    pam_succeed_if.so user != root quiet
 auth       sufficient  pam_succeed_if.so user ingroup nopasswdlogin
 auth       include     system-auth
@@ -10,3 +12,4 @@
 session    include     system-auth
 session    required    pam_loginuid.so
 session    optional    pam_console.so
+session    optional    pam_gnome_keyring.so auto_start 
