diff -ur gdm-2.32.1-ylmf3/data/gdm gdm-2.32.1-ylmf3.new//data/gdm
--- gdm-2.32.1-ylmf3/data/gdm	2012-03-01 15:07:28.904892095 +0800
+++ gdm-2.32.1-ylmf3.new//data/gdm	2012-03-01 15:08:21.925881996 +0800
@@ -11,5 +11,4 @@
 session    optional    pam_keyinit.so force revoke
 session    include     system-auth
 session    required    pam_loginuid.so
-session    optional    pam_console.so
 session    optional    pam_gnome_keyring.so auto_start 
diff -ur gdm-2.32.1-ylmf3/data/gdm-autologin gdm-2.32.1-ylmf3.new//data/gdm-autologin
--- gdm-2.32.1-ylmf3/data/gdm-autologin	2011-03-08 01:40:37.000000000 +0800
+++ gdm-2.32.1-ylmf3.new//data/gdm-autologin	2012-03-01 15:08:32.732750565 +0800
@@ -7,4 +7,3 @@
 session    optional    pam_keyinit.so force revoke
 session    include     system-auth
 session    required    pam_loginuid.so
-session    optional    pam_console.so
