diff -ur x86-mingw32-build-1.0/x86-mingw32-build.sh x86-mingw32-build-1.0.new//x86-mingw32-build.sh
--- x86-mingw32-build-1.0/x86-mingw32-build.sh	2009-03-16 07:28:22.000000000 +0800
+++ x86-mingw32-build-1.0.new//x86-mingw32-build.sh	2011-06-01 15:52:21.499949229 +0800
@@ -35,7 +35,8 @@
 # compiler tool suite, and the operating system element, which may be used
 # to qualify source package names.
 #
-TARGET=${1-${TARGET-${TARGET_CPU-"i386"}-${TARGET_OS="mingw32"}}}
+#TARGET=${1-${TARGET-${TARGET_CPU-"i386"}-${TARGET_OS="mingw32"}}}
+TARGET="i586-mingw32"
 TARGET_OS=${TARGET_OS-"`echo $TARGET | sed 's,^.*-mingw,mingw,'`"}
 
 # Check how the user wants to progress the build, (default is `interactive'),
