diff -ur nautilus-2.30.1-ylmf2/src/file-manager/fm-directory-view.c nautilus-2.30.1-ylmf2.new/src/file-manager/fm-directory-view.c
--- nautilus-2.30.1-ylmf2/src/file-manager/fm-directory-view.c	2010-04-26 15:50:42.000000000 +0800
+++ nautilus-2.30.1-ylmf2.new/src/file-manager/fm-directory-view.c	2011-05-18 16:02:49.841437371 +0800
@@ -277,6 +277,9 @@
 static void     fm_directory_view_create_links_for_files       (FMDirectoryView      *view,
 								GList                *files,
 								GArray               *item_locations);
+static void     fm_directory_view_desktop_links_for_files (FMDirectoryView *view,
+                                				GList *files,
+                                				GArray *relative_item_points);
 static void     trash_or_delete_files                          (GtkWindow            *parent_window,
 								const GList          *files,
 								gboolean              delete_if_all_already_in_trash,
@@ -1103,6 +1106,28 @@
 }
 
 static void
+action_desktop_link_callback (GtkAction *action,
+                            gpointer callback_data)
+{
+    FMDirectoryView *view;
+    GList *selection;
+    GArray *selected_item_locations;
+
+    g_assert (FM_IS_DIRECTORY_VIEW (callback_data));
+
+    view = FM_DIRECTORY_VIEW (callback_data);
+    selection = fm_directory_view_get_selection (view);
+    if (selection_not_empty_in_menu_callback (view, selection)) {
+        selected_item_locations = fm_directory_view_get_selected_icon_locations (view);
+        fm_directory_view_desktop_links_for_files (view, selection, selected_item_locations);
+        g_array_free (selected_item_locations, TRUE);
+    }
+
+    nautilus_file_list_free (selection);
+}
+
+
+static void
 action_select_all_callback (GtkAction *action, 
 			    gpointer callback_data)
 {
@@ -1429,6 +1454,25 @@
 	g_free (parent_uri);
 }
 
+static gboolean
+action_computer_icon(GList *files)
+{
+       gboolean ret = FALSE;
+       gsize length;
+       char *file_uris;
+
+       length = g_list_length(files);
+       if(length == 1){
+               file_uris = nautilus_file_get_uri(files->data);
+               if(strcmp(file_uris,"x-nautilus-desktop:///computer") == 0 ||
+               strcmp(file_uris,"computer:///") == 0){
+                       ret = g_spawn_command_line_async("gnome-system-monitor -s",NULL);
+               }
+               g_free(file_uris);
+       }
+       return ret;
+}
+
 static void
 action_properties_callback (GtkAction *action,
 			    gpointer callback_data)
@@ -1441,7 +1485,8 @@
         view = FM_DIRECTORY_VIEW (callback_data);
 	selection = fm_directory_view_get_selection (view);
 
-	fm_properties_window_present (selection, GTK_WIDGET (view));
+	if(!action_computer_icon(selection))
+		fm_properties_window_present (selection, GTK_WIDGET (view));
 
         nautilus_file_list_free (selection);
 }
@@ -1460,7 +1505,8 @@
 	if (view->details->directory_as_file != NULL) {
 		files = g_list_append (NULL, nautilus_file_ref (view->details->directory_as_file));
 
-		fm_properties_window_present (files, GTK_WIDGET (view));
+		if(!action_computer_icon(files))
+			fm_properties_window_present (files, GTK_WIDGET (view));
 
 		nautilus_file_list_free (files);
 	}
@@ -3685,6 +3731,41 @@
 }
 
 static void
+fm_directory_view_desktop_links_for_files (FMDirectoryView *view, GList *files,
+                                         GArray *relative_item_points)
+{
+       GList *uris;
+       char *dir_uri;
+       CopyMoveDoneData *copy_move_done_data;
+       g_assert (relative_item_points->len == 0
+                 || g_list_length (files) == relative_item_points->len);
+       
+        g_assert (FM_IS_DIRECTORY_VIEW (view));
+        g_assert (files != NULL);
+
+       /* create a list of URIs */
+       uris = NULL;
+       g_list_foreach (files, prepend_uri_one, &uris);
+       uris = g_list_reverse (uris);
+
+        g_assert (g_list_length (uris) == g_list_length (files));
+
+       /* offset the drop locations a bit so that we don't pile
+        * up the icons on top of each other
+        */
+       offset_drop_points (relative_item_points,
+                           DUPLICATE_HORIZONTAL_ICON_OFFSET,
+                           DUPLICATE_VERTICAL_ICON_OFFSET);
+
+        copy_move_done_data = pre_copy_move (view);
+       dir_uri = g_filename_to_uri(g_get_user_special_dir(G_USER_DIRECTORY_DESKTOP),NULL,NULL);
+       nautilus_file_operations_copy_move (uris, relative_item_points, dir_uri, GDK_ACTION_LINK, 
+                                           GTK_WIDGET (view), copy_move_done_callback, copy_move_done_data);
+       g_free (dir_uri);
+       eel_g_list_free_deep (uris);
+}
+
+static void
 fm_directory_view_duplicate_selection (FMDirectoryView *view, GList *files,
 				       GArray *relative_item_points)
 {
@@ -7282,6 +7363,10 @@
   /* label, accelerator */       N_("Ma_ke Link"), "<control>M",
   /* tooltip */                  N_("Create a symbolic link for each selected item"),
                                  G_CALLBACK (action_create_link_callback) },
+  /* name, stock id */         { "Desktop Link", NULL,
+  /* label, accelerator */       N_("Ma_ke Link on desktop"), "<control>L",
+  /* tooltip */                  N_("Create a symbolic link for each selected item on desktop"),
+                                 G_CALLBACK (action_desktop_link_callback) },
   /* name, stock id */         { "Rename", NULL,
   /* label, accelerator */       N_("_Rename..."), "F2",
   /* tooltip */                  N_("Rename selected item"),
@@ -8652,6 +8737,12 @@
 	return FALSE;
 }
 
+static gboolean
+fm_directory_view_is_desktop(FMDirectoryView *view)
+{
+       return (nautilus_window_info_get_window_type(view->details->window) == NAUTILUS_WINDOW_DESKTOP);
+}
+
 static void
 real_update_menus (FMDirectoryView *view)
 {
@@ -8707,6 +8798,10 @@
 	gtk_action_set_sensitive (action,
 				  selection_count == 1 &&
 				  fm_directory_view_can_rename_file (view, selection->data));
+	
+	if(fm_directory_view_is_desktop(view)){
+	               gtk_action_set_visible(action,gtk_action_get_sensitive(action));
+	      }
 
 	action = gtk_action_group_get_action (view->details->dir_action_group,
 					      FM_ACTION_NEW_FOLDER);
@@ -8901,6 +8996,12 @@
 		      NULL);
 	gtk_action_set_sensitive (action, can_delete_files);
 
+	if(fm_directory_view_is_desktop(view)){
+	              gtk_action_set_visible (action, can_delete_files);
+	               if(!can_delete_files)
+	                       show_separate_delete_command = can_delete_files;
+	}
+
 	action = gtk_action_group_get_action (view->details->dir_action_group,
 					      FM_ACTION_DELETE);
 	gtk_action_set_visible (action, show_separate_delete_command);
@@ -8930,6 +9031,13 @@
 			      	"Ma_ke Links",
 				selection_count),
 		      NULL);
+
+	gtk_action_set_visible(action,can_link_files);
+	
+	action = gtk_action_group_get_action (view->details->dir_action_group,
+	                                             "Desktop Link");
+	gtk_action_set_sensitive (action, can_link_files);
+	gtk_action_set_visible(action,can_link_files);
 	
 	action = gtk_action_group_get_action (view->details->dir_action_group,
 					      FM_ACTION_PROPERTIES);
diff -ur nautilus-2.30.1-ylmf2/src/file-manager/fm-icon-view.c nautilus-2.30.1-ylmf2.new/src/file-manager/fm-icon-view.c
--- nautilus-2.30.1-ylmf2/src/file-manager/fm-icon-view.c	2010-01-22 23:45:17.000000000 +0800
+++ nautilus-2.30.1-ylmf2.new/src/file-manager/fm-icon-view.c	2011-05-18 16:05:28.808101931 +0800
@@ -1493,14 +1493,14 @@
 
 static const GtkActionEntry icon_view_entries[] = {
   /* name, stock id, label */  { "Arrange Items", NULL, N_("Arran_ge Items") }, 
-  /* name, stock id */         { "Stretch", NULL,
-  /* label, accelerator */       N_("Stretc_h Icon..."), NULL,
-  /* tooltip */                  N_("Make the selected icon stretchable"),
-                                 G_CALLBACK (action_stretch_callback) },
-  /* name, stock id */         { "Unstretch", NULL,
-  /* label, accelerator */       N_("Restore Icons' Original Si_zes"), NULL,
-  /* tooltip */                  N_("Restore each selected icon to its original size"),
-                                 G_CALLBACK (action_unstretch_callback) },
+// /* name, stock id */         { "Stretch", NULL,
+//  /* label, accelerator */       N_("Stretc_h Icon..."), NULL,
+//  /* tooltip */                  N_("Make the selected icon stretchable"),
+//                                 G_CALLBACK (action_stretch_callback) },
+//  /* name, stock id */         { "Unstretch", NULL,
+//  /* label, accelerator */       N_("Restore Icons' Original Si_zes"), NULL,
+//  /* tooltip */                  N_("Restore each selected icon to its original size"),
+//                                 G_CALLBACK (action_unstretch_callback) },
   /* name, stock id */         { "Clean Up", NULL,
   /* label, accelerator */       N_("Clean _Up by Name"), NULL,
   /* tooltip */                  N_("Reposition icons to better fit in the window and avoid overlapping"),
diff -ur nautilus-2.30.1-ylmf2/src/file-manager/fm-properties-window.c nautilus-2.30.1-ylmf2.new/src/file-manager/fm-properties-window.c
--- nautilus-2.30.1-ylmf2/src/file-manager/fm-properties-window.c	2010-03-10 21:28:38.000000000 +0800
+++ nautilus-2.30.1-ylmf2.new/src/file-manager/fm-properties-window.c	2011-05-18 16:06:09.744768056 +0800
@@ -87,7 +87,7 @@
 static GHashTable *windows;
 static GHashTable *pending_lists;
 
-struct FMPropertiesWindowDetails {	
+struct FMPropertiesWindowDetails {
 	GList *original_files;
 	GList *target_files;
 	
diff -ur nautilus-2.30.1-ylmf2/src/file-manager/nautilus-directory-view-ui.xml nautilus-2.30.1-ylmf2.new/src/file-manager/nautilus-directory-view-ui.xml
--- nautilus-2.30.1-ylmf2/src/file-manager/nautilus-directory-view-ui.xml	2011-05-18 15:44:36.404785192 +0800
+++ nautilus-2.30.1-ylmf2.new/src/file-manager/nautilus-directory-view-ui.xml	2011-05-18 16:09:13.278098957 +0800
@@ -66,6 +66,7 @@
 		<placeholder name="File Items Placeholder">
 			<menuitem name="Duplicate" action="Duplicate"/>
 			<menuitem name="Create Link" action="Create Link"/>
+			<menuitem name="Desktop Link" action="Desktop Link"/>
 			<menuitem name="Rename" action="Rename"/>
 			<menu action="CopyToMenu">
 				<menuitem name="Copy to next pane" action="Copy to next pane"/>
@@ -158,13 +159,15 @@
 	<separator name="File actions separator"/>
 	<placeholder name="File Actions">
 		<menuitem name="Create Link" action="Create Link"/>
+		<menuitem name="Desktop Link" action="Desktop Link"/>
 		<menuitem name="Rename" action="Rename"/>
-		<menu action="CopyToMenu">
+		<!-- menu action="CopyToMenu">
 			<menuitem name="Copy to next pane" action="Copy to next pane"/>
 		</menu>
 		<menu action="MoveToMenu">
 			<menuitem name="Move to next pane" action="Move to next pane"/>
 		</menu>
+		</menu -->
 	</placeholder>
 	<separator name="Dangerous separator"/>
 	<placeholder name="Dangerous File Actions">
