diff -ur nautilus-2.32.1/libnautilus-private/nautilus-autorun.c nautilus-2.32.1.new//libnautilus-private/nautilus-autorun.c
--- nautilus-2.32.1/libnautilus-private/nautilus-autorun.c	2010-11-13 19:59:01.000000000 +0800
+++ nautilus-2.32.1.new//libnautilus-private/nautilus-autorun.c	2011-09-14 10:18:57.681686922 +0800
@@ -1106,6 +1106,9 @@
 			if (eel_preferences_get_boolean (NAUTILUS_PREFERENCES_MEDIA_AUTOMOUNT_OPEN))
 				open_folder = TRUE;
 		}
+	       /// No matter what state it is, open_folder when NAUTILUS_PREFERENCES_MEDIA_AUTOMOUNT_OPEN == TRUE
+	        if (eel_preferences_get_boolean (NAUTILUS_PREFERENCES_MEDIA_AUTOMOUNT_OPEN))
+		        open_folder = TRUE;
 	}
 
 	/* only open the folder once.. */
diff -ur nautilus-2.32.1/po/zh_CN.po nautilus-2.32.1.new//po/zh_CN.po
--- nautilus-2.32.1/po/zh_CN.po	2010-11-13 19:59:01.000000000 +0800
+++ nautilus-2.32.1.new//po/zh_CN.po	2011-09-14 10:19:32.612599291 +0800
@@ -37,6 +37,10 @@
 msgid "Starting %s"
 msgstr "正在启动 %s"
 
+#
+msgid "_Automatic mount volumes" 
+msgstr "自动加载磁盘(_A)"
+
 #: ../cut-n-paste-code/libegg/eggdesktopfile.c:1100
 #, c-format
 msgid "Application does not accept documents on command line"
diff -ur nautilus-2.32.1/po/zh_TW.po nautilus-2.32.1.new//po/zh_TW.po
--- nautilus-2.32.1/po/zh_TW.po	2010-11-13 19:59:01.000000000 +0800
+++ nautilus-2.32.1.new//po/zh_TW.po	2011-09-14 10:19:56.456554973 +0800
@@ -35,6 +35,10 @@
 msgid "Starting %s"
 msgstr "準備啟動 %s"
 
+#
+msgid "_Automatic mount volumes"
+msgstr "自動掛載磁盤(_A)"
+
 #: ../cut-n-paste-code/libegg/eggdesktopfile.c:1100
 #, c-format
 msgid "Application does not accept documents on command line"
diff -ur nautilus-2.32.1/src/nautilus-application.c nautilus-2.32.1.new//src/nautilus-application.c
--- nautilus-2.32.1/src/nautilus-application.c	2010-11-13 19:59:01.000000000 +0800
+++ nautilus-2.32.1.new//src/nautilus-application.c	2011-09-14 10:24:08.319287566 +0800
@@ -220,6 +220,69 @@
 	g_volume_mount_finish (G_VOLUME (source_object), res, NULL);
 }
 
+/* add by Ylmf OS
+ * author wkt
+ * */
+static gboolean
+is_mounted(const char *devfile)
+{
+    FILE *fp = NULL;
+    char *line = NULL;
+    ssize_t res_n;
+    size_t n;
+    gboolean res = FALSE;
+
+    fp = fopen("/proc/mounts","r");
+    if(fp == NULL){
+        return FALSE;
+    }
+
+    while((res_n = getline(&line,&n,fp)) != -1){
+        if(res_n < 1|| line[0] != '/'){
+            continue;
+        }
+        char *pt;
+        for(pt = line;*pt;pt++){
+            if(*pt == ' ' || *pt == '\t' ||*pt == '\n'){
+                *pt = 0;
+                break;
+            }
+        }
+        if(g_strcmp0(devfile,line) == 0){
+            res = TRUE;
+            break;
+        }
+    }
+    if(line){
+        g_free(line);
+    }
+    fclose(fp);
+    return res;
+}
+ 
+static gboolean
+ylmf_g_volume_should_automount(GVolume *volume)
+{
+       char *path = NULL;
+       gboolean should_automount = TRUE;
+
+       path = g_volume_get_identifier(volume,G_VOLUME_IDENTIFIER_KIND_UNIX_DEVICE);
+       
+       if(path == NULL||
+         is_mounted(path)||
+         g_str_has_prefix(path,"/dev/loop") ||
+         g_str_has_prefix(path,"/dev/mapper/")||
+         g_str_has_prefix(path,"/dev/dm-")){
+               should_automount = FALSE;
+       }
+///    g_warning("automount %s:%d",path,should_automount);
+       g_free(path);
+       return should_automount;
+}
+/*
+ * end of adding codes
+ * */
+
 static void
 automount_all_volumes (NautilusApplication *application)
 {
@@ -233,7 +296,7 @@
 		for (l = volumes; l != NULL; l = l->next) {
 			volume = l->data;
 			
-			if (!g_volume_should_automount (volume) ||
+			if (!ylmf_g_volume_should_automount (volume) ||
 			    !g_volume_can_mount (volume)) {
 				continue;
 			}
@@ -1525,7 +1588,7 @@
 		       NautilusApplication *application)
 {
 	if (eel_preferences_get_boolean (NAUTILUS_PREFERENCES_MEDIA_AUTOMOUNT) &&
-	    g_volume_should_automount (volume) &&
+	    ylmf_g_volume_should_automount (volume) &&
 	    g_volume_can_mount (volume)) {
 		nautilus_file_operations_mount_volume (NULL, volume, TRUE);
 	} else {
diff -ur nautilus-2.32.1/src/nautilus-file-management-properties.c nautilus-2.32.1.new//src/nautilus-file-management-properties.c
--- nautilus-2.32.1/src/nautilus-file-management-properties.c	2010-11-13 19:59:01.000000000 +0800
+++ nautilus-2.32.1.new//src/nautilus-file-management-properties.c	2011-09-14 10:25:18.107500108 +0800
@@ -69,6 +69,7 @@
 #define NAUTILUS_FILE_MANAGEMENT_PROPERTIES_SHOW_HIDDEN_WIDGET "hidden_files_checkbutton"
 #define NAUTILUS_FILE_MANAGEMENT_PROPERTIES_TREE_VIEW_FOLDERS_WIDGET "treeview_folders_checkbutton"
 #define NAUTILUS_FILE_MANAGEMENT_PROPERTIES_MEDIA_AUTOMOUNT_OPEN "media_automount_open_checkbutton"
+#define NAUTILUS_FILE_MANAGEMENT_PROPERTIES_MEDIA_AUTOMOUNT "media_automount_checkbutton"
 #define NAUTILUS_FILE_MANAGEMENT_PROPERTIES_MEDIA_AUTORUN_NEVER "media_autorun_never_checkbutton"
 
 /* int enums */
@@ -716,6 +717,10 @@
 					      NAUTILUS_PREFERENCES_MEDIA_AUTORUN_NEVER);
 
 	eel_preferences_builder_connect_bool (builder,
+	                                      NAUTILUS_FILE_MANAGEMENT_PROPERTIES_MEDIA_AUTOMOUNT,
+	                                      NAUTILUS_PREFERENCES_MEDIA_AUTOMOUNT);
+
+	eel_preferences_builder_connect_bool (builder,
 					      NAUTILUS_FILE_MANAGEMENT_PROPERTIES_TRASH_CONFIRM_WIDGET,
 					      NAUTILUS_PREFERENCES_CONFIRM_TRASH);
 	eel_preferences_builder_connect_bool (builder,
diff -ur nautilus-2.32.1/src/nautilus-file-management-properties.ui nautilus-2.32.1.new//src/nautilus-file-management-properties.ui
--- nautilus-2.32.1/src/nautilus-file-management-properties.ui	2010-11-13 19:59:01.000000000 +0800
+++ nautilus-2.32.1.new//src/nautilus-file-management-properties.ui	2011-09-14 10:27:15.433434746 +0800
@@ -3038,6 +3038,24 @@
                     <property name="fill">False</property>
                   </packing>
                 </child>
+	        <child>
+	          <object class="GtkCheckButton" id="media_automount_checkbutton">
+	                     <property name="visible">True</property>
+	                     <property name="can_focus">True</property>
+	                     <property name="label" translatable="yes">_Automatic mount volumes</property>
+	                     <property name="use_underline">True</property>
+	                     <property name="relief">GTK_RELIEF_NORMAL</property>
+	                     <property name="focus_on_click">True</property>
+	                     <property name="active">False</property>
+	                     <property name="inconsistent">False</property>
+	                     <property name="draw_indicator">True</property>
+	            </object>
+	            <packing>
+	                     <property name="padding">0</property>
+	                     <property name="expand">False</property>
+	                     <property name="fill" >False</property>
+	             </packing>
+	       </child>
               </object>
               <packing>
                 <property name="tab_expand">False</property>
