diff -c -a -b -x 'Makefile*' nautilus-2.30.1-ylmf3/po/en_GB.po new/po/en_GB.po
*** nautilus-2.30.1-ylmf3/po/en_GB.po	2011-07-20 14:22:29.247566841 +0800
--- new/po/en_GB.po	2011-07-20 09:13:39.547950840 +0800
***************
*** 6603,6608 ****
--- 6603,6613 ----
  msgid "_Search"
  msgstr "_Search"
  
+ 
+ #: ../src/nautilus-navigation-window-menus.c:892
+ msgid "Show Drive Letter"
+ msgstr "Show Drive Letter"
+ 
  #: ../src/nautilus-navigation-window-pane.c:396
  msgid "_Close Tab"
  msgstr "_Close Tab"

diff -c -a -b -x 'Makefile*' nautilus-2.30.1-ylmf3/po/zh_CN.po new/po/zh_CN.po
*** nautilus-2.30.1-ylmf3/po/zh_CN.po	2011-07-20 14:22:29.280899993 +0800
--- new/po/zh_CN.po	2011-07-19 21:49:47.343402818 +0800
***************
*** 6315,6320 ****
--- 6315,6324 ----
  msgid "_Search"
  msgstr "搜索(_S)"
  
+ #: ../src/nautilus-navigation-window-menus.c:892
+ msgid "Show Drive Letter"
+ msgstr "显示盘符"
+ 
  #: ../src/nautilus-navigation-window.c:185
  msgid "Toggle between button and text-based location bar"
  msgstr "在按钮和基于文本的位置栏之间切换"

diff -c -a -b -x 'Makefile*' nautilus-2.30.1-ylmf3/po/zh_TW.po new/po/zh_TW.po
*** nautilus-2.30.1-ylmf3/po/zh_TW.po	2011-07-20 14:22:28.970901672 +0800
--- new/po/zh_TW.po	2011-07-20 09:08:41.139567458 +0800
***************
*** 6479,6484 ****
--- 6479,6488 ----
  msgid "_Search"
  msgstr "搜尋(_S)"
  
+ #: ../src/nautilus-navigation-window-menus.c:892
+ msgid "Show Drive Letter"
+ msgstr "顯示盤符"
+ 
  #: ../src/nautilus-navigation-window-pane.c:396
  msgid "_Close Tab"
  msgstr "關閉分頁(_C)"

--- nautilus-2.30.1-ylmf3/libnautilus-private/nautilus-global-preferences.c	2011-07-21 16:13:16.061969682 +0800
+++ nautilus/libnautilus-private/nautilus-global-preferences.c	2011-07-21 10:07:25.154221293 +0800
@@ -519,10 +519,16 @@
 	  default_network_link_name, g_free,
 	},
 
+	{ NAUTILUS_PREFERENCES_SHOW_DRIVE_LETTER,
+	  PREFERENCE_BOOLEAN,
+	  GINT_TO_POINTER (TRUE)
+	},
+
 	{ NAUTILUS_PREFERENCES_MEDIA_AUTOMOUNT,
 	  PREFERENCE_BOOLEAN,
 	  GINT_TO_POINTER (TRUE)
 	},
+	
 	{ NAUTILUS_PREFERENCES_MEDIA_AUTOMOUNT_OPEN,
 	  PREFERENCE_BOOLEAN,
 	  GINT_TO_POINTER (TRUE)

--- nautilus-2.30.1-ylmf3/libnautilus-private/nautilus-global-preferences.h	2011-07-21 16:13:15.948636964 +0800
+++ nautilus/libnautilus-private/nautilus-global-preferences.h	2011-07-21 10:05:53.384718450 +0800
@@ -53,6 +53,9 @@
 #define NAUTILUS_PREFERENCES_MEDIA_AUTOMOUNT	                "preferences/media_automount"
 #define NAUTILUS_PREFERENCES_MEDIA_AUTOMOUNT_OPEN		"preferences/media_automount_open"
 
+/* Show drive letter options */
+#define NAUTILUS_PREFERENCES_SHOW_DRIVE_LETTER                  "preferences/show_drive_letter"
+
 /* Autorun options */
 #define NAUTILUS_PREFERENCES_MEDIA_AUTORUN_NEVER                "preferences/media_autorun_never"
 #define NAUTILUS_PREFERENCES_MEDIA_AUTORUN_X_CONTENT_START_APP  "preferences/media_autorun_x_content_start_app"

--- nautilus-2.30.1-ylmf3/src/nautilus-navigation-window-menus.c	2011-07-21 16:13:25.235253321 +0800
+++ nautilus/src/nautilus-navigation-window-menus.c	2011-07-21 13:23:40.040431258 +0800
@@ -278,6 +278,25 @@
 	}
 }
 
+static void
+action_show_drive_letter_callback(GtkAction *action,
+				  gpointer user_data)
+{
+	NautilusNavigationWindow *window;
+
+	window = NAUTILUS_NAVIGATION_WINDOW (user_data);
+
+	if (gtk_toggle_action_get_active (GTK_TOGGLE_ACTION (action)))
+	{
+		eel_preferences_set_boolean(NAUTILUS_PREFERENCES_SHOW_DRIVE_LETTER, TRUE);
+	}
+	else
+	{
+		eel_preferences_set_boolean(NAUTILUS_PREFERENCES_SHOW_DRIVE_LETTER, FALSE);
+	}
+}
+
+
 void
 nautilus_navigation_window_update_show_hide_menu_items (NautilusNavigationWindow *window) 
 {
@@ -871,6 +890,11 @@
   /* tooltip */              N_("Open an extra folder view side-by-side"),
                              G_CALLBACK (action_split_view_callback),
   /* is_active */            FALSE },
+    /* name, stock id */     { "Show Drive Letter", NULL,
+  /* label, accelerator */   N_("Show Drive Letter"), NULL,
+  /* tooltip */              N_("Show drive letter in the sidebar"),
+                             G_CALLBACK (action_show_drive_letter_callback),
+  /* is_active */            TRUE },
 };
 
 void 

--- 2.30.1/src/nautilus-places-sidebar.c	2011-07-21 17:17:44.844344022 +0800
+++ nautilus/src/nautilus-places-sidebar.c	2011-07-21 17:11:18.853101783 +0800
@@ -48,12 +48,18 @@
 #include <libnautilus-private/nautilus-window-slot-info.h>
 #include <gio/gio.h>
 
+#include <dbus/dbus-glib.h>
+
 #include "nautilus-bookmark-list.h"
 #include "nautilus-places-sidebar.h"
 #include "nautilus-window.h"
 
 #define EJECT_BUTTON_XPAD 5
 
+#define DBUS_NAME "org.ylmf.nautilus"
+#define INTERFACE "org.ylmf.nautilus.ShowDL"
+                  
+
 typedef struct {
 	GtkScrolledWindow  parent;
 	GtkTreeView        *tree_view;
@@ -284,6 +290,13 @@
 	GFile *root;
 	NautilusWindowSlotInfo *slot;
 	char *tooltip;
+	char buf[1024];
+	char cmd[1024];
+
+	memset(buf, '\0', sizeof(buf));
+	memset(cmd, '\0', sizeof(cmd));
+
+	
 
 	selection = gtk_tree_view_get_selection (sidebar->tree_view);
 	gtk_list_store_clear (sidebar->store);
@@ -365,9 +378,104 @@
 					root = g_mount_get_default_location (mount);
 					mount_uri = g_file_get_uri (root);
 					name = g_mount_get_name (mount);
+					if (eel_preferences_get_boolean(NAUTILUS_PREFERENCES_SHOW_DRIVE_LETTER))
+					{
+						DBusGConnection *bus;
+						DBusGProxy *remote_object;
+						DBusGProxy *remote_object_introspectable;
+						GError *error = NULL;
+						char *reply;
+						char *introspect_data;
+						FILE *stream;
+
+
+						g_type_init ();
+
+						bus = dbus_g_bus_get (DBUS_BUS_SYSTEM, &error);
+						if (!bus){
+							g_error("Couldn't connect to system bus :%s", error->message);
+							g_error_free(error);
+							error = NULL;
+						}
+						/*
+						remote_object_introspectable = dbus_g_proxy_new_for_name (bus,
+													  DBUS_NAME,
+													  "/",
+													  "org.freedesktop.DBus.Introspectable");
+						if (!dbus_g_proxy_call (remote_object_introspectable, "Introspect", &error,
+									G_TYPE_INVALID,
+									G_TYPE_STRING, &introspect_data, G_TYPE_INVALID))
+							g_error("Failed to complete Introspect:%s", error->message);
+
+						g_print ("%s\n", introspect_data);
+						g_free (introspect_data);
+						*/
+
+						remote_object = dbus_g_proxy_new_for_name (bus,
+											   DBUS_NAME,
+											   "/",
+											   INTERFACE
+											   );
+						strcpy(buf, g_file_get_path(root));
+						
+						sprintf(cmd, "mount | grep '%s' | awk '{print $1}'", buf);
+						stream = popen(cmd, "r");
+
+						memset(buf, '\0', sizeof(buf));
+						memset(cmd, '\0', sizeof(cmd));
+						
+						fread(buf, sizeof(char), 9, stream);
+
+						if(buf[6] == 'd')
+						{
+						
+							dbus_g_proxy_set_default_timeout(remote_object,1000000000);
+							if (!dbus_g_proxy_call (remote_object, "showdriveletter", &error,
+										G_TYPE_STRING, buf,
+										G_TYPE_INVALID,
+										G_TYPE_STRING, &reply,
+										G_TYPE_INVALID)){
+								g_error("Failed to complete \"%s\":%s\n","org.ylmf.nautilus.interface",error->message);
+								g_error_free(error);
+								error = NULL;
+							}
+						
+							memset(buf, '\0', sizeof(buf));
+							memset(cmd, '\0', sizeof(cmd));
+							//		g_print("driveletter:%s\n",  reply);
+						}
+						else
+						{
+							reply = NULL;
+						}
+						
+						
+
+						if(reply == NULL)
+						{
+							strcpy(cmd, g_mount_get_name(mount));
+						}
+						else
+						{
+							strcpy(cmd, reply);
+						}
+												
+
+						//	g_print("driveletter:%s\n",reply);
+						g_free(reply);
+						pclose(stream);
+						g_object_unref (G_OBJECT (remote_object));
+						g_object_unref (G_OBJECT (remote_object_introspectable));
+					}
+					else
+					{
+						strcpy(cmd, g_mount_get_name(mount));
+
+					}
+					
 					tooltip = g_file_get_parse_name (root);
 					last_iter = add_place (sidebar, PLACES_MOUNTED_VOLUME,
-							       name, icon, mount_uri,
+							       cmd, icon, mount_uri,
 							       drive, volume, mount, 0, tooltip);
 					if (eel_strcmp (location, mount_uri) == 0) {
 						gtk_tree_selection_select_iter (selection, &last_iter);
@@ -2722,6 +2830,13 @@
 	g_signal_connect_object (sidebar->volume_monitor, "drive_changed",
 				 G_CALLBACK (drive_changed_callback), sidebar, 0);
 
+	eel_preferences_add_callback_while_alive(NAUTILUS_PREFERENCES_SHOW_DRIVE_LETTER,
+						 update_places,
+						 sidebar,
+						 G_OBJECT(sidebar));
+
+
+
 	update_places (sidebar);
 }
 


--- nautilus-2.30.1-ylmf3/src/nautilus-navigation-window-ui.xml	2011-07-21 16:13:24.395257870 +0800
+++ nautilus/src/nautilus-navigation-window-ui.xml	2011-07-21 10:34:29.148756692 +0800
@@ -25,6 +25,7 @@
 			<menuitem name="Show Hide Location Bar" action="Show Hide Location Bar"/>
 			<menuitem name="Show Hide Statusbar" action="Show Hide Statusbar"/>
 			<menuitem name="Show Hide Extra Pane" action="Show Hide Extra Pane"/>
+			<menuitem name="Show Drive Letter" action="Show Drive Letter"/>
 		</placeholder>
 	</menu>
         <placeholder name="Other Menus">
