diff -ur /usr/src/mingw32-build//x86-mingw32-build.sh x86-mingw32-build-1.0.1/x86-mingw32-build.sh
--- /usr/src/mingw32-build//x86-mingw32-build.sh	2010-03-11 06:16:55.000000000 +0800
+++ x86-mingw32-build-1.0.1/x86-mingw32-build.sh	2011-07-01 14:09:02.699761181 +0800
@@ -35,8 +35,11 @@
 # compiler tool suite, and the operating system element, which may be used
 # to qualify source package names.
 #
-TARGET=${1-${TARGET-${TARGET_CPU-"i386"}-${TARGET_OS="mingw32"}}}
-TARGET_OS=${TARGET_OS-"`echo $TARGET | sed 's,^.*-mingw,mingw,'`"}
+#TARGET=${1-${TARGET-${TARGET_CPU-"i386"}-${TARGET_OS="mingw32"}}}
+#TARGET_OS=${TARGET_OS-"`echo $TARGET | sed 's,^.*-mingw,mingw,'`"}
+
+TARGET=i586-mingw32msvc
+TARGET_OS=mingw32
 
 # Check how the user wants to progress the build, (default is `interactive'),
 # then hand off package selection and set-up to the appropriate helper script.
