--- opt/libreoffice/share/xdg//impress.desktop	2011-02-16 12:58:41.000000000 +0800
+++ opt/libreoffice/share/xdg.new/impress.desktop	2011-03-11 10:35:50.336996002 +0800
@@ -7,6 +7,9 @@
 Exec=libreoffice -impress %U
 MimeType=application/vnd.oasis.opendocument.presentation;application/vnd.oasis.opendocument.presentation-template;application/vnd.sun.xml.impress;application/vnd.sun.xml.impress.template;application/vnd.stardivision.impress;application/mspowerpoint;application/vnd.ms-powerpoint;application/vnd.openxmlformats-officedocument.presentationml.presentation;application/vnd.ms-powerpoint.presentation.macroenabled.12;application/vnd.openxmlformats-officedocument.presentationml.template;application/vnd.ms-powerpoint.template.macroenabled.12;
 Name=LibreOffice 3.3 Impress
+Name[zh_CN]=LibreOffice 3.3 幻灯片
+Name[zh_TW]=LibreOffice 3.3 簡報
+Name=LibreOffice 3.3 Impress
 Name[pt_BR]=BrOffice 3.3 Impress
 GenericName=Presentation
 GenericName[en]=Presentation

--- usr/share/applications/libreoffice-impress.desktop 	2011-02-16 12:58:41.000000000 +0800
+++ usr/share/applications/libreoffice-impress.desktop	2011-03-11 10:35:50.336996002 +0800
@@ -7,6 +7,9 @@
 Exec=libreoffice -impress %U
 MimeType=application/vnd.oasis.opendocument.presentation;application/vnd.oasis.opendocument.presentation-template;application/vnd.sun.xml.impress;application/vnd.sun.xml.impress.template;application/vnd.stardivision.impress;application/mspowerpoint;application/vnd.ms-powerpoint;application/vnd.openxmlformats-officedocument.presentationml.presentation;application/vnd.ms-powerpoint.presentation.macroenabled.12;application/vnd.openxmlformats-officedocument.presentationml.template;application/vnd.ms-powerpoint.template.macroenabled.12;
 Name=LibreOffice 3.3 Impress
+Name[zh_CN]=LibreOffice 3.3 幻灯片
+Name[zh_TW]=LibreOffice 3.3 簡報
+Name=LibreOffice 3.3 Impress
 Name[pt_BR]=BrOffice 3.3 Impress
 GenericName=Presentation
 GenericName[en]=Presentation
