diff -ur evolution-data-server-2.32.3-r1/addressbook/libedata-book/e-data-book.c evolution-data-server-2.32.3-r1.new//addressbook/libedata-book/e-data-book.c
--- evolution-data-server-2.32.3-r1/addressbook/libedata-book/e-data-book.c	2011-04-21 19:35:36.000000000 +0000
+++ evolution-data-server-2.32.3-r1.new//addressbook/libedata-book/e-data-book.c	2012-09-03 14:48:26.159727061 +0000
@@ -22,6 +22,7 @@
 
 #include <unistd.h>
 #include <stdlib.h>
+#undef G_DISABLE_DEPRECATED
 #include <glib/gi18n.h>
 #include <glib-object.h>
 #include <gio/gio.h>
diff -ur evolution-data-server-2.32.3-r1/addressbook/libedata-book/e-data-book-factory.c evolution-data-server-2.32.3-r1.new//addressbook/libedata-book/e-data-book-factory.c
--- evolution-data-server-2.32.3-r1/addressbook/libedata-book/e-data-book-factory.c	2011-04-21 19:36:35.000000000 +0000
+++ evolution-data-server-2.32.3-r1.new//addressbook/libedata-book/e-data-book-factory.c	2012-09-03 14:48:43.759726108 +0000
@@ -24,6 +24,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#undef G_DISABLE_DEPRECATED
 #include <glib-object.h>
 #include <glib/gi18n.h>
 #include <libebackend/e-data-server-module.h>
diff -ur evolution-data-server-2.32.3-r1/calendar/libedata-cal/e-data-cal-factory.c evolution-data-server-2.32.3-r1.new//calendar/libedata-cal/e-data-cal-factory.c
--- evolution-data-server-2.32.3-r1/calendar/libedata-cal/e-data-cal-factory.c	2011-04-21 19:36:35.000000000 +0000
+++ evolution-data-server-2.32.3-r1.new//calendar/libedata-cal/e-data-cal-factory.c	2012-09-03 14:49:04.896391631 +0000
@@ -28,6 +28,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#undef G_DISABLE_DEPRECATED
 #include <glib/gi18n.h>
 #include <glib-object.h>
 
diff -ur evolution-data-server-2.32.3-r1/libedataserver/e-data-server-util.c evolution-data-server-2.32.3-r1.new//libedataserver/e-data-server-util.c
--- evolution-data-server-2.32.3-r1/libedataserver/e-data-server-util.c	2011-04-21 19:35:37.000000000 +0000
+++ evolution-data-server-2.32.3-r1.new//libedataserver/e-data-server-util.c	2012-09-03 14:48:08.896394663 +0000
@@ -26,6 +26,8 @@
 #include <time.h>
 #include <unistd.h>
 
+#undef G_DISABLE_DEPRECATED
+
 #include <glib.h>
 
 #ifdef G_OS_WIN32
