diff -ur r8101-1.020.00/src/Makefile r8101-1.020.00.new/src/Makefile
--- r8101-1.020.00/src/Makefile	2010-09-20 17:55:57.000000000 +0800
+++ r8101-1.020.00.new/src/Makefile	2011-07-07 16:26:54.349555510 +0800
@@ -31,7 +31,7 @@
 
 KVER		:= $(shell uname -r)
 KDIR		:= /lib/modules/$(KVER)/build
-KMISC		:= /lib/modules/$(KVER)/kernel/drivers/net/
+KMISC		:= $(DESTDIR)/lib/modules/$(KVER)/kernel/drivers/net/
 KEXT		:= $(shell echo $(KVER) | sed -ne 's/^2\.[567]\..*/k/p')o
 KFLAG		:= 2$(shell echo $(KVER) | sed -ne 's/^2\.[4]\..*/4/p')x
 
@@ -51,6 +51,7 @@
 
 
 install:
+	install -d $(KMISC)
 	install -m 744 -c r8101.$(KEXT) $(KMISC)
 
 ifneq ($(KFLAG),24x)
