diff -ur emr-3.0/emr.desktop.in emr-3.0.new//emr.desktop.in
--- emr-3.0/emr.desktop.in	2006-11-25 12:55:38.000000000 +0800
+++ emr-3.0.new//emr.desktop.in	2012-04-19 10:52:58.754403073 +0800
@@ -2,9 +2,9 @@
 Encoding=UTF-8
 Version=1.0
 Type=Application
-Exec=@BINDIR@/emr
-Icon=emr.png
-Path=@DATADIR@/emr
+Exec=emr
+Icon=emr
+Path=/usr/share/emr
 Terminal=0
 
 Name=Excalibur: Morgana's Revenge
diff -ur emr-3.0/emr.in emr-3.0.new//emr.in
--- emr-3.0/emr.in	2007-06-02 17:08:04.000000000 +0800
+++ emr-3.0.new//emr.in	2012-04-19 10:54:27.982936624 +0800
@@ -48,7 +48,7 @@
 fi
 
 # Default locations of EMR and Aleph One data files.
-test -z "$ALEPHONE_DATA" && ALEPHONE_DATA='@DATADIR@/emr'
+test -z "$ALEPHONE_DATA" && ALEPHONE_DATA='/usr/share/emr'
 export ALEPHONE_DATA
 
 exec @ALEPHONE@ "$@"
diff -ur emr-3.0/Makefile emr-3.0.new//Makefile
--- emr-3.0/Makefile	2007-01-26 11:30:54.000000000 +0800
+++ emr-3.0.new//Makefile	2012-04-19 10:52:58.754403073 +0800
@@ -2,7 +2,7 @@
 # Excalibur: Morgana's Revenge will be installed.
 
 # The directory under which most files are installed.
-PREFIX=/usr/local
+PREFIX=$(DESTDIR)/usr
 
 # The directory where user programs are installed.  "emr" goes there.
 BINDIR=$(PREFIX)/bin
@@ -16,11 +16,11 @@
 EMRDOCDIR=$(DOCDIR)/emr
 
 # The directory where man pages are installed.
-MANDIR=$(PREFIX)/man
+MANDIR=$(PREFIX)/share/man
 MAN6DIR=$(MANDIR)/man6
 
 # The path to the actual alephone executable.
-ALEPHONE=/usr/local/bin/alephone
+ALEPHONE=/usr/bin/alephone
 
 # ----------------
 CP=/bin/cp
@@ -53,14 +53,15 @@
 	$(INSTALLDATA) emr.png $(ICONDIR)
 
 installdirs:
-	test -e $(MANDIR) || mkdir $(MANDIR)
-	test -e $(MAN6DIR) || mkdir $(MAN6DIR)
-	test -e $(DATADIR) || mkdir $(DATADIR)
-	test -e $(DESKTOPDIR) || mkdir $(DESKTOPDIR)
-	test -e $(ICONDIR) || mkdir $(ICONDIR)
-	test -e $(SCENARIODIR) || mkdir $(SCENARIODIR)
-	test -e $(DOCDIR) || mkdir $(DOCDIR)
-	test -e $(EMRDOCDIR) || mkdir $(EMRDOCDIR)
+	test -e $(MANDIR) || mkdir -p $(MANDIR)
+	test -e $(BINDIR) || mkdir -p $(BINDIR)
+	test -e $(MAN6DIR) || mkdir -p $(MAN6DIR)
+	test -e $(DATADIR) || mkdir -p $(DATADIR)
+	test -e $(DESKTOPDIR) || mkdir -p $(DESKTOPDIR)
+	test -e $(ICONDIR) || mkdir -p $(ICONDIR)
+	test -e $(SCENARIODIR) || mkdir -p $(SCENARIODIR)
+	test -e $(DOCDIR) || mkdir -p $(DOCDIR)
+	test -e $(EMRDOCDIR) || mkdir -p $(EMRDOCDIR)
 
 uninstall:
 	-$(RM) $(BINDIR)/emr
