--- opt/libreoffice/share/xdg//writer.desktop	2011-02-16 12:58:41.000000000 +0800
+++ opt/libreoffice/share/xdg.new/writer.desktop	2011-03-11 10:39:48.004996002 +0800
@@ -7,6 +7,9 @@
 Exec=libreoffice -writer %U
 MimeType=application/vnd.oasis.opendocument.text;application/vnd.oasis.opendocument.text-template;application/vnd.oasis.opendocument.text-web;application/vnd.oasis.opendocument.text-master;application/vnd.sun.xml.writer;application/vnd.sun.xml.writer.template;application/vnd.sun.xml.writer.global;application/vnd.stardivision.writer;application/msword;application/vnd.ms-word;application/x-doc;application/rtf;text/rtf;application/vnd.wordperfect;application/wordperfect;application/vnd.openxmlformats-officedocument.wordprocessingml.document;application/vnd.ms-word.document.macroenabled.12;application/vnd.openxmlformats-officedocument.wordprocessingml.template;application/vnd.ms-word.template.macroenabled.12;
 Name=LibreOffice 3.3 Writer
+Name[zh_CN]=LibreOffice 3.3 文档
+Name[zh_TW]=LibreOffice 3.3 文書處理器
+Name=LibreOffice 3.3 Writer
 Name[pt_BR]=BrOffice 3.3 Writer
 GenericName=Word Processor
 GenericName[en]=Word Processor

--- usr/share/applications/libreoffice-writer.desktop	2011-02-16 12:58:41.000000000 +0800
+++ usr/share/applications/libreoffice-writer.desktop	2011-03-11 10:39:48.004996002 +0800
@@ -7,6 +7,9 @@
 Exec=libreoffice -writer %U
 MimeType=application/vnd.oasis.opendocument.text;application/vnd.oasis.opendocument.text-template;application/vnd.oasis.opendocument.text-web;application/vnd.oasis.opendocument.text-master;application/vnd.sun.xml.writer;application/vnd.sun.xml.writer.template;application/vnd.sun.xml.writer.global;application/vnd.stardivision.writer;application/msword;application/vnd.ms-word;application/x-doc;application/rtf;text/rtf;application/vnd.wordperfect;application/wordperfect;application/vnd.openxmlformats-officedocument.wordprocessingml.document;application/vnd.ms-word.document.macroenabled.12;application/vnd.openxmlformats-officedocument.wordprocessingml.template;application/vnd.ms-word.template.macroenabled.12;
 Name=LibreOffice 3.3 Writer
+Name[zh_CN]=LibreOffice 3.3 文档
+Name[zh_TW]=LibreOffice 3.3 文書處理器
+Name=LibreOffice 3.3 Writer
 Name[pt_BR]=BrOffice 3.3 Writer
 GenericName=Word Processor
 GenericName[en]=Word Processor




