diff -ur nv/Makefile.kbuild nv.new//Makefile.kbuild
--- usr/src/nv/Makefile.kbuild	2011-04-15 00:26:35.000000000 +0800
+++ usr/src/nv/Makefile.kbuild	2011-11-04 14:00:30.404887042 +0800
@@ -288,8 +288,8 @@
 #
 
 module-install: suser-sanity-check module
-	@mkdir -p $(MODULE_ROOT)/video; \
-	install -m 0664 -o root -g root $(MODULE_OBJECT) $(MODULE_ROOT)/video; \
+	@mkdir -p $(DESTDIR)/$(MODULE_ROOT)/video; \
+	install -m 0664 -o root -g root $(MODULE_OBJECT) $(DESTDIR)/$(MODULE_ROOT)/video; \
 	PATH="$(PATH):/bin:/sbin" depmod -ae;
 
 #
@@ -298,8 +298,7 @@
 #
 
 package-install: module-install rmmod-sanity-check
-	PATH="$(PATH):/bin:/sbin" modprobe $(MODULE_NAME) && \
-	echo "$(MODULE_OBJECT) installed successfully.";
+	echo "$(MODULE_OBJECT) compiled successfully.";
 
 #
 # Build an object file suitable for further processing by the installer and
