diff -ur samba4-4.0.0-alpha11/librpc/ndr/libndr.h samba4-4.0.0-alpha11.new//librpc/ndr/libndr.h
--- samba4-4.0.0-alpha11/librpc/ndr/libndr.h	2010-01-11 13:35:28.000000000 +0800
+++ samba4-4.0.0-alpha11.new//librpc/ndr/libndr.h	2011-11-22 11:20:52.431677959 +0800
@@ -436,7 +436,7 @@
 			 struct ndr_token_list **list, 
 			 const void *key, 
 			 uint32_t value);
-enum ndr_err_code ndr_token_retrieve_cmp_fn(struct ndr_token_list **list, const void *key, uint32_t *v, int(*_cmp_fn)(const void*,const void*), bool _remove_tok);
+/*enum ndr_err_code ndr_token_retrieve_cmp_fn(struct ndr_token_list **list, const void *key, uint32_t *v, int(*_cmp_fn)(const void*,const void*), bool _remove_tok);*/
 enum ndr_err_code ndr_token_retrieve(struct ndr_token_list **list, const void *key, uint32_t *v);
 uint32_t ndr_token_peek(struct ndr_token_list **list, const void *key);
 enum ndr_err_code ndr_pull_array_size(struct ndr_pull *ndr, const void *p);
