--- a/Makefile	2010-10-14 23:06:47.000000000 +0800
+++ b/Makefile	2010-12-30 21:08:30.549000075 +0800
@@ -106,7 +106,7 @@
 
 ######################################################################
 
-qemu-img.o: qemu-img-cmds.h
+qemu-img.o: config-host.h qemu-img-cmds.h
 qemu-img.o qemu-tool.o qemu-nbd.o qemu-io.o: $(GENERATED_HEADERS)
 
 qemu-img$(EXESUF): qemu-img.o qemu-tool.o qemu-error.o $(block-obj-y) $(qobject-obj-y)
