diff -ur mediastreamer-2.8.2/include/mediastreamer2/msvideo.h mediastreamer-2.8.2.new//include/mediastreamer2/msvideo.h
--- mediastreamer-2.8.2/include/mediastreamer2/msvideo.h	2012-02-17 09:16:20.000000000 +0000
+++ mediastreamer-2.8.2.new//include/mediastreamer2/msvideo.h	2012-07-02 17:00:17.384572515 +0000
@@ -22,6 +22,13 @@
 
 #include <mediastreamer2/msfilter.h>
 
+#ifndef INT64_C
+#define INT64_C(c) (c ## LL)
+#define UINT64_C(c) (c ## ULL)
+#endif
+#include <libavutil/avutil.h>
+#include <libavcodec/version.h>
+
 /* some global constants for video MSFilter(s) */
 #define MS_VIDEO_SIZE_SQCIF_W 128
 #define MS_VIDEO_SIZE_SQCIF_H 96
@@ -206,7 +213,9 @@
 MS2_PUBLIC int ms_pix_fmt_to_ffmpeg(MSPixFmt fmt);
 MS2_PUBLIC MSPixFmt ffmpeg_pix_fmt_to_ms(int fmt);
 MS2_PUBLIC MSPixFmt ms_fourcc_to_pix_fmt(uint32_t fourcc);
+#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(54, 0, 100)
 MS2_PUBLIC void ms_ffmpeg_check_init(void);
+#endif
 MS2_PUBLIC int ms_yuv_buf_init_from_mblk(MSPicture *buf, mblk_t *m);
 MS2_PUBLIC int ms_yuv_buf_init_from_mblk_with_size(MSPicture *buf, mblk_t *m, int w, int h);
 MS2_PUBLIC int ms_picture_init_from_mblk_with_size(MSPicture *buf, mblk_t *m, MSPixFmt fmt, int w, int h);
diff -ur mediastreamer-2.8.2/src/h264dec.c mediastreamer-2.8.2.new//src/h264dec.c
--- mediastreamer-2.8.2/src/h264dec.c	2012-02-17 09:16:20.000000000 +0000
+++ mediastreamer-2.8.2.new//src/h264dec.c	2012-07-02 17:00:49.352397164 +0000
@@ -44,7 +44,9 @@
 static void ffmpeg_init(){
 	static bool_t done=FALSE;
 	if (!done){
+	#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(54, 0, 100)
 		avcodec_init();
+	#endif
 		avcodec_register_all();
 		done=TRUE;
 	}
diff -ur mediastreamer-2.8.2/src/mscommon.c mediastreamer-2.8.2.new//src/mscommon.c
--- mediastreamer-2.8.2/src/mscommon.c	2012-02-22 16:29:04.000000000 +0000
+++ mediastreamer-2.8.2.new//src/mscommon.c	2012-07-02 17:02:23.772484042 +0000
@@ -34,8 +34,17 @@
 #include "mediastreamer2/msfilter.h"
 #include <ortp/ortp_srtp.h>
 
+#ifndef INT64_C
+#define INT64_C(c) (c ## LL)
+#define UINT64_C(c) (c ## ULL)
+#endif
+#include <libavutil/avutil.h>
+#include <libavcodec/version.h>
+
 extern void __register_ffmpeg_encoders_if_possible(void);
+#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(54, 0, 100)
 extern void ms_ffmpeg_check_init();
+#endif
 extern bool_t libmsandroiddisplay_init(void);
 extern void libmsandroiddisplaybad_init(void);
 extern void libmsandroidopengldisplay_init(void);
@@ -645,7 +654,9 @@
 		}
 	}
 #if !defined(NO_FFMPEG)
+#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(54, 0, 100)
 	ms_ffmpeg_check_init();
+#endif
 	__register_ffmpeg_encoders_if_possible();
 #endif
 #endif
diff -ur mediastreamer-2.8.2/src/videodec.c mediastreamer-2.8.2.new//src/videodec.c
--- mediastreamer-2.8.2/src/videodec.c	2012-02-17 09:16:20.000000000 +0000
+++ mediastreamer-2.8.2.new//src/videodec.c	2012-07-02 17:03:32.128293453 +0000
@@ -28,9 +28,16 @@
 #include "mediastreamer2/msticker.h"
 #include "rfc2429.h"
 
+#ifndef INT64_C
+#define INT64_C(c) (c ## LL)
+#define UINT64_C(c) (c ## ULL)
+#endif
+#include <libavutil/avutil.h>
+#include <libavcodec/version.h>
 
+#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(54, 0, 100)
 extern void ms_ffmpeg_check_init();
-
+#endif
 
 typedef struct DecState{
 	AVCodecContext av_context;
@@ -50,7 +57,9 @@
 
 static void dec_init(MSFilter *f, enum CodecID cid){
 	DecState *s=(DecState *)ms_new0(DecState,1);
+	#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(54, 0, 100)
 	ms_ffmpeg_check_init();
+	#endif
 	
 	avcodec_get_context_defaults(&s->av_context);
 	s->av_codec=NULL;
diff -ur mediastreamer-2.8.2/src/videoenc.c mediastreamer-2.8.2.new//src/videoenc.c
--- mediastreamer-2.8.2/src/videoenc.c	2012-02-17 09:16:20.000000000 +0000
+++ mediastreamer-2.8.2.new//src/videoenc.c	2012-07-02 17:08:29.739073917 +0000
@@ -36,8 +36,9 @@
 #include "rfc2429.h"
 
 #define RATE_CONTROL_MARGIN 15000 /*bits/second*/
-
+#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(54, 0, 100)
 static bool_t avcodec_initialized=FALSE;
+#endif
 
 #ifdef ENABLE_LOG_FFMPEG
 
@@ -50,7 +51,7 @@
 }
 
 #endif
-
+#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(54, 0, 100)
 void ms_ffmpeg_check_init(){
 	if(!avcodec_initialized){
 		avcodec_init();
@@ -62,6 +63,7 @@
 #endif
 	}
 }
+#endif
 
 /* the goal of this small object is to tell when to send I frames at startup:
 at 2 and 4 seconds*/
@@ -201,7 +203,9 @@
 {
 	EncState *s=(EncState *)ms_new(EncState,1);
 	f->data=s;
+	#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(54, 0, 100)
 	ms_ffmpeg_check_init();
+	#endif
 	s->profile=0;/*always default to profile 0*/
 	s->comp_buf=NULL;
 	s->fps=15;
@@ -300,9 +304,13 @@
 	if (s->profile==0){
 		s->codec=CODEC_ID_H263;
 	}else{
+	#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(54, 0, 100)
 		c->flags|=CODEC_FLAG_H263P_UMV;
+	#endif
 		c->flags|=CODEC_FLAG_AC_PRED;
+	#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(54, 0, 100)
 		c->flags|=CODEC_FLAG_H263P_SLICE_STRUCT;
+	#endif
 		/*
 		c->flags|=CODEC_FLAG_OBMC;
 		c->flags|=CODEC_FLAG_AC_PRED;
@@ -372,7 +380,11 @@
 	// assume video size is CIF or QCIF
 	if (context->width == 352 && context->height == 288) header->b_wptr[1] = 0x60;
 	else header->b_wptr[1] = 0x40;
+	#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(54, 0, 100)
 	if (context->coded_frame->pict_type != FF_I_TYPE) header->b_wptr[1] |= 0x10;
+	#else
+	if (context->coded_frame->pict_type != AV_PICTURE_TYPE_I) header->b_wptr[1] |= 0x10;
+	#endif
 	header->b_wptr += 4;
 	header->b_cont = *packet;
 	*packet = header;
@@ -779,7 +791,11 @@
 		s->req_vfu=TRUE;
 	}
 	if (s->req_vfu){
+	#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(54, 0, 100)
 		pict.pict_type=FF_I_TYPE;
+	#else
+		pict.pict_type=AV_PICTURE_TYPE_I;
+	#endif
 		s->req_vfu=FALSE;
 	}
 	comp_buf->b_rptr=comp_buf->b_wptr=comp_buf->b_datap->db_base;
@@ -799,7 +815,11 @@
 		if (s->framenum==1){
 			video_starter_first_frame (&s->starter,f->ticker->time);
 		}
+		#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(54, 0, 100)
 		if (c->coded_frame->pict_type==FF_I_TYPE){
+		#else
+			if (c->coded_frame->pict_type==AV_PICTURE_TYPE_I){
+		#endif
 			ms_message("Emitting I-frame");
 		}
 		comp_buf->b_wptr+=error;
@@ -1067,7 +1087,9 @@
 #endif
 
 void __register_ffmpeg_encoders_if_possible(void){
+	#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(54, 0, 100)
 	ms_ffmpeg_check_init();
+	#endif
 	if (avcodec_find_encoder(CODEC_ID_MPEG4))
 		ms_filter_register(&ms_mpeg4_enc_desc);
 	if (avcodec_find_encoder(CODEC_ID_H263)){
