diff -ur uget-1.8.0/uget-gtk/UgSetting.c uget-1.8.0.new/uget-gtk/UgSetting.c
--- uget-1.8.0/uget-gtk/UgSetting.c	2011-04-15 12:51:28.000000000 +0800
+++ uget-1.8.0.new/uget-gtk/UgSetting.c	2011-07-04 14:56:45.489728149 +0800
@@ -455,7 +455,7 @@
 	setting->plugin.aria2.launch = TRUE;
 	setting->plugin.aria2.shutdown = TRUE;
 	setting->plugin.aria2.path = g_strdup ("aria2c");
-	setting->plugin.aria2.args = g_strdup ("--enable-xml-rpc");
+	setting->plugin.aria2.args = g_strdup ("--enable-rpc");
 	setting->plugin.aria2.uri  = g_strdup ("http://localhost:6800/rpc");
 
 	// "FolderList"
