diff -ur gnome-settings-daemon-2.30.2-ylmf1/plugins/xsettings/gsd-xsettings-manager.c gnome-settings-daemon-2.30.2-ylmf1.new/plugins/xsettings/gsd-xsettings-manager.c
--- gnome-settings-daemon-2.30.2-ylmf1/plugins/xsettings/gsd-xsettings-manager.c	2010-05-20 23:23:19.000000000 +0800
+++ gnome-settings-daemon-2.30.2-ylmf1.new/plugins/xsettings/gsd-xsettings-manager.c	2011-05-16 09:52:42.876550253 +0800
@@ -759,12 +759,23 @@
         int         n_screens;
         gboolean    res;
         gboolean    terminated;
+	gint        tries = 0;
 
         display = gdk_display_get_default ();
         n_screens = gdk_display_get_n_screens (display);
 
-        res = xsettings_manager_check_running (gdk_x11_display_get_xdisplay (display),
-                                               gdk_screen_get_number (gdk_screen_get_default ()));
+        //res = xsettings_manager_check_running (gdk_x11_display_get_xdisplay (display),
+        //                                       gdk_screen_get_number (gdk_screen_get_default ()));
+
+        do {
+                res = xsettings_manager_check_running (gdk_x11_display_get_xdisplay (display),
+                                                       gdk_screen_get_number (gdk_screen_get_default ()));
+                if (res) {
+                        g_usleep (100000);
+                        tries++;
+                }
+        } while (res && tries <= 20);
+
         if (res) {
                 g_warning ("You can only run one xsettings manager at a time; exiting");
                 return FALSE;
