diff -ur gnome-panel-2.32.1/gnome-panel/panel-toplevel.c gnome-panel-2.32.1.new//gnome-panel/panel-toplevel.c
--- gnome-panel-2.32.1/gnome-panel/panel-toplevel.c	2010-08-04 19:22:35.000000000 +0800
+++ gnome-panel-2.32.1.new//gnome-panel/panel-toplevel.c	2011-09-16 16:47:23.096692756 +0800
@@ -3540,6 +3540,10 @@
 {
 	g_return_if_fail (PANEL_IS_TOPLEVEL (toplevel));
 
+	GConfClient *client;
+		client = panel_gconf_get_client ();
+		gconf_client_set_int(client,"/apps/ymenu/flag",1,NULL);
+
 	if (toplevel->priv->state != PANEL_STATE_NORMAL)
 		return;
 
@@ -3621,6 +3625,10 @@
 {
 	g_return_if_fail (PANEL_IS_TOPLEVEL (toplevel));
 
+	GConfClient *client;
+		client = panel_gconf_get_client ();
+		gconf_client_set_int(client,"/apps/ymenu/flag",0,NULL);
+
 	if (toplevel->priv->state == PANEL_STATE_NORMAL)
 		return;
 
