--- gnome-panel-2.30.2/applets/notification_area/na-tray.c	2010-05-26 16:31:06.000000000 +0800
+++ gnome-panel-2.30.2/applets/notification_area/na-tray.c	2011-02-18 17:39:51.426656523 +0800
@@ -736,6 +736,25 @@
 na_tray_size_allocate (GtkWidget        *widget,
                        GtkAllocation    *allocation)
 {
+
+  GtkAllocation child_allocation;
+  child_allocation = *allocation;
+
+  switch(NA_TRAY(widget)->priv->orientation)
+  {
+    case GTK_ORIENTATION_VERTICAL:
+      child_allocation.width = allocation->width*3.0/5;
+      break;
+    case GTK_ORIENTATION_HORIZONTAL:
+      child_allocation.height = allocation->height*3.0/5;
+      break;
+    default:
+        break;
+  }
+  child_allocation.x = allocation->x + (allocation->width - child_allocation.width)/2.0;
+  child_allocation.y = allocation->y + (allocation->height - child_allocation.height)/2.0;
+  allocation = &child_allocation;
+
   gtk_widget_size_allocate (gtk_bin_get_child (GTK_BIN (widget)), allocation);
 }

