diff -urN gnome-panel-2.30.2/gnome-panel/panel-toplevel.c gnome-panel-hide-2.30.2/gnome-panel/panel-toplevel.c
--- gnome-panel-2.30.2/gnome-panel/panel-toplevel.c	2010-06-23 01:27:02.000000000 +0800
+++ gnome-panel-hide-2.30.2/gnome-panel/panel-toplevel.c	2011-07-06 21:39:26.467228364 +0800
@@ -3502,6 +3502,10 @@
 		     GtkDirectionType  direction)
 {
 	g_return_if_fail (PANEL_IS_TOPLEVEL (toplevel));
+	
+	GConfClient *client;
+        client = panel_gconf_get_client ();
+        gconf_client_set_int(client,"/apps/ymenu/flag",1,NULL);
 
 	if (toplevel->priv->state != PANEL_STATE_NORMAL)
 		return;
@@ -3583,6 +3587,10 @@
 panel_toplevel_unhide (PanelToplevel *toplevel)
 {
 	g_return_if_fail (PANEL_IS_TOPLEVEL (toplevel));
+	
+	GConfClient *client;
+        client = panel_gconf_get_client ();
+        gconf_client_set_int(client,"/apps/ymenu/flag",0,NULL);
 
 	if (toplevel->priv->state == PANEL_STATE_NORMAL)
 		return;
