diff -ruN gnome-mplayer-1.0.5.orig/DOCS/keyboard_shortcuts.txt gnome-mplayer-1.0.5/DOCS/keyboard_shortcuts.txt
--- gnome-mplayer-1.0.5.orig/DOCS/keyboard_shortcuts.txt	2008-04-29 05:13:23.000000000 +0800
+++ gnome-mplayer-1.0.5/DOCS/keyboard_shortcuts.txt	2012-03-13 17:38:33.236575086 +0800
@@ -4,9 +4,10 @@
 Left Arrow - Fast Reverse
 Page Up - Seek 10min forward
 Page Down - Seek 10min backward
-Up Arrow - Seek 1 min forward
-Down Arrow - Seek 1 min backward
+Up Arrow - Volume down 10
+Down Arrow - Volume up 10
 Space - Pause
+
 p - Pause
 m - Mute
 
@@ -24,8 +25,8 @@
 { - Half speed playback
 } - Double speed playback
 
-9 - Volume down 10
-0 - Volume up 10
+9 - Seek 1 min forward
+0 - Seek 1 min backward
 
 # - Switch Audio Track
 j - Subtitle select
diff -ruN gnome-mplayer-1.0.5.orig/src/gui.c gnome-mplayer-1.0.5/src/gui.c
--- gnome-mplayer-1.0.5.orig/src/gui.c	2011-12-17 03:55:51.000000000 +0800
+++ gnome-mplayer-1.0.5/src/gui.c	2012-03-13 17:32:34.299909878 +0800
@@ -1744,33 +1744,24 @@
             if (title_is_menu) {
                 return gmtk_media_player_send_key_press_event(GMTK_MEDIA_PLAYER(media), event, data);
             } else {
-                if (gmtk_media_player_get_state(GMTK_MEDIA_PLAYER(media)) != MEDIA_STATE_STOP
-                    && !gtk_check_menu_item_get_active(GTK_CHECK_MENU_ITEM(menuitem_view_playlist)))
-                    gmtk_media_player_seek(GMTK_MEDIA_PLAYER(media), 60.0, SEEK_RELATIVE);
-                if (gmtk_media_player_get_state(GMTK_MEDIA_PLAYER(media)) == MEDIA_STATE_PAUSE) {
-                    gmtk_media_player_seek(GMTK_MEDIA_PLAYER(media), 0.0, SEEK_RELATIVE);
-                    idledata->position += 60;
-                    if (idledata->position > idledata->length)
-                        idledata->position = 0;
-                    gmtk_media_tracker_set_percentage(tracker, idledata->position / idledata->length);
-                }
+#ifdef GTK2_12_ENABLED
+            gtk_scale_button_set_value(GTK_SCALE_BUTTON(vol_slider),
+                                       gtk_scale_button_get_value(GTK_SCALE_BUTTON(vol_slider)) + 0.10);
+#else
+            gtk_range_set_value(GTK_RANGE(vol_slider), gtk_range_get_value(GTK_RANGE(vol_slider)) + 0.10);
+#endif
             }
-
             return FALSE;
         case GDK_Down:
             if (title_is_menu) {
                 return gmtk_media_player_send_key_press_event(GMTK_MEDIA_PLAYER(media), event, data);
             } else {
-                if (gmtk_media_player_get_state(GMTK_MEDIA_PLAYER(media)) != MEDIA_STATE_STOP
-                    && !gtk_check_menu_item_get_active(GTK_CHECK_MENU_ITEM(menuitem_view_playlist)))
-                    gmtk_media_player_seek(GMTK_MEDIA_PLAYER(media), -60.0, SEEK_RELATIVE);
-                if (gmtk_media_player_get_state(GMTK_MEDIA_PLAYER(media)) == MEDIA_STATE_PAUSE) {
-                    gmtk_media_player_seek(GMTK_MEDIA_PLAYER(media), 0.0, SEEK_RELATIVE);
-                    idledata->position -= 60;
-                    if (idledata->position < 0)
-                        idledata->position = 0;
-                    gmtk_media_tracker_set_percentage(tracker, idledata->position / idledata->length);
-                }
+#ifdef GTK2_12_ENABLED
+                gtk_scale_button_set_value(GTK_SCALE_BUTTON(vol_slider),
+                        gtk_scale_button_get_value(GTK_SCALE_BUTTON(vol_slider)) - 0.10);
+#else
+                gtk_range_set_value(GTK_RANGE(vol_slider), gtk_range_get_value(GTK_RANGE(vol_slider)) - 0.10);
+#endif
             }
             return FALSE;
         case GDK_Return:
@@ -1848,20 +1839,28 @@
             gmtk_media_player_set_attribute_double(GMTK_MEDIA_PLAYER(media), ATTRIBUTE_SPEED_SET, 1.0);
             return FALSE;
         case GDK_9:
-#ifdef GTK2_12_ENABLED
-            gtk_scale_button_set_value(GTK_SCALE_BUTTON(vol_slider),
-                                       gtk_scale_button_get_value(GTK_SCALE_BUTTON(vol_slider)) - 0.10);
-#else
-            gtk_range_set_value(GTK_RANGE(vol_slider), gtk_range_get_value(GTK_RANGE(vol_slider)) - 0.10);
-#endif
+            if (gmtk_media_player_get_state(GMTK_MEDIA_PLAYER(media)) != MEDIA_STATE_STOP
+                    && !gtk_check_menu_item_get_active(GTK_CHECK_MENU_ITEM(menuitem_view_playlist)))
+                gmtk_media_player_seek(GMTK_MEDIA_PLAYER(media), -60.0, SEEK_RELATIVE);
+            if (gmtk_media_player_get_state(GMTK_MEDIA_PLAYER(media)) == MEDIA_STATE_PAUSE) {
+                gmtk_media_player_seek(GMTK_MEDIA_PLAYER(media), 0.0, SEEK_RELATIVE);
+                idledata->position -= 60;
+                if (idledata->position < 0)
+                    idledata->position = 0;
+                gmtk_media_tracker_set_percentage(tracker, idledata->position / idledata->length);
+            }
             return FALSE;
         case GDK_0:
-#ifdef GTK2_12_ENABLED
-            gtk_scale_button_set_value(GTK_SCALE_BUTTON(vol_slider),
-                                       gtk_scale_button_get_value(GTK_SCALE_BUTTON(vol_slider)) + 0.10);
-#else
-            gtk_range_set_value(GTK_RANGE(vol_slider), gtk_range_get_value(GTK_RANGE(vol_slider)) + 0.10);
-#endif
+            if (gmtk_media_player_get_state(GMTK_MEDIA_PLAYER(media)) != MEDIA_STATE_STOP
+                    && !gtk_check_menu_item_get_active(GTK_CHECK_MENU_ITEM(menuitem_view_playlist)))
+                gmtk_media_player_seek(GMTK_MEDIA_PLAYER(media), 60.0, SEEK_RELATIVE);
+            if (gmtk_media_player_get_state(GMTK_MEDIA_PLAYER(media)) == MEDIA_STATE_PAUSE) {
+                gmtk_media_player_seek(GMTK_MEDIA_PLAYER(media), 0.0, SEEK_RELATIVE);
+                idledata->position += 60;
+                if (idledata->position > idledata->length)
+                    idledata->position = 0;
+                gmtk_media_tracker_set_percentage(tracker, idledata->position / idledata->length);
+            }
             return FALSE;
         case GDK_numbersign:
             gmtk_media_player_send_command(GMTK_MEDIA_PLAYER(media), COMMAND_SWITCH_AUDIO);
