diff -ur gnome-mplayer-1.0.2-ylmf1/src/thread.c gnome-mplayer-1.0.2-ylmf1.new/src/thread.c
--- gnome-mplayer-1.0.2-ylmf1/src/thread.c	2011-03-03 04:06:15.000000000 +0800
+++ gnome-mplayer-1.0.2-ylmf1.new/src/thread.c	2011-05-16 14:54:20.599541968 +0800
@@ -1299,8 +1299,8 @@
 
     if (!use_hw_audio) {
 #ifndef OS_WIN32
-        argv[arg++] = g_strdup_printf("-af-add");
-        argv[arg++] = g_strdup_printf("export=%s:512", idledata->af_export);
+//        argv[arg++] = g_strdup_printf("-af-add");
+//        argv[arg++] = g_strdup_printf("export=%s:512", idledata->af_export);
 #endif
     }
 
