--- chrome/browser/search_engines/template_url_prepopulate_data.cc	2012-07-10 07:49:00.000000000 +0000
+++ chrome/browser/search_engines/template_url_prepopulate_data.cc.new	2012-07-26 15:07:16.708794964 +0000
@@ -255,7 +255,7 @@
   L"\x767e\x5ea6",
   L"baidu.com",
   "http://www.baidu.com/favicon.ico",
-  "http://www.baidu.com/s?wd={searchTerms}",
+  "http://www.baidu.com/s?wd={searchTerms}&tn=ylmf_dg&ch=5",
   "GB2312",
   NULL,
   NULL,
