--- ./webkit/glue/user_agent.cc	2012-07-10 07:48:15.000000000 +0000
+++ ./webkit/glue/user_agent.cc.new	2012-10-17 14:15:48.096655503 +0000
@@ -131,7 +131,7 @@
   base::StringAppendF(
       &user_agent,
       "Mozilla/5.0 (%s%s) AppleWebKit/%d.%d"
-      " (KHTML, like Gecko) %s Safari/%d.%d",
+      " (KHTML, like Gecko) StartOS/5.0 %s Safari/%d.%d",
       kUserAgentPlatform,
       webkit_glue::BuildOSCpuInfo().c_str(),
       WEBKIT_VERSION_MAJOR,
