diff -ur chromium-15.0.876.0/content/browser/download/base_file.cc chromium-15.0.876.0.new/content/browser/download/base_file.cc
--- chromium-15.0.876.0/content/browser/download/base_file.cc	2011-09-10 09:08:12.000000000 +0000
+++ chromium-15.0.876.0.new/content/browser/download/base_file.cc	2011-12-06 10:22:23.700241260 +0000
@@ -305,16 +305,19 @@
     int stat_error = stat(new_path.value().c_str(), &st);
     bool stat_succeeded = (stat_error == 0);
     if (!stat_succeeded)
-      return LOG_ERROR("stat", net::MapSystemError(errno));
+      LOG_ERROR("stat", net::MapSystemError(errno));
 
     // TODO(estade): Move() falls back to copying and deleting when a simple
     // rename fails. Copying sucks for large downloads. crbug.com/8737
     if (!file_util::Move(full_path_, new_path))
       return LOG_ERROR("Move", net::MapSystemError(errno));
 
-    int chmod_error = chmod(new_path.value().c_str(), st.st_mode);
-    if (chmod_error < 0)
-      return LOG_ERROR("chmod", net::MapSystemError(errno));
+    if (stat_succeeded) {
+       // On Windows file systems (FAT, NTFS), chmod fails.  This is OK.
+       int chmod_error = chmod(new_path.value().c_str(), st.st_mode);
+       if (chmod_error < 0)
+         LOG_ERROR("chmod", net::MapSystemError(errno));
+      }
   }
 #endif
 
