diff -ur chromium-15.0.876.0/chrome/browser/search_engines/template_url_prepopulate_data.h chromium-15.0.876.0.new//chrome/browser/search_engines/template_url_prepopulate_data.h
--- chromium-15.0.876.0/chrome/browser/search_engines/template_url_prepopulate_data.h	2011-09-10 09:08:59.000000000 +0000
+++ chromium-15.0.876.0.new//chrome/browser/search_engines/template_url_prepopulate_data.h	2011-12-06 02:07:04.011714617 +0000
@@ -8,6 +8,7 @@
 
 #include <stddef.h>
 #include <vector>
+#include <cstddef>
 
 class GURL;
 class PrefService;
diff -ur chromium-15.0.876.0/ppapi/cpp/paint_aggregator.h chromium-15.0.876.0.new//ppapi/cpp/paint_aggregator.h
--- chromium-15.0.876.0/ppapi/cpp/paint_aggregator.h	2011-09-10 09:08:10.000000000 +0000
+++ chromium-15.0.876.0.new//ppapi/cpp/paint_aggregator.h	2011-12-06 02:07:00.661662868 +0000
@@ -10,6 +10,7 @@
 
 #include "ppapi/cpp/point.h"
 #include "ppapi/cpp/rect.h"
+#include <cstddef>
 
 /// @file
 /// This file defines the API to aggregate multiple invalidation and scroll
diff -ur chromium-15.0.876.0/third_party/ots/src/os2.cc chromium-15.0.876.0.new//third_party/ots/src/os2.cc
--- chromium-15.0.876.0/third_party/ots/src/os2.cc	2011-09-10 09:09:26.000000000 +0000
+++ chromium-15.0.876.0.new//third_party/ots/src/os2.cc	2011-12-06 02:07:21.205316368 +0000
@@ -5,6 +5,7 @@
 #include "os2.h"
 
 #include "head.h"
+#include <cstddef>
 
 // OS/2 - OS/2 and Windows Metrics
 // http://www.microsoft.com/opentype/otspec/os2.htm
diff -ur chromium-15.0.876.0/third_party/tcmalloc/chromium/src/base/stl_allocator.h chromium-15.0.876.0.new//third_party/tcmalloc/chromium/src/base/stl_allocator.h
--- chromium-15.0.876.0/third_party/tcmalloc/chromium/src/base/stl_allocator.h	2011-09-10 09:08:18.000000000 +0000
+++ chromium-15.0.876.0.new//third_party/tcmalloc/chromium/src/base/stl_allocator.h	2011-12-06 02:07:45.602363430 +0000
@@ -39,6 +39,7 @@
 
 #include <stddef.h>   // for std::ptrdiff_t
 #include <limits>
+#include <cstddef>
 
 #include "base/logging.h"
 
diff -ur chromium-15.0.876.0/third_party/tcmalloc/chromium/src/base/vdso_support.h chromium-15.0.876.0.new//third_party/tcmalloc/chromium/src/base/vdso_support.h
--- chromium-15.0.876.0/third_party/tcmalloc/chromium/src/base/vdso_support.h	2011-09-10 09:08:18.000000000 +0000
+++ chromium-15.0.876.0.new//third_party/tcmalloc/chromium/src/base/vdso_support.h	2011-12-06 02:08:10.672646700 +0000
@@ -38,6 +38,7 @@
 
 #define HAVE_VDSO_SUPPORT 1
 
+#include <cstddef>
 #include <stdlib.h>     // for NULL
 #include <link.h>  // for ElfW
 
diff -ur chromium-15.0.876.0/third_party/tcmalloc/chromium/src/common.cc chromium-15.0.876.0.new//third_party/tcmalloc/chromium/src/common.cc
--- chromium-15.0.876.0/third_party/tcmalloc/chromium/src/common.cc	2011-09-10 09:08:18.000000000 +0000
+++ chromium-15.0.876.0.new//third_party/tcmalloc/chromium/src/common.cc	2011-12-06 02:08:29.716205021 +0000
@@ -33,6 +33,7 @@
 #include "config.h"
 #include "common.h"
 #include "system-alloc.h"
+#include <cstddef>
 
 namespace tcmalloc {
 
diff -ur chromium-15.0.876.0/third_party/tcmalloc/chromium/src/symbolize.h chromium-15.0.876.0.new//third_party/tcmalloc/chromium/src/symbolize.h
--- chromium-15.0.876.0/third_party/tcmalloc/chromium/src/symbolize.h	2011-09-10 09:08:18.000000000 +0000
+++ chromium-15.0.876.0.new//third_party/tcmalloc/chromium/src/symbolize.h	2011-12-06 02:08:53.753167239 +0000
@@ -40,6 +40,7 @@
 #endif
 #include <stddef.h>  // for NULL
 #include <map>
+#include <cstddef>
 
 using std::map;
 
diff -ur chromium-15.0.876.0/ui/gfx/codec/jpeg_codec.h chromium-15.0.876.0.new//ui/gfx/codec/jpeg_codec.h
--- chromium-15.0.876.0/ui/gfx/codec/jpeg_codec.h	2011-09-10 09:09:03.000000000 +0000
+++ chromium-15.0.876.0.new//ui/gfx/codec/jpeg_codec.h	2011-12-06 02:09:07.070003003 +0000
@@ -8,6 +8,7 @@
 
 #include <stddef.h>
 #include <vector>
+#include <cstddef>
 
 #include "ui/base/ui_export.h"
 
