--- chrome/browser/search_engines/template_url_prepopulate_data.cc	2012-04-04 09:36:53.000000000 +0800
+++ chrome/browser/search_engines/template_url_prepopulate_data.cc.new	2012-04-12 14:58:55.849834103 +0800
@@ -272,7 +272,7 @@
   L"\x767e\x5ea6",
   L"baidu.com",
   "http://www.baidu.com/favicon.ico",
-  L"http://www.baidu.com/s?wd={searchTerms}",
+  L"http://www.baidu.com/s?wd={searchTerms}&tn=ylmf_dg&ch=5",
   "GB2312",
   NULL,
   NULL,
