diff -ur chromium-13.0.762.0/chrome/tools/build/linux/chrome-wrapper chromium-13.0.762.0.new//chrome/tools/build/linux/chrome-wrapper
--- chromium-13.0.762.0/chrome/tools/build/linux/chrome-wrapper	2011-05-11 10:00:55.000000000 +0800
+++ chromium-13.0.762.0.new//chrome/tools/build/linux/chrome-wrapper	2011-05-12 17:48:58.517417513 +0800
@@ -7,11 +7,13 @@
 # Running Chromium via this script makes it possible to set Chromium as the
 # default browser directly out of a compile, without needing to package it.
 
-DESKTOP="chromium-devel"
+DESKTOP="chromium"
 TITLE="Chromium"
+APP="chromium"
+BIN="/usr/lib/chromium/chromium"
 
 usage() {
-  echo "$0 [--gdb] [--help] [--man-page] [--] [chrome-options]"
+  echo "$0 [--gdb] [--help] [--man-page] [--] [chromium-options]"
   echo
   echo "        --gdb                   Start within gdb"
   echo "        --help                  This help screen"
@@ -58,7 +60,7 @@
 Version=1.0
 Encoding=UTF-8
 Name=$TITLE
-Exec=$CHROME_WRAPPER %U
+Exec=$APP %U
 Terminal=false
 Icon=$HERE/product_logo_48.png
 Type=Application
@@ -92,8 +94,8 @@
 # Always use our ffmpeg and other shared libs.
 export LD_LIBRARY_PATH="$HERE:$HERE/lib:$HERE/lib.target${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}"
 
-MISSING_LIBS=$(ldd "$HERE/chrome" 2> /dev/null |grep "not found$" | cut -d" " -f 1|sed 's/\t//')
-CHROME_ARCH=$(check_executable "$HERE/chrome")
+MISSING_LIBS=$(ldd "$BIN" 2> /dev/null |grep "not found$" | cut -d" " -f 1|sed 's/\t//')
+CHROME_ARCH=$(check_executable "$BIN")
 uname -m | grep -qs x86_64
 if [ $? = 1 ]; then
     LIBDIRS="/lib /lib32 /usr/lib /usr/lib32"
@@ -135,7 +137,7 @@
 
 # Custom version string for this release. This can be used to add a downstream
 # vendor string or release channel information.
-export CHROME_VERSION_EXTRA="custom"
+export CHROME_VERSION_EXTRA="Ylmf OS 4.0"
 
 exists_desktop_file || generate_desktop_file
 
@@ -152,7 +154,7 @@
         usage
         exit 0 ;;
     "--man-page")
-        exec man "$HERE/../../chrome/app/resources/manpage.1.in" ;;
+        exec man "$APP" ;;
     *)
         ARGS=( "${ARGS[@]}" "$1" ) ;;
     esac
@@ -160,4 +162,12 @@
 done
 set -- "${ARGS[@]}" "$@"
 
-exec $CMD_PREFIX "$HERE/chrome" "$@"
+#cp default Preferences to HOME
+#     
+if [ ! -d $HOME/.config/chromium ]; then
+	mkdir -p $HOME/.config
+	tar xf /usr/lib/chromium/chromium.tar.bz2 -C $HOME/.config/
+	chown -R $USER:$USER $HOME/.config/chromium 
+fi
+flash_setting_dir="$HOME/.macromedia/Flash_Player/macromedia.com/support/flashplayer/sys"
+if [ ! -d $flash_setting_dir ]; then
+	mkdir -p $flash_setting_dir
+	cp -a /usr/lib/chromium/settings.sol $flash_setting_dir
+	chown -R $USER:$USER $flash_setting_dir/settings.sol
+fi
+exec $CMD_PREFIX "$BIN" "$@"
