diff -ur chromium-13.0.762.0-ylmf1/chrome/browser/search_engines/template_url_prepopulate_data.cc chromium-13.0.762.0-ylmf1.new/chrome/browser/search_engines/template_url_prepopulate_data.cc
--- chromium-13.0.762.0-ylmf1/chrome/browser/search_engines/template_url_prepopulate_data.cc	2011-05-10 14:19:15.000000000 +0800
+++ chromium-13.0.762.0-ylmf1.new/chrome/browser/search_engines/template_url_prepopulate_data.cc	2011-05-18 13:49:15.044876901 +0800
@@ -268,7 +268,7 @@
   L"\x767e\x5ea6",
   L"baidu.com",
   "http://www.baidu.com/favicon.ico",
-  L"http://www.baidu.com/s?wd={searchTerms}",
+  L"http://www.baidu.com/s?wd={searchTerms}&tn=ylmf_dg&ch=5",
   "GB2312",
   NULL,
   NULL,
