diff -ur ndiswrapper-1.57/driver/Makefile ndiswrapper-1.57.new//driver/Makefile
--- ndiswrapper-1.57/driver/Makefile	2011-12-31 20:30:15.000000000 +0000
+++ ndiswrapper-1.57.new//driver/Makefile	2012-08-13 17:27:45.605971941 +0000
@@ -36,7 +36,7 @@
 $(error Cannot find kernel version in $(KBUILD), is it configured?)
 endif
 
-KVERS := $(shell sed -ne 's/"//g;s/^\#define UTS_RELEASE //p' $(VERSION_H))
+KVERS := $(shell uname -r)
 
 ifeq (,$(KVERS))
 $(error Cannot find UTS_RELEASE in $(VERSION_H), has 'make modules_prepare' been called?)
@@ -196,15 +196,9 @@
 		{ echo "$(MODULE)" is not for Linux $(KVERS); exit 1; }
 	mkdir -p -m 755 $(DESTDIR)$(INST_DIR)
 	install -m 0644 $(MODULE) $(DESTDIR)$(INST_DIR)
-ifndef DESTDIR
-	-/sbin/depmod -a $(KVERS)
-endif
 
 uninstall:
 	rm -f $(DESTDIR)$(INST_DIR)/$(MODULE)
-ifndef DESTDIR
-	-/sbin/depmod -a $(KVERS)
-endif
 
 dist:
 	@for file in $(DISTFILES); do \
diff -ur ndiswrapper-1.57/Makefile ndiswrapper-1.57.new//Makefile
--- ndiswrapper-1.57/Makefile	2011-12-31 20:30:15.000000000 +0000
+++ ndiswrapper-1.57.new//Makefile	2012-08-13 17:30:57.782628206 +0000
@@ -24,6 +24,14 @@
 $(SUBDIRS):
 	$(MAKE) -C $@
 
+.PHONY: utils
+utils:
+	$(MAKE) -C utils
+
+PHONY: driver
+driver:
+	$(MAKE) -C driver
+
 .PHONY: install
 install:
 	$(MAKE) -C driver install
@@ -32,6 +40,17 @@
 	install -m 644 ndiswrapper.8 $(DESTDIR)$(mandir)/man8
 	install -m 644 loadndisdriver.8 $(DESTDIR)$(mandir)/man8
 
+.PHONY: install-utils
+install-utils:
+	$(MAKE) -C utils install
+	mkdir -p -m 0755 $(mandir)/man8
+	install -m 644 ndiswrapper.8 $(mandir)/man8
+	install -m 644 loadndisdriver.8 $(mandir)/man8
+
+.PHONY: install-driver
+install-driver:
+	$(MAKE) -C driver install
+
 .PHONY: clean distclean
 clean:
 	$(MAKE) -C driver clean
