diff -ur ndiswrapper-1.56/driver/wrapndis.c ndiswrapper-1.56.new/driver/wrapndis.c
--- ndiswrapper-1.56/driver/wrapndis.c	2011-05-12 11:18:53.344394300 +0800
+++ ndiswrapper-1.56.new/driver/wrapndis.c	2011-05-12 11:20:34.844392955 +0800
@@ -13,16 +13,16 @@
  *
  */
 
-#include "ndis.h"
-#include "iw_ndis.h"
-#include "pnp.h"
-#include "loader.h"
-#include "wrapndis.h"
 #include <linux/inetdevice.h>
 #include <linux/ip.h>
 #include <linux/tcp.h>
 #include <linux/udp.h>
 #include <linux/in.h>
+#include "ndis.h"
+#include "iw_ndis.h"
+#include "pnp.h"
+#include "loader.h"
+#include "wrapndis.h"
 #include "wrapper.h"
 
 /* Functions callable from the NDIS driver */
