--- Makefile	2011-10-25 11:35:56.648014169 +0800
+++ Makefile.new	2011-10-25 11:37:02.856020906 +0800
@@ -7,7 +7,7 @@
 
 DISTFILES=AUTHORS ChangeLog INSTALL Makefile README ndiswrapper.spec \
 				   ndiswrapper.8 loadndisdriver.8
-DIST_SUBDIRS=utils driver
+DIST_SUBDIRS=driver
 
 DESTDIR =
 mandir = $(DESTDIR)$(shell [ -d /usr/man/man8 ] && echo /usr/man || echo /usr/share/man )
@@ -18,11 +18,6 @@
 
 all:
 	+make -C driver
-	+make -C utils
-
-.PHONY: utils
-utils:
-	+make -C utils
 
 .PHONY: driver
 driver:
@@ -31,17 +26,6 @@
 .PHONY: install
 install:
 	+make -C driver install
-	+make -C utils install
-	mkdir -p -m 0755 $(mandir)/man8
-	install -m 644 ndiswrapper.8 $(mandir)/man8
-	install -m 644 loadndisdriver.8 $(mandir)/man8
-
-.PHONY: install-utils
-install-utils:
-	+make -C utils install
-	mkdir -p -m 0755 $(mandir)/man8
-	install -m 644 ndiswrapper.8 $(mandir)/man8
-	install -m 644 loadndisdriver.8 $(mandir)/man8
 
 .PHONY: install-driver
 install-driver:
@@ -50,13 +34,11 @@
 .PHONY: clean distclean
 clean:
 	+make -C driver clean
-	+make -C utils clean
 	rm -f *~
 	rm -fr ${distdir} ${distdir}.tar.gz patch-stamp
 
 distclean: clean
 	+make -C driver distclean
-	+make -C utils distclean
 	rm -f .\#*
 
 uninstall:
@@ -96,7 +78,6 @@
 	  fi; \
 	done
 	+make -C driver distdir=../${distdir}/driver dist
-	+make -C utils distdir=../${distdir}/utils dist
 
 	# Update version in dist rpm spec file - don't crash if it fails
 	-sed -i "s/\%define\s\+ndiswrapper_version\s\+[^\}]\+\}/%define ndiswrapper_version $(DRIVER_VERSION)\}/" $(distdir)/ndiswrapper.spec
