diff -ruN gvfs-1.6.7.orig/daemon/trashlib/trashwatcher.c gvfs-1.6.7/daemon/trashlib/trashwatcher.c
--- gvfs-1.6.7.orig/daemon/trashlib/trashwatcher.c	2011-03-21 23:42:18.000000000 +0800
+++ gvfs-1.6.7/daemon/trashlib/trashwatcher.c	2011-10-31 15:04:39.036653036 +0800
@@ -250,14 +250,20 @@
   GFile *homedir_trashdir;
   TrashWatcher *watcher;
   GFile *user_datadir;
+  GVolumeMonitor *V_mon;
 
   watcher = g_slice_new (TrashWatcher);
   watcher->root = root;
   watcher->mounts = NULL;
   watcher->watching = FALSE;
   watcher->mount_monitor = g_unix_mount_monitor_new ();
-  g_signal_connect_swapped (watcher->mount_monitor, "mounts_changed",
-                            G_CALLBACK (trash_watcher_remount), watcher);
+  /* g_signal_connect_swapped (watcher->mount_monitor, "mounts_changed",
+                            G_CALLBACK (trash_watcher_remount), watcher); */
+  
+  /* Editing by StartOS developer for fixing Trash real time to show Trash files */
+  V_mon = g_volume_monitor_get();
+  g_signal_connect_swapped (V_mon, "volume-changed", 
+                   	    G_CALLBACK (trash_watcher_remount), watcher);
 
   user_datadir = g_file_new_for_path (g_get_user_data_dir ());
   homedir_trashdir = g_file_get_child (user_datadir, "Trash/files");
