diff -ur libgnomeprint-2.18.8/Makefile.am libgnomeprint-2.18.8.new/Makefile.am
--- libgnomeprint-2.18.8/Makefile.am	2010-02-09 20:32:51.000000000 +0800
+++ libgnomeprint-2.18.8.new/Makefile.am	2011-09-13 17:48:02.599561517 +0800
@@ -1,4 +1,4 @@
-SUBDIRS = libgnomeprint filters tests data po doc
+SUBDIRS = libgnomeprint filters tests data po 
 
 EXTRA_DIST = \
     MAINTAINERS \
diff -ur libgnomeprint-2.18.8/Makefile.in libgnomeprint-2.18.8.new/Makefile.in
--- libgnomeprint-2.18.8/Makefile.in	2010-09-28 17:29:23.000000000 +0800
+++ libgnomeprint-2.18.8.new/Makefile.in	2011-09-13 17:48:13.389715750 +0800
@@ -277,7 +277,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = libgnomeprint filters tests data po doc
+SUBDIRS = libgnomeprint filters tests data po 
 EXTRA_DIST = \
     MAINTAINERS \
     ChangeLog \
