--- ca-certificates-20110421/Makefile	2011-04-22 01:44:18.000000000 +0800
+++ ca-certificates-20110421.new/Makefile	2011-06-08 15:52:24.189713063 +0800
@@ -17,10 +17,11 @@
 
 install:
 	for dir in $(SUBDIRS); do \
-	  mkdir $(DESTDIR)/$(CERTSDIR)/$$dir; \
+	  mkdir -p $(DESTDIR)/$(CERTSDIR)/$$dir; \
 	  $(MAKE) -C $$dir install CERTSDIR=$(DESTDIR)/$(CERTSDIR)/$$dir; \
 	done
 	for dir in sbin; do \
+	  mkdir -p $(DESTDIR)/usr/$$dir; \
 	  $(MAKE) -C $$dir install CERTSDIR=$(DESTDIR)/$(CERTSDIR)/$$dir; \
 	done
 
