diff -u -r gftp-2.0.19/docs/gftp.desktop gftp-2.0.19.new/docs/gftp.desktop
--- gftp-2.0.19/docs/gftp.desktop	2007-01-10 09:24:50.000000000 +0800
+++ gftp-2.0.19.new/docs/gftp.desktop	2011-04-18 20:23:53.704283177 +0800
@@ -1,8 +1,10 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=gFTP
+Name[zh_CN]=gFTP ftp客户端工具
 Comment=Download and upload files using multiple file transfer protocols
 Comment[fr]=Télécharge des fichiers en utilisant le protocole FTP
+Comment[zh_CN]=一个用 Gtk 开发的多线程 FTP 客户端工具.
 GenericName=gFTP
 Exec=gftp %u
 Terminal=false
