--- mono-2.8/mcs/tools/gacutil/driver.cs	2010-09-10 21:55:45.000000000 +0100
+++ mono-2.8/mcs/tools/gacutil/driver-new.cs	2010-09-15 13:32:30.335396575 +0100
@@ -808,8 +808,8 @@
 		private static string EnsureLib (string dir)
 		{
 			DirectoryInfo d = new DirectoryInfo (dir);
-			if (d.Name == "lib")
+			if (d.Name == "lib" || d.Name == "lib64")
 				return dir;
 			return Path.Combine (dir, "lib");
 		}
 
--- mono-2.6.4/data/mint.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/mint.pc-new.in	2010-04-27 00:34:27.000000000 +0100
@@ -1,6 +1,6 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${pcfiledir}/../..
-libdir=${prefix}/lib
+libdir=@LIBDIR@
 includedir=${prefix}/include/mono-@API_VER@
 
 
--- mono-2.6.4/data/mono.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/mono.pc-new.in	2010-04-27 00:34:57.000000000 +0100
@@ -1,6 +1,6 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${pcfiledir}/../..
-libdir=${prefix}/@reloc_libdir@
+libdir=@LIBDIR@
 includedir=${prefix}/include/mono-@API_VER@
 sysconfdir=@sysconfdir@
 
--- mono-2.6.4/data/mono.web.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/mono.web.pc-new.in	2010-04-27 00:40:27.000000000 +0100
@@ -1,4 +1,4 @@
 Name: Mono.Web
 Description: Mono ASP.NET extension classes
 Version: 2.0.0.0
-Libs: -r:@prefix@/lib/mono/gac/Mono.Web/2.0.0.0__0738eb9f132ed756/Mono.Web.dll
+Libs: -r:@LIBDIR@/mono/gac/Mono.Web/2.0.0.0__0738eb9f132ed756/Mono.Web.dll

--- mono-2.6.4/data/mono-lineeditor.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/mono-lineeditor.pc-new.in	2010-04-27 00:37:17.000000000 +0100
@@ -1,5 +1,5 @@
 prefix=${pcfiledir}/../..
-assemblies_dir=${prefix}/lib/mono-source-libs
+assemblies_dir=@LIBDIR@/mono-source-libs
 Sources=${assemblies_dir}/getline.cs
 
 Name: Mono.Terminal.LineEditor
--- mono-2.6.4/data/mono-nunit.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/mono-nunit.pc-new.in	2010-04-27 00:37:48.000000000 +0100
@@ -1,6 +1,6 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+libdir=@LIBDIR@
 
 
 Name: Mono NUnit
--- mono-2.6.4/data/mono-options.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/mono-options.pc-new.in	2010-04-27 00:38:21.000000000 +0100
@@ -1,5 +1,5 @@
 prefix=${pcfiledir}/../..
-assemblies_dir=${prefix}/lib/mono-source-libs
+assemblies_dir=@LIBDIR@/mono-source-libs
 Sources=${assemblies_dir}/Options.cs
 
 Name: Mono.Options
--- mono-2.6.4/data/system.web.extensions.design_1.0.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/system.web.extensions.design_1.0.pc-new.in	2010-04-27 00:38:43.000000000 +0100
@@ -1,4 +1,4 @@
 Name: System.Web.Extensions.Design
 Description: System.Web.Extensions.Design ASP.NET 2.0 add-on
 Version: 1.0.61025.0
-Libs: -r:@prefix@/lib/mono/gac/System.Web.Extensions.Design/1.0.61025.0__31bf3856ad364e35/System.Web.Extensions.Design.dll
+Libs: -r:@LIBDIR@/mono/gac/System.Web.Extensions.Design/1.0.61025.0__31bf3856ad364e35/System.Web.Extensions.Design.dll
--- mono-2.6.4/data/system.web.extensions_1.0.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/system.web.extensions_1.0.pc-new.in	2010-04-27 00:39:00.000000000 +0100
@@ -1,4 +1,4 @@
 Name: System.Web.Extensions
 Description: System.Web.Extensions ASP.NET 2.0 add-on
 Version: 1.0.61025.0
-Libs: -r:@prefix@/lib/mono/gac/System.Web.Extensions/1.0.61025.0__31bf3856ad364e35/System.Web.Extensions.dll
+Libs: -r:@LIBDIR@/mono/gac/System.Web.Extensions/1.0.61025.0__31bf3856ad364e35/System.Web.Extensions.dll
--- mono-2.6.4/data/system.web.mvc.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/system.web.mvc.pc-new.in	2010-04-27 00:39:16.000000000 +0100
@@ -1,4 +1,4 @@
 Name: System.Web.Mvc
 Description: System.Web.Mvc - ASP.NET MVC
 Version: 1.0.0.0
-Libs: -r:@prefix@/lib/mono/gac/System.Web.Mvc/1.0.0.0__31bf3856ad364e35/System.Web.Mvc.dll
+Libs: -r:@LIBDIR@/mono/gac/System.Web.Mvc/1.0.0.0__31bf3856ad364e35/System.Web.Mvc.dll

--- mono-2.8/configure.in	2010-09-10 21:55:45.000000000 +0100
+++ mono-2.8/configure-new.in	2010-09-15 14:01:05.640710876 +0100
@@ -2886,7 +2886,7 @@
     echo "prefix=$prefix" > $mcs_topdir/build/config.make
     echo "exec_prefix=$exec_prefix" >> $mcs_topdir/build/config.make
     echo "sysconfdir=$sysconfdir" >> $mcs_topdir/build/config.make
-    echo 'mono_libdir=${exec_prefix}/lib' >> $mcs_topdir/build/config.make
+    echo 'mono_libdir=@LIBDIR@' >> $mcs_topdir/build/config.make
     echo 'MCS_FLAGS = $(PLATFORM_DEBUG_FLAGS)' >> $mcs_topdir/build/config.make
     echo 'IL_FLAGS = /debug' >> $mcs_topdir/build/config.make
     echo "RUNTIME = $mono_build_root/runtime/mono-wrapper" >> $mcs_topdir/build/config.make
@@ -2937,7 +2937,7 @@
 
       echo "prefix=$prefix" > $srcdir/$olivedir/build/config.make
       echo "exec_prefix=$exec_prefix" >> $srcdir/$olivedir/build/config.make
-      echo 'mono_libdir=${exec_prefix}/lib' >> $srcdir/$olivedir/build/config.make
+      echo 'mono_libdir=@LIBDIR@' >> $srcdir/$olivedir/build/config.make
       echo 'MCS_FLAGS = $(PLATFORM_DEBUG_FLAGS)' >> $srcdir/$olivedir/build/config.make
       echo "RUNTIME = $mono_build_root/runtime/mono-wrapper" >> $srcdir/$olivedir/build/config.make
       echo "MONO_VERSION = $myver" >> $srcdir/$olivedir/build/config.make
--- mono-2.8/data/monodoc.pc.in	2010-09-10 21:55:45.000000000 +0100
+++ mono-2.8/data/monodoc.pc-new.in	2010-09-22 12:43:38.220677135 +0100
@@ -1,9 +1,9 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-libdir=@prefix@/lib
+libdir=@LIBDIR@
 sourcesdir=${libdir}/monodoc/sources
 
 Name: Monodoc
 Description: Monodoc - Mono Documentation Tools
 Version: @VERSION@
-Libs: /r:${prefix}/lib/mono/monodoc/monodoc.dll 
+Libs: /r:${libdir}/mono/monodoc/monodoc.dll 
--- mono-2.8/data/mono-cairo.pc.in	2010-09-10 21:55:45.000000000 +0100
+++ mono-2.8/data/mono-cairo.pc-new.in	2010-09-22 12:42:53.873541601 +0100
@@ -1,9 +1,9 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${pcfiledir}/../..
-libdir=${prefix}/@reloc_libdir@
+libdir=@LIBDIR@
 includedir=${prefix}/include
 
 Name: Mono.Cairo
 Description: Cairo bindings for Mono
 Version: @VERSION@
-Libs: -r:${prefix}/lib/mono/2.0/Mono.Cairo.dll
+Libs: -r:${libdir}/mono/2.0/Mono.Cairo.dll
--- mono-2.8/data/system.web.mvc2.pc.in	2010-09-10 21:55:45.000000000 +0100
+++ mono-2.8/data/system.web.mvc2.pc-new.in	2010-09-22 12:44:16.452759229 +0100
@@ -1,4 +1,4 @@
 Name: System.Web.Mvc2
 Description: System.Web.Mvc - ASP.NET MVC v2
 Version: 2.0.0.0
-Libs: -r:@prefix@/lib/mono/gac/System.Web.Mvc/2.0.0.0__31bf3856ad364e35/System.Web.Mvc.dll
+Libs: -r:@LIBDIR@/mono/gac/System.Web.Mvc/2.0.0.0__31bf3856ad364e35/System.Web.Mvc.dll
--- mono-2.8/docs/Makefile.in	2010-09-10 21:58:21.000000000 +0100
+++ mono-2.8/docs/Makefile-new.in	2010-09-22 12:51:28.223926963 +0100
@@ -252,7 +252,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-sourcesdir = $(prefix)/lib/monodoc/sources
+sourcesdir = @LIBDIR@/monodoc/sources
 sources_DATA = \
 	mono-file-formats.source \
 	mono-tools.source \
--- mono-2.8/docs/Makefile.am	2010-09-10 21:55:45.000000000 +0100
+++ mono-2.8/docs/Makefile-new.am	2010-09-22 12:49:54.369453974 +0100
@@ -1,4 +1,4 @@
-sourcesdir = $(prefix)/lib/monodoc/sources
+sourcesdir = @LIBDIR@/monodoc/sources
 
 sources_DATA = \
 	mono-file-formats.source \
--- mono-2.8/data/mono-2.pc.in	2010-09-10 21:55:45.000000000 +0100
+++ mono-2.8/data/mono-2.pc-new.in	2010-09-22 12:54:54.445011590 +0100
@@ -1,6 +1,6 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${pcfiledir}/../..
-libdir=${prefix}/@reloc_libdir@
+libdir=@LIBDIR@
 includedir=${prefix}/include/mono-@API_VER@
 sysconfdir=@sysconfdir@
 
--- mono-2.10/mono/metadata/assembly.c	2011-01-31 20:16:05.000000000 +0000
+++ mono-2.10/mono/metadata/assembly-new.c	2011-02-03 19:04:44.865273789 +0000
@@ -578,7 +578,7 @@
 	}
 
 	config = g_build_filename (base, "etc", NULL);
-	lib = g_build_filename (base, "lib", NULL);
+	lib = g_build_filename (base, MONO_RELOC_LIBDIR, NULL);
 	mono = g_build_filename (lib, "mono/2.0", NULL);
 	if (stat (mono, &buf) == -1)
 		fallback ();
@@ -2136,7 +2136,7 @@
 	if (extra_gac_paths) {
 		paths = extra_gac_paths;
 		while (!res && *paths) {
-			gacpath = g_build_path (G_DIR_SEPARATOR_S, *paths, "lib", "mono", "gac", aname->name, NULL);
+			gacpath = g_build_path (G_DIR_SEPARATOR_S, *paths, MONO_RELOC_LIBDIR, "mono", "gac", aname->name, NULL);
 			res = probe_for_partial_name (gacpath, fullname, aname, status);
 			g_free (gacpath);
 			paths++;
@@ -2266,7 +2266,7 @@
 		paths = extra_gac_paths;
 		while (!image && *paths) {
 			fullpath = g_build_path (G_DIR_SEPARATOR_S, *paths,
-					"lib", "mono", "gac", subpath, NULL);
+					MONO_RELOC_LIBDIR, "mono", "gac", subpath, NULL);
 			image = mono_image_open (fullpath, NULL);
 			g_free (fullpath);
 			paths++;
@@ -2572,7 +2572,7 @@
 	if (extra_gac_paths) {
 		paths = extra_gac_paths;
 		while (!result && *paths) {
-			fullpath = g_build_path (G_DIR_SEPARATOR_S, *paths, "lib", "mono", "gac", subpath, NULL);
+			fullpath = g_build_path (G_DIR_SEPARATOR_S, *paths, MONO_RELOC_LIBDIR, "mono", "gac", subpath, NULL);
 			result = mono_assembly_open_full (fullpath, status, refonly);
 			g_free (fullpath);
 			paths++;
--- mono-2.10/data/cecil.pc.in	2011-01-31 20:16:05.000000000 +0000
+++ mono-2.10/data/cecil.pc-new.in	2011-02-03 19:10:50.904123791 +0000
@@ -1,4 +1,4 @@
-assemblies_dir=@prefix@/lib/mono
+assemblies_dir=@LIBDIR@/mono
 Libraries=${assemblies_dir}/gac/Mono.Cecil/0.9.4.0__0738eb9f132ed756/Mono.Cecil.dll
 
 Name: Mono Internal -- Do not use.
--- mono-2.10/scripts/Makefile.am	2011-01-31 20:16:05.000000000 +0000
+++ mono-2.10/scripts/Makefile-new.am	2011-02-03 19:14:37.488167039 +0000
@@ -171,7 +171,7 @@
 endif
 else
 plat_bindir = $(bindir)
-mono_instdir = $(prefix)/lib/mono
+mono_instdir = @LIBDIR@/mono
 endif
 
 
--- mono-2.10/mcs/build/config-default.make	2011-01-31 20:16:05.000000000 +0000
+++ mono-2.10/mcs/build/config-default-new.make	2011-02-03 20:01:19.554349792 +0000
@@ -19,7 +19,7 @@
 endif
 prefix = /usr/local
 exec_prefix = $(prefix)
-mono_libdir = $(exec_prefix)/lib
+mono_libdir = @LIBDIR@
 sysconfdir = $(prefix)/etc
 #RUNTIME = mono
 RUNTIME = false
