diff -ruN plymouth-0.8.6-orig/src/libply/ply-key-file.c plymouth-0.8.6/src/libply/ply-key-file.c
--- plymouth-0.8.6-orig/src/libply/ply-key-file.c	2012-09-17 15:02:17.616571842 +0800
+++ plymouth-0.8.6/src/libply/ply-key-file.c	2012-09-17 15:03:30.263238211 +0800
@@ -313,6 +313,12 @@
   return  ply_hashtable_lookup (key_file->groups, (void *) group_name);
 }
 
+bool ply_key_file_has_group (ply_key_file_t *key_file, 
+			     const char *group_name)
+{
+  return  ply_hashtable_lookup (key_file->groups, (void *) group_name) ? true : false;
+}
+
 static ply_key_file_entry_t *
 ply_key_file_find_entry (ply_key_file_t       *key_file,
                          ply_key_file_group_t *group,
diff -ruN plymouth-0.8.6-orig/src/libply/ply-key-file.h plymouth-0.8.6/src/libply/ply-key-file.h
--- plymouth-0.8.6-orig/src/libply/ply-key-file.h	2012-09-17 15:02:17.616571842 +0800
+++ plymouth-0.8.6/src/libply/ply-key-file.h	2012-09-17 15:03:57.489904813 +0800
@@ -45,6 +45,8 @@
 void ply_key_file_foreach_entry (ply_key_file_t              *key_file,
                                  ply_key_file_foreach_func_t  func,
                                  void                        *user_data);
+bool ply_key_file_has_group (ply_key_file_t *key_file, 
+			     const char *group_name);
 #endif
 
 #endif /* PLY_KEY_FILE_H */
diff -ruN plymouth-0.8.6-orig/src/libply-splash-core/ply-text-progress-bar.c plymouth-0.8.6/src/libply-splash-core/ply-text-progress-bar.c
--- plymouth-0.8.6-orig/src/libply-splash-core/ply-text-progress-bar.c	2012-09-17 15:02:17.616571842 +0800
+++ plymouth-0.8.6/src/libply-splash-core/ply-text-progress-bar.c	2012-09-17 15:09:56.519903309 +0800
@@ -51,13 +51,16 @@
 
 #include <linux/kd.h>
 
+#include <sys/utsname.h>
+#include <ctype.h>
+
 #ifndef FRAMES_PER_SECOND
 #define FRAMES_PER_SECOND 5
 #endif
 
 #define NUMBER_OF_INDICATOR_COLUMNS 6
 
-static char *os_string;
+static char *os_string = "";
 
 struct _ply_text_progress_bar
 {
@@ -104,6 +107,7 @@
 
   buf = NULL;
 
+  buf = calloc (sbuf.st_size + 1, sizeof(char));
   fd = open (RELEASE_FILE, O_RDONLY);
   if (fd == -1)
     goto out;
@@ -113,7 +117,6 @@
     goto out;
   }
 
-  buf = calloc (sbuf.st_size + 1, sizeof(char));
   read (fd, buf, sbuf.st_size);
   close (fd);
 
@@ -160,9 +163,23 @@
 
 out:
   free (buf);
+  if (strcmp(os_string, ""))
+             return;
+  struct  utsname str_uname;
+  int i, len;
+  if (uname(&str_uname))
+      asprintf(&os_string, " %s", "StartOS");
+  else
+  {   
+      len = strlen(str_uname.release);
+      for (i = 0; i < len; i++)
+      {   
+	  if (islower(str_uname.release[i]))
+	      str_uname.release[i] = toupper(str_uname.release[i]);
+      }    
 
-  if (os_string == NULL)
-    os_string = strdup ("");
+      asprintf(&os_string, " %s", str_uname.release);
+  }
 }
 
 void
diff -ruN plymouth-0.8.6-orig/src/libply-splash-graphics/ply-throbber.h plymouth-0.8.6/src/libply-splash-graphics/ply-throbber.h
--- plymouth-0.8.6-orig/src/libply-splash-graphics/ply-throbber.h	2012-09-17 15:02:17.613238528 +0800
+++ plymouth-0.8.6/src/libply-splash-graphics/ply-throbber.h	2012-09-17 15:11:22.173236299 +0800
@@ -33,6 +33,11 @@
 
 typedef struct _ply_throbber ply_throbber_t;
 
+#ifndef FRAMES_PER_SECOND
+long int frames_per_second  = 30; 
+#define FRAMES_PER_SECOND frames_per_second
+#endif
+
 #ifndef PLY_HIDE_FUNCTION_DECLARATIONS
 ply_throbber_t *ply_throbber_new (const char *image_dir,
                                   const char *frames_prefix);
diff -ruN plymouth-0.8.6-orig/src/plugins/renderers/frame-buffer/plugin.c plymouth-0.8.6/src/plugins/renderers/frame-buffer/plugin.c
--- plymouth-0.8.6-orig/src/plugins/renderers/frame-buffer/plugin.c	2012-09-17 15:02:17.609905224 +0800
+++ plymouth-0.8.6/src/plugins/renderers/frame-buffer/plugin.c	2012-09-17 15:15:07.123235387 +0800
@@ -41,6 +41,7 @@
 #include <sys/types.h>
 #include <values.h>
 #include <unistd.h>
+#include <errno.h>
 
 #include <linux/fb.h>
 
@@ -255,6 +256,21 @@
   else
     backend->device_name =
       strdup (PLY_FRAME_BUFFER_DEFAULT_FB_DEVICE_NAME);
+ 
+  if (access(backend->device_name, F_OK))
+  {
+      ply_trace ("invalid frame buffer device : %s : %s", backend->device_name, strerror(errno));
+      if (access("/dev/fb0", F_OK))
+      {
+	  ply_trace ("device /dev/fb0 : %s", strerror(errno));
+	  goto out_if;
+      }
+      backend->device_name = strdup ("/dev/fb0");
+  }
+
+ out_if:
+
+
 
   ply_trace ("creating renderer backend for device %s", backend->device_name);
 
diff -ruN plymouth-0.8.6-orig/src/plugins/splash/script/plugin.c plymouth-0.8.6/src/plugins/splash/script/plugin.c
--- plymouth-0.8.6-orig/src/plugins/splash/script/plugin.c	2012-09-17 15:02:17.613238528 +0800
+++ plymouth-0.8.6/src/plugins/splash/script/plugin.c	2012-09-17 15:17:12.976568210 +0800
@@ -66,7 +66,8 @@
 #include <linux/kd.h>
 
 #ifndef FRAMES_PER_SECOND
-#define FRAMES_PER_SECOND 50
+long int frames_per_second = 50;
+#define FRAMES_PER_SECOND frames_per_second
 #endif
 
 struct _ply_boot_splash_plugin
@@ -168,6 +169,7 @@
 static ply_boot_splash_plugin_t *
 create_plugin (ply_key_file_t *key_file)
 {
+  char *frame_freq = NULL;
   ply_boot_splash_plugin_t *plugin;
   plugin = calloc (1, sizeof (ply_boot_splash_plugin_t));
   plugin->image_dir = ply_key_file_get_value (key_file, 
@@ -176,6 +178,14 @@
   plugin->script_filename = ply_key_file_get_value (key_file,
                                                     "script",
                                                     "ScriptFile");
+  frame_freq = ply_key_file_get_value (key_file,
+                                       "StartOSExt",
+                                       "FramesPerSecond");
+  if (frame_freq)
+  {
+      frames_per_second = strtol(frame_freq, NULL, 0);
+      free (frame_freq);
+  }
 
   plugin->script_env_vars = ply_list_new ();
   ply_key_file_foreach_entry (key_file, add_script_env_var, plugin->script_env_vars);
diff -ruN plymouth-0.8.6-orig/src/plugins/splash/throbgress/plugin.c plymouth-0.8.6/src/plugins/splash/throbgress/plugin.c
--- plymouth-0.8.6-orig/src/plugins/splash/throbgress/plugin.c	2012-09-17 15:02:17.613238528 +0800
+++ plymouth-0.8.6/src/plugins/splash/throbgress/plugin.c	2012-09-17 15:35:07.806563856 +0800
@@ -103,6 +103,8 @@
   uint32_t root_is_mounted : 1;
   uint32_t is_visible : 1;
   uint32_t is_animating : 1;
+  uint32_t ylmfext : 1;
+  char *imageprefix;
 };
 
 ply_boot_splash_plugin_interface_t * ply_boot_splash_plugin_get_interface (void);
@@ -119,9 +121,17 @@
   view->display = display;
 
   view->entry = ply_entry_new (plugin->image_dir);
-  view->throbber = ply_throbber_new (plugin->image_dir,
+  if (plugin->imageprefix)
+  {
+      view->throbber = ply_throbber_new (plugin->image_dir,
+                                     plugin->imageprefix);
+  }
+  else
+  {
+      view->throbber = ply_throbber_new (plugin->image_dir,
                                      "throbber-");
-  view->progress_bar = ply_progress_bar_new ();
+      view->progress_bar = ply_progress_bar_new ();
+  }
   view->label = ply_label_new ();
   view->message_label = ply_label_new ();
   ply_label_set_text (view->message_label, "");
@@ -136,7 +146,8 @@
 
   ply_entry_free (view->entry);
   ply_throbber_free (view->throbber);
-  ply_progress_bar_free (view->progress_bar);
+  if (!view->plugin->ylmfext)
+      ply_progress_bar_free (view->progress_bar);
   ply_label_free (view->label);
   ply_label_free (view->message_label);
 
@@ -287,6 +298,7 @@
 
   unsigned long screen_width, screen_height;
   long width, height;
+  long tho_y;
 
   assert (view != NULL);
 
@@ -306,14 +318,23 @@
 
   width = ply_throbber_get_width (view->throbber);
   height = ply_throbber_get_height (view->throbber);
+  if (view->plugin->ylmfext)
+  {
+      tho_y = (screen_height - height) / 2;
+  }
+  else
+  {
+      tho_y = view->logo_area.y + view->logo_area.height + height / 2;
+      ply_progress_bar_show (view->progress_bar,
+	      view->display,
+	      0, screen_height - ply_progress_bar_get_height (view->progress_bar));
+  }
+
   ply_throbber_start (view->throbber,
                       plugin->loop,
                       view->display,
-                      screen_width / 2.0 - width / 2.0,
-                      view->logo_area.y + view->logo_area.height + height / 2);
-  ply_progress_bar_show (view->progress_bar,
-                         view->display,
-                         0, screen_height - ply_progress_bar_get_height (view->progress_bar));
+                      (screen_width - width) / 2,
+                      tho_y);
   view_redraw (view);
 }
 
@@ -408,7 +429,6 @@
     plugin->background_start_color = PLYMOUTH_BACKGROUND_START_COLOR;
 
   free (color);
-
   color = ply_key_file_get_value (key_file, "throbgress", "BackgroundEndColor");
 
   if (color != NULL)
@@ -417,6 +437,21 @@
     plugin->background_end_color = PLYMOUTH_BACKGROUND_END_COLOR;
 
   free (color);
+ 
+  /* StartOS extension */
+  plugin->ylmfext = false;
+  plugin->imageprefix=NULL;
+  char *frame_freq;
+  if (ply_key_file_has_group (key_file, "StartOSExt"))
+  {
+      plugin->ylmfext = true;
+      plugin->imageprefix = ply_key_file_get_value (key_file, "StartOSExt", "ImagePrefix");
+      if ((frame_freq = ply_key_file_get_value(key_file, "StartOSExt", "FramesPerSecond"))) 
+      {
+	  frames_per_second = strtol (frame_freq, NULL, 0);
+	  free (frame_freq);
+      }
+  }
 
   return plugin;
 }
@@ -441,6 +476,8 @@
 
   free_views (plugin);
 
+  if (plugin->imageprefix)
+      free (plugin->imageprefix);
   free (plugin);
 }
 
@@ -546,7 +583,8 @@
       view = ply_list_node_get_data (node);
       next_node = ply_list_get_next_node (plugin->views, node);
 
-      ply_progress_bar_hide (view->progress_bar);
+      if (!plugin->ylmfext)
+      	ply_progress_bar_hide (view->progress_bar);
       if (trigger != NULL)
         ply_trigger_ignore_next_pull (trigger);
       ply_throbber_stop (view->throbber, trigger);
@@ -593,6 +631,13 @@
                                               &view->lock_area,
                                               lock_data);
     }
+  else if (view->plugin->ylmfext)
+  {
+      y = (height - ply_throbber_get_height(view->throbber)) / 2;
+      ply_throbber_draw_area (view->throbber,
+                              pixel_buffer, x, y, width, height);
+  }
+
   else
     {
       draw_logo (view, pixel_buffer);
@@ -656,15 +701,18 @@
                     ply_boot_splash_mode_t    mode)
 {
   assert (plugin != NULL);
-  assert (plugin->logo_image != NULL);
 
   plugin->loop = loop;
   plugin->mode = mode;
 
-  ply_trace ("loading logo image");
-  if (!ply_image_load (plugin->logo_image))
-    return false;
-
+  if (!plugin->ylmfext)
+  {
+      assert (plugin->logo_image != NULL);
+      ply_trace ("loading logo image");
+      if (!ply_image_load (plugin->logo_image))
+	  return false;
+  }
+ 
   ply_trace ("loading lock image");
   if (!ply_image_load (plugin->lock_image))
     return false;
@@ -706,6 +754,8 @@
   ply_list_node_t *node;
   double total_duration;
 
+  if (plugin->ylmfext)
+      return ;
   total_duration = duration / percent_done;
 
   /* Fun made-up smoothing function to make the growth asymptotic:
diff -ruN plymouth-0.8.6-orig/src/plymouthd.conf plymouth-0.8.6/src/plymouthd.conf
--- plymouth-0.8.6-orig/src/plymouthd.conf	2012-09-17 15:02:17.609905224 +0800
+++ plymouth-0.8.6/src/plymouthd.conf	2012-09-17 15:37:50.063229864 +0800
@@ -1,3 +1,3 @@
 # Administrator customizations go in this file
 #[Daemon]
-#Theme=fade-in
+#Theme=startos
diff -ruN plymouth-0.8.6-orig/src/plymouthd.defaults plymouth-0.8.6/src/plymouthd.defaults
--- plymouth-0.8.6-orig/src/plymouthd.defaults	2012-09-17 15:02:17.609905224 +0800
+++ plymouth-0.8.6/src/plymouthd.defaults	2012-09-17 15:38:40.713229659 +0800
@@ -1,4 +1,4 @@
 # Distribution defaults. Changes to this file will get overwritten during
 # upgrades.
 [Daemon]
-Theme=fade-in
+Theme=startos
