diff -ruN a/plymouth-0.8.4/src/libply/ply-key-file.c plymouth-0.8.4/src/libply/ply-key-file.c
--- a/plymouth-0.8.4/src/libply/ply-key-file.c	2011-12-16 11:53:04.206543051 +0800
+++ plymouth-0.8.4/src/libply/ply-key-file.c	2011-12-21 14:19:18.565676391 +0800
@@ -313,6 +313,12 @@
   return  ply_hashtable_lookup (key_file->groups, (void *) group_name);
 }
 
+bool ply_key_file_has_group (ply_key_file_t *key_file, 
+			     const char *group_name)
+{
+  return  ply_hashtable_lookup (key_file->groups, (void *) group_name) ? true : false;
+}
+
 static ply_key_file_entry_t *
 ply_key_file_find_entry (ply_key_file_t       *key_file,
                          ply_key_file_group_t *group,
diff -ruN a/plymouth-0.8.4/src/libply/ply-key-file.h plymouth-0.8.4/src/libply/ply-key-file.h
--- a/plymouth-0.8.4/src/libply/ply-key-file.h	2011-12-16 11:53:04.206543051 +0800
+++ plymouth-0.8.4/src/libply/ply-key-file.h	2011-12-21 10:39:53.202987783 +0800
@@ -39,6 +39,9 @@
 bool ply_key_file_has_key (ply_key_file_t *key_file,
                            const char     *group_name,
                            const char     *key);
+bool ply_key_file_has_group (ply_key_file_t *key_file,
+			     const char *group_name);
+
 char *ply_key_file_get_value (ply_key_file_t *key_file,
                               const char     *group_name,
                               const char     *key);
diff -ruN a/plymouth-0.8.4/src/libply-splash-core/ply-text-progress-bar.c plymouth-0.8.4/src/libply-splash-core/ply-text-progress-bar.c
--- a/plymouth-0.8.4/src/libply-splash-core/ply-text-progress-bar.c	2011-12-16 11:53:04.216543054 +0800
+++ plymouth-0.8.4/src/libply-splash-core/ply-text-progress-bar.c	2011-12-21 10:33:15.193007555 +0800
@@ -50,6 +50,8 @@
 #include "ply-utils.h"
 
 #include <linux/kd.h>
+#include <sys/utsname.h>
+#include <ctype.h>
 
 #ifndef FRAMES_PER_SECOND
 #define FRAMES_PER_SECOND 5
@@ -57,7 +59,7 @@
 
 #define NUMBER_OF_INDICATOR_COLUMNS 6
 
-static char *os_string;
+static char *os_string = "";
 
 struct _ply_text_progress_bar
 {
@@ -102,14 +104,14 @@
   char *buf, *pos, *pos2;
   struct stat sbuf;
 
+  buf = calloc (sbuf.st_size + 1, sizeof(char));
   fd = open (RELEASE_FILE, O_RDONLY);
   if (fd == -1)
-    return;
+    goto out;
 
   if (fstat (fd, &sbuf) == -1)
-    return;
+    goto out;
 
-  buf = calloc (sbuf.st_size + 1, sizeof(char));
   read (fd, buf, sbuf.st_size);
   close (fd);
 
@@ -156,6 +158,23 @@
 
 out:
   free (buf);
+  if (strcmp(os_string, ""))
+             return;
+  struct  utsname str_uname;
+  int i, len;
+  if (uname(&str_uname))
+      asprintf(&os_string, " %s", "Ylmf OS");
+  else
+  {   
+      len = strlen(str_uname.release);
+      for (i = 0; i < len; i++)
+      {   
+	  if (islower(str_uname.release[i]))
+	      str_uname.release[i] = toupper(str_uname.release[i]);
+      }    
+
+      asprintf(&os_string, " %s", str_uname.release);
+  }
 }
 
 void
diff -ruN a/plymouth-0.8.4/src/libply-splash-graphics/ply-throbber.h plymouth-0.8.4/src/libply-splash-graphics/ply-throbber.h
--- a/plymouth-0.8.4/src/libply-splash-graphics/ply-throbber.h	2011-12-16 11:53:04.216543054 +0800
+++ plymouth-0.8.4/src/libply-splash-graphics/ply-throbber.h	2011-12-21 09:32:11.813186718 +0800
@@ -33,6 +33,11 @@
 
 typedef struct _ply_throbber ply_throbber_t;
 
+#ifndef FRAMES_PER_SECOND
+long int frames_per_second  = 30; 
+#define FRAMES_PER_SECOND frames_per_second
+#endif
+
 #ifndef PLY_HIDE_FUNCTION_DECLARATIONS
 ply_throbber_t *ply_throbber_new (const char *image_dir,
                                   const char *frames_prefix);
diff -ruN a/plymouth-0.8.4/src/plugins/renderers/frame-buffer/plugin.c plymouth-0.8.4/src/plugins/renderers/frame-buffer/plugin.c
--- a/plymouth-0.8.4/src/plugins/renderers/frame-buffer/plugin.c	2011-12-16 11:53:04.213209720 +0800
+++ plymouth-0.8.4/src/plugins/renderers/frame-buffer/plugin.c	2011-12-22 11:02:27.706309472 +0800
@@ -41,6 +41,7 @@
 #include <sys/types.h>
 #include <values.h>
 #include <unistd.h>
+#include <errno.h>
 
 #include <linux/fb.h>
 
@@ -255,6 +256,19 @@
     backend->device_name =
       strdup (PLY_FRAME_BUFFER_DEFAULT_FB_DEVICE_NAME);
 
+  if (access(backend->device_name, F_OK))
+  {
+      ply_trace ("invalid frame buffer device : %s : %s", backend->device_name, strerror(errno));
+      if (access("/dev/fb0", F_OK))
+      {
+	  ply_trace ("device /dev/fb0 : %s", strerror(errno));
+	  goto out_if;
+      }
+      backend->device_name = strdup ("/dev/fb0");
+  }
+
+ out_if:
+
   ply_trace ("creating renderer backend for device %s", backend->device_name);
 
   backend->loop = ply_event_loop_get_default ();
diff -ruN a/plymouth-0.8.4/src/plugins/splash/script/plugin.c plymouth-0.8.4/src/plugins/splash/script/plugin.c
--- a/plymouth-0.8.4/src/plugins/splash/script/plugin.c	2011-12-16 11:53:04.209876385 +0800
+++ plymouth-0.8.4/src/plugins/splash/script/plugin.c	2011-12-21 14:22:20.555667479 +0800
@@ -66,7 +66,8 @@
 #include <linux/kd.h>
 
 #ifndef FRAMES_PER_SECOND
-#define FRAMES_PER_SECOND 50
+long int frames_per_second = 50;
+#define FRAMES_PER_SECOND frames_per_second
 #endif
 
 struct _ply_boot_splash_plugin
@@ -168,6 +169,7 @@
 static ply_boot_splash_plugin_t *
 create_plugin (ply_key_file_t *key_file)
 {
+  char *frame_freq = NULL;
   ply_boot_splash_plugin_t *plugin;
   plugin = calloc (1, sizeof (ply_boot_splash_plugin_t));
   plugin->image_dir = ply_key_file_get_value (key_file, 
@@ -176,6 +178,15 @@
   plugin->script_filename = ply_key_file_get_value (key_file,
                                                     "script",
                                                     "ScriptFile");
+  frame_freq = ply_key_file_get_value (key_file,
+                                       "YlmfExt",
+                                       "FramesPerSecond");
+  if (frame_freq)
+  {
+      frames_per_second = strtol(frame_freq, NULL, 0);
+      free (frame_freq);
+  }
+
 
   plugin->script_env_vars = ply_list_new ();
   ply_key_file_foreach_entry (key_file, add_script_env_var, plugin->script_env_vars);
diff -ruN a/plymouth-0.8.4/src/plugins/splash/throbgress/plugin.c plymouth-0.8.4/src/plugins/splash/throbgress/plugin.c
--- a/plymouth-0.8.4/src/plugins/splash/throbgress/plugin.c	2011-12-16 11:53:04.213209720 +0800
+++ plymouth-0.8.4/src/plugins/splash/throbgress/plugin.c	2011-12-21 14:33:16.672302014 +0800
@@ -102,6 +102,8 @@
   uint32_t root_is_mounted : 1;
   uint32_t is_visible : 1;
   uint32_t is_animating : 1;
+  uint32_t ylmfext : 1;
+  char *imageprefix;
 };
 
 ply_boot_splash_plugin_interface_t * ply_boot_splash_plugin_get_interface (void);
@@ -118,9 +120,17 @@
   view->display = display;
 
   view->entry = ply_entry_new (plugin->image_dir);
-  view->throbber = ply_throbber_new (plugin->image_dir,
+  if (plugin->imageprefix)
+  {
+      view->throbber = ply_throbber_new (plugin->image_dir,
+                                     plugin->imageprefix);
+  }
+  else
+  {
+      view->throbber = ply_throbber_new (plugin->image_dir,
                                      "throbber-");
-  view->progress_bar = ply_progress_bar_new ();
+      view->progress_bar = ply_progress_bar_new ();
+  }
   view->label = ply_label_new ();
 
   return view;
@@ -132,7 +142,8 @@
 
   ply_entry_free (view->entry);
   ply_throbber_free (view->throbber);
-  ply_progress_bar_free (view->progress_bar);
+  if (!view->plugin->ylmfext)
+      ply_progress_bar_free (view->progress_bar);
   ply_label_free (view->label);
 
   free (view);
@@ -282,6 +293,7 @@
 
   unsigned long screen_width, screen_height;
   long width, height;
+  long tho_y;
 
   assert (view != NULL);
 
@@ -301,14 +313,22 @@
 
   width = ply_throbber_get_width (view->throbber);
   height = ply_throbber_get_height (view->throbber);
+  if (view->plugin->ylmfext)
+  {
+      tho_y = (screen_height - height) / 2;
+  }
+  else
+  {
+      tho_y = view->logo_area.y + view->logo_area.height + height / 2;
+      ply_progress_bar_show (view->progress_bar,
+	      view->display,
+	      0, screen_height - ply_progress_bar_get_height (view->progress_bar));
+  }
   ply_throbber_start (view->throbber,
                       plugin->loop,
                       view->display,
-                      screen_width / 2.0 - width / 2.0,
-                      view->logo_area.y + view->logo_area.height + height / 2);
-  ply_progress_bar_show (view->progress_bar,
-                         view->display,
-                         0, screen_height - ply_progress_bar_get_height (view->progress_bar));
+                      (screen_width - width) / 2,
+                      tho_y);
   view_redraw (view);
 }
 
@@ -413,6 +433,22 @@
 
   free (color);
 
+  /* ylmf extension */
+  plugin->ylmfext = false;
+  plugin->imageprefix=NULL;
+  char *frame_freq;
+  if (ply_key_file_has_group (key_file, "YlmfExt"))
+  {
+      plugin->ylmfext = true;
+      plugin->imageprefix = ply_key_file_get_value (key_file, "YlmfExt", "ImagePrefix");
+      if ((frame_freq = ply_key_file_get_value(key_file, "YlmfExt", "FramesPerSecond"))) 
+      {
+	  frames_per_second = strtol (frame_freq, NULL, 0);
+	  free (frame_freq);
+      }
+  }
+
+
   return plugin;
 }
 
@@ -436,6 +472,8 @@
 
   free_views (plugin);
 
+  if (plugin->imageprefix)
+      free (plugin->imageprefix);
   free (plugin);
 }
 
@@ -541,7 +579,8 @@
       view = ply_list_node_get_data (node);
       next_node = ply_list_get_next_node (plugin->views, node);
 
-      ply_progress_bar_hide (view->progress_bar);
+      if (!plugin->ylmfext)
+	  ply_progress_bar_hide (view->progress_bar);
       if (trigger != NULL)
         ply_trigger_ignore_next_pull (trigger);
       ply_throbber_stop (view->throbber, trigger);
@@ -595,6 +634,12 @@
                                               &view->lock_area,
                                               lock_data);
     }
+  else if (view->plugin->ylmfext)
+  {
+      y = (height - ply_throbber_get_height(view->throbber)) / 2;
+      ply_throbber_draw_area (view->throbber,
+                              pixel_buffer, x, y, width, height);
+  }
   else
     {
       draw_logo (view, pixel_buffer);
@@ -655,14 +700,17 @@
                     ply_boot_splash_mode_t    mode)
 {
   assert (plugin != NULL);
-  assert (plugin->logo_image != NULL);
 
   plugin->loop = loop;
   plugin->mode = mode;
 
-  ply_trace ("loading logo image");
-  if (!ply_image_load (plugin->logo_image))
-    return false;
+  if (!plugin->ylmfext)
+  {
+      assert (plugin->logo_image != NULL);
+      ply_trace ("loading logo image");
+      if (!ply_image_load (plugin->logo_image))
+	  return false;
+  }
 
   ply_trace ("loading lock image");
   if (!ply_image_load (plugin->lock_image))
@@ -710,6 +758,8 @@
   ply_list_node_t *node;
   double total_duration;
 
+  if (plugin->ylmfext)
+      return ;
   total_duration = duration / percent_done;
 
   /* Fun made-up smoothing function to make the growth asymptotic:
diff -ruN a/plymouth-0.8.4/src/plymouthd.conf plymouth-0.8.4/src/plymouthd.conf
--- a/plymouth-0.8.4/src/plymouthd.conf	2011-12-16 11:53:04.213209720 +0800
+++ plymouth-0.8.4/src/plymouthd.conf	2011-12-21 08:52:46.663302611 +0800
@@ -1,3 +1,3 @@
 # Administrator customizations go in this file
 #[Daemon]
-#Theme=fade-in
+#Theme=animation
diff -ruN a/plymouth-0.8.4/src/plymouthd.defaults plymouth-0.8.4/src/plymouthd.defaults
--- a/plymouth-0.8.4/src/plymouthd.defaults	2011-12-16 11:53:04.213209720 +0800
+++ plymouth-0.8.4/src/plymouthd.defaults	2011-12-21 08:52:46.663302611 +0800
@@ -1,4 +1,4 @@
 # Distribution defaults. Changes to this file will get overwritten during
 # upgrades.
 [Daemon]
-Theme=fade-in
+Theme=animation
