diff -ur plymouth-0.8.2-ylmf4/src/plymouthd.conf plymouth-0.8.2-ylmf4.new//src/plymouthd.conf
--- plymouth-0.8.2-ylmf4/src/plymouthd.conf	2010-03-24 06:02:10.000000000 +0800
+++ plymouth-0.8.2-ylmf4.new//src/plymouthd.conf	2011-11-18 10:17:29.755633249 +0800
@@ -1,3 +1,3 @@
 # Administrator customizations go in this file
-#[Daemon]
-#Theme=fade-in
+[Daemon]
+Theme=ylmfos
diff -ur plymouth-0.8.2-ylmf4/src/plymouthd.defaults plymouth-0.8.2-ylmf4.new//src/plymouthd.defaults
--- plymouth-0.8.2-ylmf4/src/plymouthd.defaults	2010-03-24 10:57:04.000000000 +0800
+++ plymouth-0.8.2-ylmf4.new//src/plymouthd.defaults	2011-11-18 10:17:19.545501917 +0800
@@ -1,4 +1,4 @@
 # Distribution defaults. Changes to this file will get overwritten during
 # upgrades.
 [Daemon]
-Theme=fade-in
+Theme=ylmfos
