diff -ur grub2-1.98-ylmf2/util/grub.d/10_linux.in grub2-1.98-ylmf2.new/util/grub.d/10_linux.in
--- grub2-1.98-ylmf2/util/grub.d/10_linux.in	2011-05-25 16:06:06.456227857 +0800
+++ grub2-1.98-ylmf2.new/util/grub.d/10_linux.in	2011-05-25 16:08:11.286226202 +0800
@@ -50,6 +50,11 @@
   LINUX_ROOT_DEVICE=UUID=${GRUB_DEVICE_UUID}
 fi
 
+# add crashkernel option if we have the required tools
+if [ -x "/usr/bin/makedumpfile" ] && [ -x "/sbin/kexec" ]; then
+    GRUB_CMDLINE_EXTRA="$GRUB_CMDLINE_EXTRA crashkernel=384M-2G:64M,2G-:128M"
+fi
+
 linux_entry ()
 {
   os="$1"
@@ -130,7 +135,7 @@
   fi
 
   linux_entry "${OS}" "${version}" false \
-      "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT}"
+  	"${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_EXTRA} ${GRUB_CMDLINE_LINUX_DEFAULT}"
   if [ "x${GRUB_DISABLE_LINUX_RECOVERY}" != "xtrue" ]; then
     linux_entry "${OS}" "${version}" true \
 	"single ${GRUB_CMDLINE_LINUX}"
