diff -ur grub2-1.98-ylmf2/util/grub.d/10_linux.in grub2-1.98-ylmf2.new/util/grub.d/10_linux.in
--- grub2-1.98-ylmf2/util/grub.d/10_linux.in	2011-05-25 16:23:16.959547536 +0800
+++ grub2-1.98-ylmf2.new/util/grub.d/10_linux.in	2011-05-25 16:25:34.619545712 +0800
@@ -69,17 +69,8 @@
   printf "menuentry \"${title}\" ${CLASS} {\n" "${os}" "${version}"
   save_default_entry | sed -e "s/^/\t/"
 
-  # Use ELILO's generic "efifb" when it's known to be available.
-  # FIXME: We need an interface to select vesafb in case efifb can't be used.
-  if [ "x$GRUB_GFXPAYLOAD_LINUX" = x ]; then
-      if grep -qx "CONFIG_FB_EFI=y" /boot/config-${version} 2> /dev/null \
-	  && grep -qx "CONFIG_VT_HW_CONSOLE_BINDING=y" /boot/config-${version} 2> /dev/null; then
-	  cat << EOF
-	set gfxpayload=keep
-EOF
-      fi
-  else
-	  cat << EOF
+  if [ "x$GRUB_GFXPAYLOAD_LINUX" != x ]; then
+        cat << EOF
 	set gfxpayload=$GRUB_GFXPAYLOAD_LINUX
 EOF
   fi
