--- util/grub.d/10_linux.in	2010-03-07 04:51:37.000000000 +0800
+++ util/grub.d/10_linux.in.new	2010-11-11 11:47:56.622000056 +0800
@@ -28,7 +28,12 @@
 CLASS="--class gnu-linux --class gnu --class os"
 
 if [ "x${GRUB_DISTRIBUTOR}" = "x" ] ; then
-  OS=GNU/Linux
+  if [ -f "/etc/lsb-release" ]; then
+     . /etc/lsb-release
+     OS=$DISTRIB_ID
+  else
+     OS=GNU/Linux
+  fi
 else
   OS="${GRUB_DISTRIBUTOR} GNU/Linux"
   CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr '[A-Z]' '[a-z]') ${CLASS}"
