diff -ur grub2-1.98-ylmf3/util/grub.d/00_header.in grub2-1.98-ylmf3.new/util/grub.d/00_header.in
--- grub2-1.98-ylmf3/util/grub.d/00_header.in	2011-06-28 15:32:50.333014965 +0800
+++ grub2-1.98-ylmf3.new/util/grub.d/00_header.in	2011-06-28 15:34:09.413013919 +0800
@@ -16,6 +16,8 @@
 # You should have received a copy of the GNU General Public License
 # along with GRUB.  If not, see <http://www.gnu.org/licenses/>.
 
+export PATH=$PATH
+
 transform="@program_transform_name@"
 
 prefix=@prefix@
diff -ur grub2-1.98-ylmf3/util/grub.d/10_linux.in grub2-1.98-ylmf3.new/util/grub.d/10_linux.in
--- grub2-1.98-ylmf3/util/grub.d/10_linux.in	2011-06-28 15:32:50.679681628 +0800
+++ grub2-1.98-ylmf3.new/util/grub.d/10_linux.in	2011-06-28 15:34:30.529680305 +0800
@@ -16,6 +16,8 @@
 # You should have received a copy of the GNU General Public License
 # along with GRUB.  If not, see <http://www.gnu.org/licenses/>.
 
+export PATH=$PATH
+
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 bindir=@bindir@
diff -ur grub2-1.98-ylmf3/util/grub.d/30_os-prober.in grub2-1.98-ylmf3.new/util/grub.d/30_os-prober.in
--- grub2-1.98-ylmf3/util/grub.d/30_os-prober.in	2011-06-28 15:32:50.056348303 +0800
+++ grub2-1.98-ylmf3.new/util/grub.d/30_os-prober.in	2011-06-28 15:34:40.706346836 +0800
@@ -16,6 +16,8 @@
 # You should have received a copy of the GNU General Public License
 # along with GRUB.  If not, see <http://www.gnu.org/licenses/>.
 
+export PATH=$PATH
+
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
