diff -ur grub2-1.98-ylmf6/util/grub.d/10_linux.in grub2-1.98-ylmf6.new/util/grub.d/10_linux.in
--- grub2-1.98-ylmf6/util/grub.d/10_linux.in	2011-07-22 16:39:31.426076448 +0800
+++ grub2-1.98-ylmf6.new/util/grub.d/10_linux.in	2011-07-22 16:40:46.157900157 +0800
@@ -74,9 +74,6 @@
     title="$(gettext_quoted "%s, with Linux %s")"
   fi
   printf "menuentry '${title}' ${CLASS} {\n" "${os}" "${version}"
-  cat << EOF
-	recordfail
-EOF
   save_default_entry | sed -e "s/^/\t/"
 
   # Use ELILO's generic "efifb" when it's known to be available.
