diff -ur grub2-1.98-ylmf4/kern/emu/hostdisk.c grub2-1.98-ylmf4.new/kern/emu/hostdisk.c
--- grub2-1.98-ylmf4/kern/emu/hostdisk.c	2011-07-04 11:47:03.426545630 +0800
+++ grub2-1.98-ylmf4.new/kern/emu/hostdisk.c	2011-07-04 11:50:03.629876576 +0800
@@ -204,9 +204,10 @@
   unsigned i;
 
   for (i = 0; i < sizeof (map) / sizeof (map[0]); i++)
-    if (map[i].drive && hook (map[i].drive))
+    {
+      if (map[i].drive  && strncmp(map[i].drive, "fd", 2) && hook (map[i].drive))
       return 1;
-
+    }
   return 0;
 }
 
