--- util/grub-mkconfig_lib.in	2010-03-07 04:51:37.000000000 +0800
+++ util/grub-mkconfig_lib.in.new	2010-10-19 11:40:39.301858438 +0800
@@ -24,6 +24,9 @@
 sbindir=@sbindir@
 pkgdatadir=${datadir}/`echo @PACKAGE_TARNAME@ | sed "${transform}"`
 
+# parepare for devicemap
+/usr/sbin/grub-mkdevicemap --no-floppy
+
 if test "x$grub_probe" = x; then
   grub_probe=${sbindir}/`echo grub-probe | sed ${transform}`
 fi
