--- a/kernel/Makefile.kbuild	2011-07-28 07:55:54.000000000 +0800
+++ b/kernel/Makefile.kbuild.new	2012-02-11 09:57:34.181122292 +0800
@@ -249,8 +249,8 @@
 #
 
 module-install: suser-sanity-check module
-	@mkdir -p $(MODULE_ROOT)/video; \
-	install -m 0664 -o root -g root $(MODULE_OBJECT) $(MODULE_ROOT)/video; \
+	@mkdir -p $(DESTDIR)/$(MODULE_ROOT)/video; \
+	install -m 0664 -o root -g root $(MODULE_OBJECT) $(DESTDIR)/$(MODULE_ROOT)/video; \
 	PATH="$(PATH):/bin:/sbin" depmod -ae;
 
 #
@@ -260,7 +260,7 @@
 
 package-install: module-install rmmod-sanity-check
 	@PATH="$(PATH):/bin:/sbin" modprobe $(MODULE_NAME) && \
-	echo "$(MODULE_OBJECT) installed successfully.";
+	echo "$(MODULE_OBJECT) compiled successfully.";	
 
 #
 # Build an object file suitable for further processing by the installer
