diff -u -r gparted-0.6.4/gparted.desktop gparted-0.6.4.new/gparted.desktop
--- gparted-0.6.4/gparted.desktop	2010-10-01 23:10:10.000000000 +0800
+++ gparted-0.6.4.new/gparted.desktop	2011-03-11 11:37:10.582996002 +0800
@@ -47,9 +47,9 @@
 Name[tr]=GParted
 Name[uk]=GParted
 Name[vi]=GParted
-Name[zh_CN]=GParted
-Name[zh_HK]=GParted
-Name[zh_TW]=GParted
+Name[zh_CN]=GParted 分区编辑器
+Name[zh_HK]=GParted 
+Name[zh_TW]=GParted 分區工具
 GenericName=Partition Editor
 GenericName[ar]=محرر الأقسام
 GenericName[bg]=Редактор на дялове
diff -u -r gparted-0.6.4/po/zh_CN.po gparted-0.6.4.new/po/zh_CN.po
--- gparted-0.6.4/po/zh_CN.po	2010-09-09 06:37:30.000000000 +0800
+++ gparted-0.6.4.new/po/zh_CN.po	2011-03-11 11:36:36.197996002 +0800
@@ -29,7 +29,7 @@
 #: ../src/Win_GParted.cc:70 ../src/Win_GParted.cc:1099
 #: ../src/Win_GParted.cc:1283
 msgid "GParted"
-msgstr "GParted"
+msgstr "GParted 分区编辑器"
 
 #: ../gparted.desktop.in.in.h:3
 msgid "GParted Partition Editor"
