diff -Nurb -x debian likewise-open-6.0.0.53010/lsass/configure.in likewise-open-6.0.0.53010/lsass/configure.in
--- likewise-open-6.0.0.53010/lsass/configure.in	2010-10-19 14:03:01.000000000 -0700
+++ likewise-open-6.0.0.53010/lsass/configure.in	2011-01-23 00:16:40.000000000 -0800
@@ -882,15 +882,6 @@
 # include <wc16str.h>
 #endif
 ])
-AC_CHECK_MEMBERS([ucontext_t.uc_mcontext,
-		  ucontext_t.uc_mcontext->__ss,
-		  ucontext_t.uc_mcontext->__ss.__rip,
-		  ucontext_t.uc_mcontext->__ss.__eip],
-[], [], [AC_INCLUDES_DEFAULT
-#ifdef HAVE_EXECINFO_H
-# include <execinfo.h>
-#endif
-])
 
 if test x"$ENABLE_COMPAT" = x"false"; then
 AC_CHECK_HEADER([reg/reg.h],,[AC_MSG_ERROR([Could not find lwreg headers])],
diff -Nurb -x debian likewise-open-6.0.0.53010/lwdns/lwdns/Makefile.am likewise-open-6.0.0.53010/lwdns/lwdns/Makefile.am
--- likewise-open-6.0.0.53010/lwdns/lwdns/Makefile.am	2010-10-19 13:54:29.000000000 -0700
+++ likewise-open-6.0.0.53010/lwdns/lwdns/Makefile.am	2011-01-23 00:45:20.000000000 -0800
@@ -34,7 +34,8 @@
     @SOCKET_LIBS@    \
     @NSL_LIBS@       \
     @PTHREAD_LIBS@   \
-    $(LWBASE_LIBS)
+    $(LWBASE_LIBS)   \
+    $(LWBASE_NOTHR_LIBS)
 
 liblwdns_la_LDFLAGS = \
     @KRB5_LDFLAGS@   \
diff -Nurb -x debian likewise-open-6.0.0.53010/lwreg/client/Makefile.am likewise-open-6.0.0.53010/lwreg/client/Makefile.am
--- likewise-open-6.0.0.53010/lwreg/client/Makefile.am	2010-10-19 13:54:30.000000000 -0700
+++ likewise-open-6.0.0.53010/lwreg/client/Makefile.am	2011-01-22 23:55:41.000000000 -0800
@@ -11,4 +11,6 @@
     @LWMSG_INCLUDES@
 
 libregclient_la_LIBADD = \
-	@top_builddir@/common/libregcommon.la
+	@top_builddir@/common/libregcommon.la \
+    @LWBASE_LIBS@ \
+    @LWMSG_LIBS@
diff -Nurb -x debian likewise-open-6.0.0.53010/netlogon/tools/set_log_level/Makefile.am likewise-open-6.0.0.53010/netlogon/tools/set_log_level/Makefile.am
--- likewise-open-6.0.0.53010/netlogon/tools/set_log_level/Makefile.am	2010-11-22 11:55:16.000000000 -0800
+++ likewise-open-6.0.0.53010/netlogon/tools/set_log_level/Makefile.am	2011-01-23 00:43:51.000000000 -0800
@@ -5,7 +5,7 @@
 lwnet_set_log_level_LDADD = \
     @top_builddir@/client/liblwnetclientapi.la \
     @top_builddir@/common/liblwnetcommon.la \
-    -llwadvapi
+    -llwadvapi -llwadvapi_nothr
 
 lwnet_set_log_level_SOURCES = \
     main.c
