diff -Nurb likewise-open-6.1.0.48254.orig/build/components/centutils.comp likewise-open-6.1.0.48254/build/components/centutils.comp
--- likewise-open-6.1.0.48254.orig/build/components/centutils.comp	2010-07-29 17:27:56.000000000 -0700
+++ likewise-open-6.1.0.48254/build/components/centutils.comp	2010-08-01 00:35:36.000000000 -0700
@@ -34,7 +34,8 @@
         --prefix=${PREFIXDIR}                  \
         --libdir=${PREFIXDIR}/${_lib} \
         --sysconfdir=${SYSCONFDIR}             \
-        --localstatedir=${LOCALSTATEDIR}
+        --localstatedir=${LOCALSTATEDIR} \
+        --datadir=${DATADIR}
 }
 
 function component_build
diff -Nurb likewise-open-6.1.0.48254.orig/build/components/dcerpc.comp likewise-open-6.1.0.48254/build/components/dcerpc.comp
--- likewise-open-6.1.0.48254.orig/build/components/dcerpc.comp	2010-07-29 17:27:56.000000000 -0700
+++ likewise-open-6.1.0.48254/build/components/dcerpc.comp	2010-08-01 00:36:06.000000000 -0700
@@ -29,6 +29,7 @@
 	--libdir=${PREFIXDIR}/${_lib} \
 	--sysconfdir=${SYSCONFDIR} \
 	--localstatedir=${LOCALSTATEDIR} \
+	--datadir=${DATADIR} \
 	--enable-dependency-tracking
 }
 
diff -Nurb likewise-open-6.1.0.48254.orig/build/components/domainjoin.comp likewise-open-6.1.0.48254/build/components/domainjoin.comp
--- likewise-open-6.1.0.48254.orig/build/components/domainjoin.comp	2010-07-29 17:27:56.000000000 -0700
+++ likewise-open-6.1.0.48254/build/components/domainjoin.comp	2010-08-01 00:36:21.000000000 -0700
@@ -42,6 +42,7 @@
         --libdir=${PREFIXDIR}/${_lib}           \
         --sysconfdir=${SYSCONFDIR}              \
         --localstatedir=${LOCALSTATEDIR}        \
+        --datadir=${DATADIR}                    \
         --disable-static                        \
         --enable-shared                         \
 	--with-gtk=${WANT_GTK} \
diff -Nurb likewise-open-6.1.0.48254.orig/build/components/eventlog.comp likewise-open-6.1.0.48254/build/components/eventlog.comp
--- likewise-open-6.1.0.48254.orig/build/components/eventlog.comp	2010-07-29 17:27:56.000000000 -0700
+++ likewise-open-6.1.0.48254/build/components/eventlog.comp	2010-08-01 00:36:35.000000000 -0700
@@ -30,7 +30,8 @@
 	--prefix=${PREFIXDIR} \
 	--libdir=${PREFIXDIR}/${_lib}  \
         --sysconfdir=${SYSCONFDIR} \
-        --localstatedir=${LOCALSTATEDIR}
+        --localstatedir=${LOCALSTATEDIR} \
+        --datadir=${DATADIR}
 }
 
 function component_build
diff -Nurb likewise-open-6.1.0.48254.orig/build/components/lsass.comp likewise-open-6.1.0.48254/build/components/lsass.comp
--- likewise-open-6.1.0.48254.orig/build/components/lsass.comp	2010-07-29 17:27:56.000000000 -0700
+++ likewise-open-6.1.0.48254/build/components/lsass.comp	2010-08-01 00:37:23.000000000 -0700
@@ -63,6 +63,7 @@
         --libdir=${PREFIXDIR}/${_lib}   \
 	--sysconfdir=${SYSCONFDIR}               \
 	--localstatedir=${LOCALSTATEDIR}         \
+    --datadir=${DATADIR}                     \
         --with-pam-libdir=${PAM_DIR}             \
         --with-nss-libdir=${NSS_DIR}   
 }
diff -Nurb likewise-open-6.1.0.48254.orig/build/components/lwdns.comp likewise-open-6.1.0.48254/build/components/lwdns.comp
--- likewise-open-6.1.0.48254.orig/build/components/lwdns.comp	2010-07-29 17:27:56.000000000 -0700
+++ likewise-open-6.1.0.48254/build/components/lwdns.comp	2010-08-01 00:37:38.000000000 -0700
@@ -31,7 +31,8 @@
 	--prefix=${PREFIXDIR}                   \
         --libdir=${PREFIXDIR}/${_lib}  \
 	--sysconfdir=${SYSCONFDIR}              \
-        --localstatedir=${LOCALSTATEDIR} 
+        --localstatedir=${LOCALSTATEDIR} \
+        --datadir=${DATADIR}
 }
 
 function component_build
diff -Nurb likewise-open-6.1.0.48254.orig/build/components/lwio.comp likewise-open-6.1.0.48254/build/components/lwio.comp
--- likewise-open-6.1.0.48254.orig/build/components/lwio.comp	2010-07-29 17:27:56.000000000 -0700
+++ likewise-open-6.1.0.48254/build/components/lwio.comp	2010-08-01 00:37:48.000000000 -0700
@@ -40,7 +40,8 @@
         --prefix=${PREFIXDIR}                    \
         --libdir=${PREFIXDIR}/${_lib}            \
         --sysconfdir=${SYSCONFDIR}               \
-        --localstatedir=${LOCALSTATEDIR}
+        --localstatedir=${LOCALSTATEDIR} \
+        --datadir=${DATADIR}
 }
 
 function component_build
diff -Nurb likewise-open-6.1.0.48254.orig/build/components/lwreg.comp likewise-open-6.1.0.48254/build/components/lwreg.comp
--- likewise-open-6.1.0.48254.orig/build/components/lwreg.comp	2010-07-29 17:27:56.000000000 -0700
+++ likewise-open-6.1.0.48254/build/components/lwreg.comp	2010-08-01 00:38:14.000000000 -0700
@@ -33,7 +33,8 @@
 	--prefix=${PREFIXDIR}            \
         --libdir=${PREFIXDIR}/${_lib}    \
         --localstatedir=${LOCALSTATEDIR} \
-	--sysconfdir=${SYSCONFDIR}
+	--sysconfdir=${SYSCONFDIR} \
+    --datadir=${DATADIR}
     exit_on_error $?
 }
 
diff -Nurb likewise-open-6.1.0.48254.orig/build/components/lwsm.comp likewise-open-6.1.0.48254/build/components/lwsm.comp
--- likewise-open-6.1.0.48254.orig/build/components/lwsm.comp	2010-07-29 17:27:56.000000000 -0700
+++ likewise-open-6.1.0.48254/build/components/lwsm.comp	2010-08-01 00:38:24.000000000 -0700
@@ -28,7 +28,8 @@
 	--prefix=${PREFIXDIR}            \
         --libdir=${PREFIXDIR}/${_lib}    \
         --localstatedir=${LOCALSTATEDIR} \
-	--sysconfdir=${SYSCONFDIR}
+	--sysconfdir=${SYSCONFDIR} \
+    --datadir=${DATADIR}
     exit_on_error $?
 }
 
diff -Nurb likewise-open-6.1.0.48254.orig/build/components/lwupgrade.comp likewise-open-6.1.0.48254/build/components/lwupgrade.comp
--- likewise-open-6.1.0.48254.orig/build/components/lwupgrade.comp	2010-07-29 17:27:56.000000000 -0700
+++ likewise-open-6.1.0.48254/build/components/lwupgrade.comp	2010-08-01 00:38:35.000000000 -0700
@@ -29,7 +29,8 @@
 	--prefix=${PREFIXDIR}            \
         --libdir=${PREFIXDIR}/${_lib}    \
         --localstatedir=${LOCALSTATEDIR} \
-	--sysconfdir=${SYSCONFDIR}
+	--sysconfdir=${SYSCONFDIR} \
+    --datadir=${DATADIR}
     exit_on_error $?
 }
 
diff -Nurb likewise-open-6.1.0.48254.orig/build/components/netlogon.comp likewise-open-6.1.0.48254/build/components/netlogon.comp
--- likewise-open-6.1.0.48254.orig/build/components/netlogon.comp	2010-07-29 17:27:56.000000000 -0700
+++ likewise-open-6.1.0.48254/build/components/netlogon.comp	2010-08-01 00:38:45.000000000 -0700
@@ -31,7 +31,8 @@
 	--prefix=${PREFIXDIR}                   \
         --libdir=${PREFIXDIR}/${_lib}  \
         --sysconfdir=${SYSCONFDIR}              \
-        --localstatedir=${LOCALSTATEDIR} 
+        --localstatedir=${LOCALSTATEDIR} \
+        --datadir=${DATADIR}
 }
 
 function component_build
diff -Nurb likewise-open-6.1.0.48254.orig/build/components/pstore.comp likewise-open-6.1.0.48254/build/components/pstore.comp
--- likewise-open-6.1.0.48254.orig/build/components/pstore.comp	2010-07-29 17:27:56.000000000 -0700
+++ likewise-open-6.1.0.48254/build/components/pstore.comp	2010-08-01 00:39:16.000000000 -0700
@@ -31,7 +31,8 @@
 	--prefix=${PREFIXDIR}                       \
         --libdir=${PREFIXDIR}/${_lib}      \
 	--sysconfdir=${SYSCONFDIR}                  \
-        --localstatedir=${LOCALSTATEDIR}  
+        --localstatedir=${LOCALSTATEDIR}  \
+        --datadir=${DATADIR}
 }
 
 function component_build
diff -Nurb likewise-open-6.1.0.48254.orig/build/components/srvsvc.comp likewise-open-6.1.0.48254/build/components/srvsvc.comp
--- likewise-open-6.1.0.48254.orig/build/components/srvsvc.comp	2010-07-29 17:27:56.000000000 -0700
+++ likewise-open-6.1.0.48254/build/components/srvsvc.comp	2010-08-01 00:39:30.000000000 -0700
@@ -30,7 +30,8 @@
 	--prefix=${PREFIXDIR} \
 	--libdir=${PREFIXDIR}/${_lib}  \
         --sysconfdir=${SYSCONFDIR} \
-        --localstatedir=${LOCALSTATEDIR}
+        --localstatedir=${LOCALSTATEDIR} \
+        --datadir=${DATADIR}
 }
 
 function component_build
diff -Nurb likewise-open-6.1.0.48254.orig/build/mkcomponent likewise-open-6.1.0.48254/build/mkcomponent
--- likewise-open-6.1.0.48254.orig/build/mkcomponent	2010-07-29 17:27:56.000000000 -0700
+++ likewise-open-6.1.0.48254/build/mkcomponent	2010-08-01 00:40:40.000000000 -0700
@@ -27,12 +27,13 @@
     PREFIXDIR="/opt/${PRODUCTNAME}"
     LIBEXECDIR="${PREFIXDIR}/bin"
     LOCALSTATEDIR="/var/lib/${PRODUCTNAME}"
+    DATADIR="/usr/share/${PRODUCTNAME}"
     STAGE_DIR="${BUILD_ROOT}/staging"
     STAGE_COMP_DIR="${STAGE_DIR}/components"
     STAGE_INSTALL_DIR="${STAGE_DIR}/install-root"
     SYSCONFDIR="/etc/${PRODUCTNAME}"
 
-    for _variable in PREFIXDIR SYSCONFDIR INITDIR LOCALSTATEDIR LIBEXECDIR STAGE_INSTALL_DIR; do
+    for _variable in PREFIXDIR SYSCONFDIR INITDIR LOCALSTATEDIR LIBEXECDIR DATADIR STAGE_INSTALL_DIR; do
         eval "if [ -n \"\${BUILD_${_variable}}\" ]; then export ${_variable}=\${BUILD_${_variable}}; fi"
         eval "if [ -n \"\${BUILD_${_upcomp}_${_variable}}\" ]; then export ${_variable}=\${BUILD_${_upcomp}_${_variable}}; fi"
         if [ -n "${BUILD_VERBOSE}" ]; then
