Index: likewise-open-5.4.0.42111/dcerpc/configure.in
===================================================================
--- likewise-open-5.4.0.42111.orig/dcerpc/configure.in	2010-03-13 04:33:44.000000000 +0000
+++ likewise-open-5.4.0.42111/dcerpc/configure.in	2010-04-09 12:16:58.556288640 +0000
@@ -16,6 +16,7 @@
 case $target_cpu in
 	i?86)	target_cpu="i386";;
 	hppa2.0*) target_cpu="hppa2.0";;
+	arm*)   target_cpu="arm";;
 esac
 
 DCERPCD_SERVICE_TYPE="dword:00000001"
@@ -527,6 +528,7 @@
 include/dce/solaris2.9/Makefile
 include/dce/solaris2.10/Makefile
 include/dce/hpux11.11/Makefile
+include/dce/arm/Makefile
 include/dce/i386/Makefile
 include/dce/hppa2.0/Makefile
 include/dce/powerpc/Makefile
Index: likewise-open-5.4.0.42111/dcerpc/include/dce/Makefile.am
===================================================================
--- likewise-open-5.4.0.42111.orig/dcerpc/include/dce/Makefile.am	2010-03-13 04:33:44.000000000 +0000
+++ likewise-open-5.4.0.42111/dcerpc/include/dce/Makefile.am	2010-04-09 12:16:58.572256590 +0000
@@ -40,4 +40,4 @@
 BUILT_SOURCES=$(IDLHEADERS)
 
 # The dist target needs to know to include these files too
-EXTRA_DIST=i386/* alpha/* AT386/* linux-gnu/* HP800/* sparc/* powerpc/* s390/* s390x/* solaris2.8/* solaris2.10/* darwin6.0/* aix5.2.0.0/* x86_64/*
+EXTRA_DIST=arm/* i386/* alpha/* AT386/* linux-gnu/* HP800/* sparc/* powerpc/* s390/* s390x/* solaris2.8/* solaris2.10/* darwin6.0/* aix5.2.0.0/* x86_64/*
Index: likewise-open-5.4.0.42111/dcerpc/include/dce/arm/Makefile.am
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ likewise-open-5.4.0.42111/dcerpc/include/dce/arm/Makefile.am	2010-04-09 12:18:04.972287953 +0000
@@ -0,0 +1,3 @@
+## Process this file with automake to produce Makefile.in
+dceincludedir = $(includedir)/dce/arm
+dceinclude_HEADERS = marshall.h ndr_rep.h ndrtypes.h
Index: likewise-open-5.4.0.42111/dcerpc/include/dce/arm/marshall.h
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ likewise-open-5.4.0.42111/dcerpc/include/dce/arm/marshall.h	2010-04-09 12:20:32.096250347 +0000
@@ -0,0 +1,38 @@
+/*
+ * 
+ * (c) Copyright 1991 OPEN SOFTWARE FOUNDATION, INC.
+ * (c) Copyright 1991 HEWLETT-PACKARD COMPANY
+ * (c) Copyright 1991 DIGITAL EQUIPMENT CORPORATION
+ * To anyone who acknowledges that this file is provided "AS IS"
+ * without any express or implied warranty:
+ *                 permission to use, copy, modify, and distribute this
+ * file for any purpose is hereby granted without fee, provided that
+ * the above copyright notices and this notice appears in all source
+ * code copies, and that none of the names of Open Software
+ * Foundation, Inc., Hewlett-Packard Company, or Digital Equipment
+ * Corporation be used in advertising or publicity pertaining to
+ * distribution of the software without specific, written prior
+ * permission.  Neither Open Software Foundation, Inc., Hewlett-
+ * Packard Company, nor Digital Equipment Corporation makes any
+ * representations about the suitability of this software for any
+ * purpose.
+ * 
+ */
+/* 
+ */
+/*
+ * marshall.h 
+ * platform dependent (OS + Architecture) file split out from stubbase.h 
+ * for DCE 1.1 code cleanup 
+ *
+ * This file is always included as part of stubbase.h 
+ */
+
+#ifndef _MARSHALL_H 
+#define _MARSHALL_H
+
+/* ARM Architecture can use the default marshalling macros */
+/* #define USE_DEFAULT_NDR_MARSHALLING_MACROS */
+
+
+#endif /* _MARSHALL_H */
Index: likewise-open-5.4.0.42111/dcerpc/include/dce/arm/ndr_rep.h
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ likewise-open-5.4.0.42111/dcerpc/include/dce/arm/ndr_rep.h	2010-04-09 12:20:37.236289232 +0000
@@ -0,0 +1,46 @@
+/*
+ * 
+ * (c) Copyright 1991 OPEN SOFTWARE FOUNDATION, INC.
+ * (c) Copyright 1991 HEWLETT-PACKARD COMPANY
+ * (c) Copyright 1991 DIGITAL EQUIPMENT CORPORATION
+ * To anyone who acknowledges that this file is provided "AS IS"
+ * without any express or implied warranty:
+ *                 permission to use, copy, modify, and distribute this
+ * file for any purpose is hereby granted without fee, provided that
+ * the above copyright notices and this notice appears in all source
+ * code copies, and that none of the names of Open Software
+ * Foundation, Inc., Hewlett-Packard Company, or Digital Equipment
+ * Corporation be used in advertising or publicity pertaining to
+ * distribution of the software without specific, written prior
+ * permission.  Neither Open Software Foundation, Inc., Hewlett-
+ * Packard Company, nor Digital Equipment Corporation makes any
+ * representations about the suitability of this software for any
+ * purpose.
+ * 
+ */
+/* 
+ */
+/*
+ * ndr_rep.h 
+ * platform dependent (OS + Architecture) file split out from stubbase.h 
+ * for DCE 1.1 code cleanup.
+ * This file contains the architecture specific definitions of the 
+ * local scaler data representation used 
+ *
+ * This file is always included as part of stubbase.h 
+ */
+
+#ifndef _NDR_REP_H 
+#define _NDR_REP_H
+
+#ifdef __ARMEL__
+
+#define NDR_LOCAL_INT_REP     ndr_c_int_little_endian
+#define NDR_LOCAL_FLOAT_REP   ndr_c_float_ieee
+#define NDR_LOCAL_CHAR_REP    ndr_c_char_ascii
+
+#else
+#error "Old ARM ABI NOT SUPPORTED!"
+#endif /* __ARMEL__ */
+
+#endif /* _NDR_REP_H */
Index: likewise-open-5.4.0.42111/dcerpc/include/dce/arm/ndrtypes.h
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ likewise-open-5.4.0.42111/dcerpc/include/dce/arm/ndrtypes.h	2010-04-09 12:20:41.844294163 +0000
@@ -0,0 +1,96 @@
+/*
+ * 
+ * (c) Copyright 1991 OPEN SOFTWARE FOUNDATION, INC.
+ * (c) Copyright 1991 HEWLETT-PACKARD COMPANY
+ * (c) Copyright 1991 DIGITAL EQUIPMENT CORPORATION
+ * To anyone who acknowledges that this file is provided "AS IS"
+ * without any express or implied warranty:
+ *                 permission to use, copy, modify, and distribute this
+ * file for any purpose is hereby granted without fee, provided that
+ * the above copyright notices and this notice appears in all source
+ * code copies, and that none of the names of Open Software
+ * Foundation, Inc., Hewlett-Packard Company, or Digital Equipment
+ * Corporation be used in advertising or publicity pertaining to
+ * distribution of the software without specific, written prior
+ * permission.  Neither Open Software Foundation, Inc., Hewlett-
+ * Packard Company, nor Digital Equipment Corporation makes any
+ * representations about the suitability of this software for any
+ * purpose.
+ * 
+ */
+/* 
+ */
+/*
+**  NAME:
+**
+**      ndrtypes.h
+**
+**  FACILITY:
+**
+**      IDL Stub Support Include File
+**
+**  ABSTRACT:
+**
+**  This file is new for DCE 1.1. This is a platform specific file that
+**  defines the base level ndr types. This file is indirectly included 
+**  in all files via the idlbase.h file. 
+**
+*/
+
+/*
+ * This particular file defines the NDR types for a little-endian 
+ * architecture. This file also depends on the presence of a ANSI 
+ * C compiler, in that it uses the signed keyword to create the 
+ * ndr_small_int type.
+ */
+
+#ifndef _NDR_TYPES_H 
+#define  _NDR_TYPES_H
+
+typedef unsigned char 		ndr_boolean;
+#define ndr_false       false
+#define ndr_true        true
+typedef unsigned char 		ndr_byte;
+
+typedef unsigned char 		ndr_char;
+
+typedef signed char 		ndr_small_int;
+
+typedef unsigned char 		ndr_usmall_int;
+
+typedef short int 		ndr_short_int;
+
+typedef unsigned short int	ndr_ushort_int;
+
+typedef long int 		ndr_long_int;
+
+typedef unsigned int     	ndr_ulong_int;
+
+/* 
+ * the reps for hyper must match the little-endian NDR rep since
+ *  defined(vax) || defined(M_I86) => defined(ALIGNED_SCALAR_ARRAYS) 
+ */
+
+struct ndr_hyper_int_rep_s_t {
+    ndr_ulong_int low; 
+    ndr_long_int high;
+};
+
+struct ndr_uhyper_int_rep_s_t {
+    ndr_ulong_int low; 
+    ndr_ulong_int high;
+};
+
+#ifdef __GNUC__
+typedef long long int		ndr_hyper_int;
+typedef unsigned long long int	ndr_uhyper_int;
+#else
+typedef struct ndr_hyper_int_rep_s_t ndr_hyper_int;
+typedef struct ndr_uhyper_int_rep_s_t ndr_uhyper_int;
+#endif /* __GNUC__ */
+
+typedef float 		        ndr_short_float;
+typedef double 			ndr_long_float;
+
+
+#endif /* _NDR_TYPES_H */
