Index: likewise-open-5.4.0.39949/build/lib/mkcomp-helper.sh
===================================================================
--- likewise-open-5.4.0.39949.orig/build/lib/mkcomp-helper.sh	2010-01-21 10:50:54.000000000 +0000
+++ likewise-open-5.4.0.39949/build/lib/mkcomp-helper.sh	2010-01-21 10:51:25.000000000 +0000
@@ -4,7 +4,8 @@
 {
     case `uname -s` in
 	Linux)
-	    cat /proc/cpuinfo | grep '^processor' | wc -l
+	    echo 1
+	    #cat /proc/cpuinfo | grep '^processor' | wc -l
 	    ;;
 	*BSD)
 	    sysctl -n hw.ncpu
