diff -ur likewise-open-6.0.0.53010/domainjoin/libdomainjoin/src/djnsswitch.c likewise-open-6.0.0.53010.new//domainjoin/libdomainjoin/src/djnsswitch.c
--- likewise-open-6.0.0.53010/domainjoin/libdomainjoin/src/djnsswitch.c	2010-11-23 03:55:16.000000000 +0800
+++ likewise-open-6.0.0.53010.new//domainjoin/libdomainjoin/src/djnsswitch.c	2011-11-22 16:49:43.470334957 +0800
@@ -1055,14 +1055,14 @@
 PREFIXDIR "/lib/*.so*            mr,\n"
 PREFIXDIR "/lib64/*.so*          mr,\n"
 "/tmp/.lwidentity/pipe              rw,\n"
-LOCALSTATEDIR "/lib/likewise/.lsassd  rw,\n"
+LOCALSTATEDIR "/.lsassd  rw,\n"
 LOCALSTATEDIR "/tmp/.lsaclient_*              rw,\n";
     else
         addString =
 PREFIXDIR "/lib/*.so*            r,\n"
 PREFIXDIR "/lib64/*.so*          r,\n"
 "/tmp/.lwidentity/pipe              rw,\n"
-LOCALSTATEDIR "/lib/likewise/.lsassd  rw,\n"
+LOCALSTATEDIR "/.lsassd  rw,\n"
 LOCALSTATEDIR "/tmp/.lsaclient_*              rw,\n";
 
 
@@ -1070,7 +1070,7 @@
     {
         LW_CLEANUP_CTERR(exc, CTCopyFileWithOriginalPerms(finalName, tempName));
         LW_CLEANUP_CTERR(exc, CTOpenFile(tempName, "a", &file));
-        LW_CLEANUP_CTERR(exc, CTFilePrintf(file, "# likewise\n%s# end likewise\n",
+	LW_CLEANUP_CTERR(exc, CTFilePrintf(file, "# -start- likewise\n%s# -end- likewise\n",
                     addString));
 
         CTSafeCloseFile(&file);
@@ -1079,7 +1079,7 @@
     }
     else
     {
-        LW_CLEANUP_CTERR(exc, CTRunSedOnFile(finalName, finalName, FALSE, "/^[ \t]*#[ \t]*likewise[ \t]*$/,/^[ \t]*#[ \t]*end likewise[ \t]*$/d"));
+        LW_CLEANUP_CTERR(exc, CTRunSedOnFile(finalName, finalName, FALSE, "/^[ \t]*#[ \t]*-start- likewise[ \t]*$/,/^[ \t]*#[ \t]*-end- likewise[ \t]*$/d"));
         LW_CLEANUP_CTERR(exc, CTRunSedOnFile(finalName, finalName, FALSE, "/^[ \t]*#[ \t]*centeris[ \t]*$/,/^[ \t]*#[ \t]*end centeris[ \t]*$/d"));
     }
 
