diff -ur likewise-open-6.0.0.53010/domainjoin/domainjoin-cli/src/main.c likewise-open-6.0.0.53010.new//domainjoin/domainjoin-cli/src/main.c
--- likewise-open-6.0.0.53010/domainjoin/domainjoin-cli/src/main.c	2010-10-20 04:54:22.000000000 +0800
+++ likewise-open-6.0.0.53010.new//domainjoin/domainjoin-cli/src/main.c	2011-11-22 16:44:34.043743863 +0800
@@ -492,6 +492,7 @@
 
     LW_TRY(exc, DJRunJoinProcess(&options, &LW_EXC));
     fprintf(stdout, "SUCCESS\n");
+    fprintf(stdout, "You should reboot this system before attempting GUI logins as a domain user.\n");
 
 cleanup:
     DJFreeJoinProcessOptions(&options);
