diff -Nurb likewise-open-6.1.0.48254.orig/srvsvc/client/defs.h likewise-open-6.1.0.48254/srvsvc/client/defs.h
--- likewise-open-6.1.0.48254.orig/srvsvc/client/defs.h	2010-07-29 17:27:58.000000000 -0700
+++ likewise-open-6.1.0.48254/srvsvc/client/defs.h	2010-07-30 10:39:39.000000000 -0700
@@ -46,7 +46,7 @@
 
 #define SRVSVC_DEFAULT_PROT_SEQ   "ncacn_np"
 #define SRVSVC_DEFAULT_ENDPOINT   "\\pipe\\srvsvc"
-#define SRVSVC_LOCAL_ENDPOINT     "/var/lib/likewise/rpc/srvsvc"
+#define SRVSVC_LOCAL_ENDPOINT     "/var/lib/likewise-open/rpc/srvsvc"
 
 #define TRY                      DCETHREAD_TRY
 #define CATCH_ALL(pDceException) DCETHREAD_CATCH_ALL(pDceException)
