--- likewise-open-before/lwbase/include/lw/types.h	2010-10-19 13:54:28.000000000 -0700
+++ likewise-open-after/lwbase/include/lw/types.h	2010-11-01 12:39:13.000000000 -0700
@@ -228,7 +228,7 @@
 #define LW_PTR_ADD(Pointer, Offset) LwRtlOffsetToPointer(Pointer, Offset)
 
 #define LW_FIELD_OFFSET(Type, Field) \
-    ((size_t)(&(((Type*)(0))->Field)))
+    offsetof(Type, Field)
 
 #define LW_FIELD_SIZE(Type, Field) \
     (sizeof(((Type*)(0))->Field))
