diff -ur likewise-open-6.0.0.53010/domainjoin/libdomainjoin/src/djdistroinfo.c likewise-open-6.0.0.53010.new//domainjoin/libdomainjoin/src/djdistroinfo.c
--- likewise-open-6.0.0.53010/domainjoin/libdomainjoin/src/djdistroinfo.c	2010-11-23 03:55:16.000000000 +0800
+++ likewise-open-6.0.0.53010.new//domainjoin/libdomainjoin/src/djdistroinfo.c	2011-11-22 16:55:10.640615867 +0800
@@ -654,7 +654,6 @@
     PSTR line = NULL; 
     BOOLEAN isEndOfFile = FALSE;
     DWORD ceError = ERROR_SUCCESS;
-    BOOLEAN bIsEnterprise = FALSE;
 
     PSTR _product = NULL;
     PSTR _version = NULL;
@@ -668,15 +667,7 @@
 #ifdef MINIMAL_JOIN
     GCE(ceError = CTOpenFile(LOCALSTATEDIR "/VERSION", "r", &versionFile));
 #else
-    ceError = CTOpenFile(PREFIXDIR "/data/ENTERPRISE_VERSION", "r", &versionFile);
-    if (ceError == 0)
-    {
-        bIsEnterprise = TRUE;
-    }
-    else
-    {
-        GCE(ceError = CTOpenFile(PREFIXDIR "/data/VERSION", "r", &versionFile));
-    }
+    GCE(ceError = CTOpenFile(PREFIXDIR "/share/likewise-open/VERSION", "r", &versionFile));
 #endif
 
     while (TRUE)
@@ -702,14 +693,8 @@
         }
     }
 
-    if (bIsEnterprise)
-    {
-        GCE(ceError = CTStrdup("Enterprise", &_product));
-    }
-    else
-    {
-        GCE(ceError = CTStrdup("Open", &_product));
-    }
+    GCE(ceError = CTStrdup("Open", &_product));
+
     if (_version == NULL)
     {
         GCE(ceError = CTStrdup("unknown", &_version));
diff -ur likewise-open-6.0.0.53010/lsass/common/utils/lsastatus.c likewise-open-6.0.0.53010.new//lsass/common/utils/lsastatus.c
--- likewise-open-6.0.0.53010/lsass/common/utils/lsastatus.c	2010-10-20 04:54:27.000000000 +0800
+++ likewise-open-6.0.0.53010.new//lsass/common/utils/lsastatus.c	2011-11-22 16:55:42.037948168 +0800
@@ -99,11 +99,7 @@
 #ifdef MINIMAL_LSASS
     versionFile = open(LOCALSTATEDIR "/VERSION", O_RDONLY, 0);
 #else
-    versionFile = open(PREFIXDIR "/data/ENTERPRISE_VERSION", O_RDONLY, 0);
-    if (versionFile < 0 && errno == ENOENT)
-    {
-        versionFile = open(PREFIXDIR "/data/VERSION", O_RDONLY, 0);
-    }
+    versionFile = open(PREFIXDIR "/share/likewise-open/VERSION", O_RDONLY, 0);
 #endif
     if (versionFile < 0)
     {
diff -ur likewise-open-6.0.0.53010/lsass/tools/lw-lsa/lw_domain_defs.h likewise-open-6.0.0.53010.new//lsass/tools/lw-lsa/lw_domain_defs.h
--- likewise-open-6.0.0.53010/lsass/tools/lw-lsa/lw_domain_defs.h	2010-10-20 04:54:28.000000000 +0800
+++ likewise-open-6.0.0.53010.new//lsass/tools/lw-lsa/lw_domain_defs.h	2011-11-22 16:56:08.411840492 +0800
@@ -45,7 +45,7 @@
 #ifndef __DEFS_H__
 #define __DEFS_H__
 
-#define LW_DOMAIN_VERSION_FILE_PATH PREFIXDIR "/data/VERSION"
+#define LW_DOMAIN_VERSION_FILE_PATH PREFIXDIR "/share/likewise-open/VERSION"
 
 typedef enum
 {
