diff -up openchange-0.9-COCHRANE/libmapi/IMSProvider.c.close-connections openchange-0.9-COCHRANE/libmapi/IMSProvider.c
--- openchange-0.9-COCHRANE/libmapi/IMSProvider.c.close-connections	2011-06-13 10:51:04.208854188 +0200
+++ openchange-0.9-COCHRANE/libmapi/IMSProvider.c	2011-06-13 10:52:32.075728473 +0200
@@ -306,6 +306,7 @@ _PUBLIC_ enum MAPISTATUS Logoff(mapi_obj
 			found = true;
 			mapi_object_release(obj_store);
 			DLIST_REMOVE(global_mapi_ctx->session, el);
+			MAPIFreeBuffer(session);
 			break;
 		}
 	}
