diff -u opt/libreoffice/share/xdg//base.desktop opt/libreoffice/share/xdg.new/base.desktop
--- opt/libreoffice/share/xdg//base.desktop	2011-02-16 12:58:41.000000000 +0800
+++ opt/libreoffice/share/xdg.new/base.desktop	2011-03-11 10:32:29.747996002 +0800
@@ -8,6 +8,8 @@
 MimeType=application/vnd.oasis.opendocument.database;application/vnd.sun.xml.base;
 Name=LibreOffice 3.3 Base
 Name[pt_BR]=BrOffice 3.3 Base
+Name[zh_CN]=LibreOffice 3.3 数据库
+Name[zh_TW]=LibreOffice 3.3 資料庫
 GenericName=Database Development
 GenericName[en]=Database Development
 GenericName[te]=డాటాబేస్ డెవలప్‌మెంట్
diff -u opt/libreoffice/share/xdg//calc.desktop opt/libreoffice/share/xdg.new/calc.desktop
--- opt/libreoffice/share/xdg//calc.desktop	2011-02-16 12:58:41.000000000 +0800
+++ opt/libreoffice/share/xdg.new/calc.desktop	2011-03-11 10:33:27.739996010 +0800
@@ -7,6 +7,9 @@
 Exec=libreoffice -calc %U
 MimeType=application/vnd.oasis.opendocument.spreadsheet;application/vnd.oasis.opendocument.spreadsheet-template;application/vnd.sun.xml.calc;application/vnd.sun.xml.calc.template;application/vnd.stardivision.calc;application/vnd.stardivision.chart;application/msexcel;application/vnd.ms-excel;application/vnd.openxmlformats-officedocument.spreadsheetml.sheet;application/vnd.ms-excel.sheet.macroenabled.12;application/vnd.openxmlformats-officedocument.spreadsheetml.template;application/vnd.ms-excel.template.macroenabled.12;application/vnd.ms-excel.sheet.binary.macroenabled.12;text/csv;application/x-dbf;
 Name=LibreOffice 3.3 Calc
+Name[zh_CN]=LibreOffice 3.3 电子表格
+Name[zh_TW]=LibreOffice 3.3 試算表
+Name=LibreOffice 3.3 Calc
 Name[pt_BR]=BrOffice 3.3 Calc
 GenericName=Spreadsheet
 GenericName[en]=Spreadsheet
diff -u opt/libreoffice/share/xdg//draw.desktop opt/libreoffice/share/xdg.new/draw.desktop
--- opt/libreoffice/share/xdg//draw.desktop	2011-02-16 12:58:41.000000000 +0800
+++ opt/libreoffice/share/xdg.new/draw.desktop	2011-03-11 10:34:47.398996001 +0800
@@ -7,6 +7,9 @@
 Exec=libreoffice -draw %U
 MimeType=application/vnd.oasis.opendocument.graphics;application/vnd.oasis.opendocument.graphics-template;application/vnd.sun.xml.draw;application/vnd.sun.xml.draw.template;application/vnd.stardivision.draw;
 Name=LibreOffice 3.3 Draw
+Name[zh_CN]=LibreOffice 3.3 绘图
+Name[zh_TW]=LibreOffice 3.3 繪圖程式
+Name=LibreOffice 3.3 Draw
 Name[pt_BR]=BrOffice 3.3 Draw
 GenericName=Drawing Program
 GenericName[en]=Drawing Program
diff -u opt/libreoffice/share/xdg//impress.desktop opt/libreoffice/share/xdg.new/impress.desktop
--- opt/libreoffice/share/xdg//impress.desktop	2011-02-16 12:58:41.000000000 +0800
+++ opt/libreoffice/share/xdg.new/impress.desktop	2011-03-11 10:35:50.336996002 +0800
@@ -7,6 +7,9 @@
 Exec=libreoffice -impress %U
 MimeType=application/vnd.oasis.opendocument.presentation;application/vnd.oasis.opendocument.presentation-template;application/vnd.sun.xml.impress;application/vnd.sun.xml.impress.template;application/vnd.stardivision.impress;application/mspowerpoint;application/vnd.ms-powerpoint;application/vnd.openxmlformats-officedocument.presentationml.presentation;application/vnd.ms-powerpoint.presentation.macroenabled.12;application/vnd.openxmlformats-officedocument.presentationml.template;application/vnd.ms-powerpoint.template.macroenabled.12;
 Name=LibreOffice 3.3 Impress
+Name[zh_CN]=LibreOffice 3.3 幻灯片
+Name[zh_TW]=LibreOffice 3.3 簡報
+Name=LibreOffice 3.3 Impress
 Name[pt_BR]=BrOffice 3.3 Impress
 GenericName=Presentation
 GenericName[en]=Presentation
diff -u opt/libreoffice/share/xdg//math.desktop opt/libreoffice/share/xdg.new/math.desktop
--- opt/libreoffice/share/xdg//math.desktop	2011-02-16 12:58:41.000000000 +0800
+++ opt/libreoffice/share/xdg.new/math.desktop	2011-03-11 10:36:59.589996001 +0800
@@ -7,6 +7,8 @@
 Exec=libreoffice -math %U
 MimeType=application/vnd.oasis.opendocument.formula;application/vnd.sun.xml.math;application/vnd.stardivision.math;
 Name=LibreOffice 3.3 Math
+Name[zh_CN]=LibreOffice 3.3 公式编辑器
+Name[zh_TW]=LibreOffice 3.3 公式編輯器
 Name[pt_BR]=BrOffice 3.3 Math
 GenericName=Formula Editor
 GenericName[en]=Formula Editor
diff -u opt/libreoffice/share/xdg//qstart.desktop opt/libreoffice/share/xdg.new/qstart.desktop
--- opt/libreoffice/share/xdg//qstart.desktop	2011-02-16 12:58:41.000000000 +0800
+++ opt/libreoffice/share/xdg.new/qstart.desktop	2011-03-11 10:37:39.723996002 +0800
@@ -6,5 +6,6 @@
 Exec=libreoffice -quickstart -nologo -nodefault
 NoDisplay=true
 Name=LibreOffice 3.3 Quickstarter
+Name[zh_CN]=LibreOffice 3.3 快速启动
 Name[pt_BR]=BrOffice 3.3 Quickstarter
 Comment=Hook for quickstarter startup
diff -u opt/libreoffice/share/xdg//startcenter.desktop opt/libreoffice/share/xdg.new/startcenter.desktop
--- opt/libreoffice/share/xdg//startcenter.desktop	2011-02-16 12:58:41.000000000 +0800
+++ opt/libreoffice/share/xdg.new/startcenter.desktop	2011-03-11 10:39:07.886995997 +0800
@@ -7,6 +7,9 @@
 Exec=libreoffice %U
 MimeType=application/vnd.openofficeorg.extension;
 Name=LibreOffice 3.3 
+Name[zh_CN]=LibreOffice 3.3 办公套件
+Name[zh_TW]=LibreOffice 3.3 辦公套件
+Name=LibreOffice 3.3 
 Name[pt_BR]=BrOffice 3.3
 GenericName=Office 
 GenericName[en]=Office
diff -u opt/libreoffice/share/xdg//writer.desktop opt/libreoffice/share/xdg.new/writer.desktop
--- opt/libreoffice/share/xdg//writer.desktop	2011-02-16 12:58:41.000000000 +0800
+++ opt/libreoffice/share/xdg.new/writer.desktop	2011-03-11 10:39:48.004996002 +0800
@@ -7,6 +7,9 @@
 Exec=libreoffice -writer %U
 MimeType=application/vnd.oasis.opendocument.text;application/vnd.oasis.opendocument.text-template;application/vnd.oasis.opendocument.text-web;application/vnd.oasis.opendocument.text-master;application/vnd.sun.xml.writer;application/vnd.sun.xml.writer.template;application/vnd.sun.xml.writer.global;application/vnd.stardivision.writer;application/msword;application/vnd.ms-word;application/x-doc;application/rtf;text/rtf;application/vnd.wordperfect;application/wordperfect;application/vnd.openxmlformats-officedocument.wordprocessingml.document;application/vnd.ms-word.document.macroenabled.12;application/vnd.openxmlformats-officedocument.wordprocessingml.template;application/vnd.ms-word.template.macroenabled.12;
 Name=LibreOffice 3.3 Writer
+Name[zh_CN]=LibreOffice 3.3 文档
+Name[zh_TW]=LibreOffice 3.3 文書處理器
+Name=LibreOffice 3.3 Writer
 Name[pt_BR]=BrOffice 3.3 Writer
 GenericName=Word Processor
 GenericName[en]=Word Processor
