diff -ur vigra1.5.0/config/Makefile.include.in vigra-1.5.0.new/config/Makefile.include.in
--- vigra1.5.0/config/Makefile.include.in	2006-12-08 04:52:06.000000000 +0800
+++ vigra-1.5.0.new/config/Makefile.include.in	2011-07-28 11:32:11.818099042 +0800
@@ -5,6 +5,8 @@
 VIGRA_VERSION = @MAJOR_VERSION@.@MINOR_VERSION@.@MICRO_VERSION@
 SHAREDLIB_VERSION = @SHAREDLIB_VERSION@
 
+DESTDIR=""
+
 vigra_builddir = @vigra_builddir@
 vigra_srcdir = @vigra_srcdir@
 vigra_testdir = $(vigra_srcdir)/test
@@ -12,10 +14,10 @@
 exec_prefix = @exec_prefix@
 pymoddir = @PYTHON_INSTALLDIR@
 python_sitedir = @python_sitedir@
-includedir = @includedir@
-libdir = @libdir@
-bindir = @bindir@
-docdir = @docdir@
+includedir = $(DESTDIR)/@includedir@
+libdir = $(DESTDIR)/@libdir@
+bindir = $(DESTDIR)/@bindir@
+docdir = $(DESTDIR)/@docdir@
 
 LIBS = @LIBS@
 PNG_LIB = @PNG_LIB@
