diff -u -r bcmwl-5.100.82.38.orig/Makefile bcmwl-5.100.82.38/Makefile
--- bcmwl-5.100.82.38.orig/Makefile	2010-12-15 10:01:09.000000000 +0800
+++ bcmwl-5.100.82.38/Makefile	2011-04-22 10:06:46.416019240 +0800
@@ -13,6 +13,14 @@
 #
 # $Id: Makefile_kbuild_portsrc,v 1.6.54.1 2010/09/27 22:18:19 Exp $
 
+ifeq ($(shell dpkg --print-architecture),amd64)
+BINARCH = x86_64
+else
+BINARCH = i386
+endif
+
+DESTDIR=
+
 obj-m              += wl.o
 
 wl-objs            := 
@@ -25,7 +33,8 @@
 EXTRA_CFLAGS       += -I$(src)/src/wl/sys -I$(src)/src/wl/phy
 #EXTRA_CFLAGS       += -DBCMDBG_ASSERT
 
-EXTRA_LDFLAGS      := $(src)/lib/wlc_hybrid.o_shipped
+#EXTRA_LDFLAGS      := $(src)/lib/wlc_hybrid.o_shipped
+EXTRA_LDFLAGS      := $(src)/lib/wlc_hybrid.o_shipped_$(BINARCH)
 
 all:
 	KBUILD_NOPEDANTIC=1 make -C /lib/modules/`uname -r`/build M=`pwd`
@@ -34,4 +43,4 @@
 	KBUILD_NOPEDANTIC=1 make -C /lib/modules/`uname -r`/build M=`pwd` clean
 
 install:
-	install -D -m 755 wl.ko /lib/modules/`uname -r`/kernel/drivers/net/wireless/wl.ko
+	install -D -m 755 wl.ko $(DESTDIR)/lib/modules/`uname -r`/kernel/drivers/net/wireless/wl.ko
diff -u -r bcmwl-5.100.82.38.orig/src/wl/sys/wl_linux.c bcmwl-5.100.82.38/src/wl/sys/wl_linux.c
--- bcmwl-5.100.82.38.orig/src/wl/sys/wl_linux.c	2010-12-15 10:01:09.000000000 +0800
+++ bcmwl-5.100.82.38/src/wl/sys/wl_linux.c	2011-04-22 10:02:55.356022303 +0800
@@ -99,6 +99,10 @@
 
 #endif 
 
+#ifndef init_MUTEX
+#define init_MUTEX(sem) sema_init(sem, 1)
+#endif
+
 static int wl_linux_watchdog(void *ctx);
 static
 int wl_found = 0;
@@ -172,6 +176,8 @@
 static void wl_free_if(wl_info_t *wl, wl_if_t *wlif);
 static void wl_get_driver_info(struct net_device *dev, struct ethtool_drvinfo *info);
 
+MODULE_LICENSE("MIXED/Proprietary");
+
 #if defined(WL_CONFIG_RFKILL)
 #include <linux/rfkill.h>
 static int wl_init_rfkill(wl_info_t *wl);
