diff -ur xulrunner-14.0.1/build/unix/mozilla.in xulrunner-14.0.1.new//build/unix/mozilla.in
--- xulrunner-14.0.1/build/unix/mozilla.in	2012-07-13 21:42:48.000000000 +0000
+++ xulrunner-14.0.1.new//build/unix/mozilla.in	2012-08-13 10:43:02.809623817 +0000
@@ -51,6 +51,7 @@
 #set -x
 
 moz_libdir=%MOZAPPDIR%
+moz_sdkbindir=%SDK_BIN_DIR%
 
 # Use run-mozilla.sh in the current dir if it exists
 # If not, then start resolving symlinks until we find run-mozilla.sh
@@ -88,9 +89,9 @@
 fi
 if [ $found = 0 ]; then
   # Check default compile-time libdir
-  if [ -x "$moz_libdir/run-mozilla.sh" ]; then
+  if [ -x "$moz_sdkbindir/run-mozilla.sh" ]; then
     dist_bin="$moz_libdir"
-    run_moz="$moz_libdir/run-mozilla.sh"
+    run_moz="$moz_sdkbindir/run-mozilla.sh"
   else
     echo "Cannot find %MOZ_APP_DISPLAYNAME% runtime directory. Exiting."
     exit 1
@@ -134,7 +135,7 @@
 
 if [ $debugging = 1 ]
 then
-  echo $dist_bin/run-mozilla.sh $script_args $dist_bin/$MOZILLA_BIN "$@"
+  echo $run_moz $script_args $dist_bin/$MOZILLA_BIN "$@"
 fi
-exec "$dist_bin/run-mozilla.sh" $script_args "$dist_bin/$MOZILLA_BIN" "$@"
+exec "$run_moz" $script_args "$dist_bin/$MOZILLA_BIN" "$@"
 # EOF.
diff -ur xulrunner-14.0.1/xulrunner/app/Makefile.in xulrunner-14.0.1.new//xulrunner/app/Makefile.in
--- xulrunner-14.0.1/xulrunner/app/Makefile.in	2012-07-13 21:43:18.000000000 +0000
+++ xulrunner-14.0.1.new//xulrunner/app/Makefile.in	2012-08-13 10:45:20.022949725 +0000
@@ -159,7 +159,7 @@
 ifeq (unix, $(MOZ_FS_LAYOUT))
 
 xulrunner:: $(topsrcdir)/build/unix/mozilla.in $(GLOBAL_DEPS)
-	cat $< | sed -e "s|%MOZAPPDIR%|$(installdir)|" \
+	cat $< | sed -e "s|%MOZAPPDIR%|$(installdir)|" -e "s|%SDK_BIN_DIR%|$(sdkdir)/sdk/bin|" \
                 -e "s|%MOZ_USER_DIR%|.mozilla/xulrunner|" \
                 -e "s|%MOZ_APP_DISPLAYNAME%|$(MOZ_APP_DISPLAYNAME)|" > $@
 	chmod +x $@
