diff -u rsyslog-5.7.10/rsyslog.service.in rsyslog-5.7.10.new//rsyslog.service.in
--- rsyslog-5.7.10/rsyslog.service.in	2011-03-28 23:23:18.000000000 +0800
+++ rsyslog-5.7.10.new//rsyslog.service.in	2011-03-31 09:09:13.688421003 +0800
@@ -3,7 +3,7 @@
 
 [Service]
 ExecStartPre=/bin/systemctl stop systemd-kmsg-syslogd.service
-ExecStart=@sbindir@/rsyslogd -n -c5
+ExecStart=@sbindir@/rsyslogd -c5
 Sockets=syslog.socket
 
 [Install]
