diff -u opt/libreoffice/share/xdg//qstart.desktop opt/libreoffice/share/xdg.new/qstart.desktop
--- opt/libreoffice/share/xdg//qstart.desktop	2011-02-16 12:58:41.000000000 +0800
+++ opt/libreoffice/share/xdg.new/qstart.desktop	2011-03-11 10:37:39.723996002 +0800
@@ -6,5 +6,6 @@
 Exec=libreoffice -quickstart -nologo -nodefault
 NoDisplay=true
 Name=LibreOffice 3.3 Quickstarter
+Name[zh_CN]=LibreOffice 3.3 快速启动
 Name[pt_BR]=BrOffice 3.3 Quickstarter
 Comment=Hook for quickstarter startup
diff -u opt/libreoffice/share/xdg//startcenter.desktop opt/libreoffice/share/xdg.new/startcenter.desktop
--- opt/libreoffice/share/xdg//startcenter.desktop	2011-02-16 12:58:41.000000000 +0800
+++ opt/libreoffice/share/xdg.new/startcenter.desktop	2011-03-11 10:39:07.886995997 +0800
@@ -7,6 +7,10 @@
 Exec=libreoffice %U
 MimeType=application/vnd.openofficeorg.extension;
 Name=LibreOffice 3.3 
+Name[zh_CN]=LibreOffice 3.3 办公套件
+Name[zh_TW]=LibreOffice 3.3 辦公套件
+Name=LibreOffice 3.3 
+NoDisplay=true
 Name[pt_BR]=BrOffice 3.3
 GenericName=Office 
 GenericName[en]=Office
