diff -ur ibus-1.4.0/ui/gtk/candidatepanel.py ibus-1.4.0.new/ui/gtk/candidatepanel.py
--- ibus-1.4.0/ui/gtk/candidatepanel.py	2011-08-11 08:49:43.000000000 +0800
+++ ibus-1.4.0.new/ui/gtk/candidatepanel.py	2011-12-22 14:27:06.328499128 +0800
@@ -84,10 +84,12 @@
         for i in range(0, 16):
             label1 = Label("%c." % ("1234567890abcdef"[i]))
             label1.set_alignment(0.0, 0.5)
+	    label1.modify_fg(gtk.STATE_NORMAL,gtk.gdk.Color("#013BC5"))
             label1.show()
 
             label2 = Label()
             label2.set_alignment(0.0, 0.5)
+	    label2.modify_fg(gtk.STATE_NORMAL,gtk.gdk.Color("#69A0EC"))
             label1.show()
 
             if self.__orientation == ibus.ORIENTATION_VERTICAL:
@@ -95,9 +97,11 @@
                 label2.set_property("xpad", 8)
                 ebox1 = EventBox()
                 ebox1.set_no_show_all(True)
+		ebox1.modify_fg(gtk.STATE_NORMAL,gtk.gdk.Color("#FFFFFF"))
                 ebox1.add(label1)
                 ebox2 = EventBox()
                 ebox2.set_no_show_all(True)
+		ebox2.modify_fg(gtk.STATE_NORMAL,gtk.gdk.Color("#FFFFFF"))
                 ebox2.add(label2)
                 self.__vbox1.pack_start(ebox1, False, False, 2)
                 self.__vbox2.pack_start(ebox2, False, False, 2)
@@ -109,6 +113,7 @@
                 hbox.pack_start(label2, False, False, 1)
                 ebox = EventBox()
                 ebox.set_no_show_all(True)
+		ebox.modify_fg(gtk.STATE_NORMAL,gtk.gdk.Color("#FFFFFF"))
                 ebox.add(hbox)
                 self.pack_start(ebox, False, False, 4)
                 self.__candidates.append((ebox,))
@@ -140,11 +145,8 @@
             if i == focus_candidate and show_cursor:
                 if attrs == None:
                     attrs = pango.AttrList()
-                color = self.__labels[i][1].style.base[gtk.STATE_SELECTED]
                 end_index = len(text.encode("utf8"))
-                attr = pango.AttrBackground(color.red, color.green, color.blue, 0, end_index)
-                attrs.change(attr)
-                color = self.__labels[i][1].style.text[gtk.STATE_SELECTED]
+		color = gtk.gdk.Color("#FF55FF")
                 attr = pango.AttrForeground(color.red, color.green, color.blue, 0, end_index)
                 attrs.insert(attr)
 
@@ -240,6 +242,7 @@
 
         # create aux label
         self.__aux_label = Label(self.__aux_string)
+	self.__aux_label.modify_fg(gtk.STATE_NORMAL,gtk.gdk.Color("#EB50A0"))
         self.__aux_label.set_attributes(self.__aux_attrs)
         self.__aux_label.set_alignment(0.0, 0.5)
         self.__aux_label.set_padding(8, 0)
@@ -519,6 +522,7 @@
     table = ibus.LookupTable()
     table.append_candidate(ibus.Text("AAA"))
     table.append_candidate(ibus.Text("BBB"))
+    table.append_candidate(ibus.Text("CCC"))
     cp = CandidatePanel()
     cp.show_all()
     cp.update_lookup_table(table, True)
