--- include/os/rt_linux.h.orig	2010-12-08 19:19:02.000000000 +0100
+++ include/os/rt_linux.h	2010-12-17 20:20:19.000000000 +0100
@@ -111,10 +111,10 @@
 
 #ifdef CONFIG_STA_SUPPORT
 #ifdef RTMP_MAC_PCI
-#define STA_PROFILE_PATH			"/etc/Wireless/RT2860STA/RT2860STA.dat"
+#define STA_PROFILE_PATH			"/etc/Wireless/RT5390STA/RT5390STA.dat"
 #define STA_DRIVER_VERSION			"2.4.0.4"
 #ifdef MULTIPLE_CARD_SUPPORT
-#define CARD_INFO_PATH			"/etc/Wireless/RT2860STA/RT2860STACard.dat"
+#define CARD_INFO_PATH			"/etc/Wireless/RT5390STA/RT5390STACard.dat"
 #endif // MULTIPLE_CARD_SUPPORT //
 #endif // RTMP_MAC_PCI //
 
@@ -1208,7 +1208,7 @@
 #define ATEDBGPRINT DBGPRINT
 #ifdef RTMP_MAC_PCI
 #ifdef CONFIG_STA_SUPPORT
-#define EEPROM_BIN_FILE_NAME  "/etc/Wireless/RT2860STA/e2p.bin"
+#define EEPROM_BIN_FILE_NAME  "/etc/Wireless/RT5390STA/e2p.bin"
 #endif // CONFIG_STA_SUPPORT //
 #endif // RTMP_MAC_PCI //
 
--- os/linux/pci_main_dev.c.orig	2010-12-08 18:13:46.000000000 +0100
+++ os/linux/pci_main_dev.c	2010-12-17 20:23:29.000000000 +0100
@@ -74,13 +74,12 @@
 #endif
 #endif // CONFIG_STA_SUPPORT //
 
-
 //
 // Our PCI driver structure
 //
 static struct pci_driver rt2860_driver =
 {
-    name:       "rt2860",
+    name:       "rt5390",
     id_table:   rt2860_pci_tbl,
     probe:      rt2860_probe,
 #if LINUX_VERSION_CODE >= 0x20412
@@ -303,9 +302,9 @@
 	}
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,0)
-	print_name = pci_dev ? pci_name(pci_dev) : "rt2860";
+	print_name = pci_dev ? pci_name(pci_dev) : "rt5390";
 #else
-	print_name = pci_dev ? pci_dev->slot_name : "rt2860";
+	print_name = pci_dev ? pci_dev->slot_name : "rt5390";
 #endif // LINUX_VERSION_CODE //
 
 	if ((rv = pci_request_regions(pci_dev, print_name)) != 0)
--- os/linux/rt_main_dev.c.orig	2010-12-17 20:24:55.000000000 +0100
+++ os/linux/rt_main_dev.c	2010-12-17 20:25:47.000000000 +0100
@@ -46,7 +46,7 @@
 MODULE_PARM_DESC (mac, "rt28xx: wireless mac addr");
 
 MODULE_LICENSE("GPL");
-
+MODULE_DESCRIPTION("RT5390 Wireless Lan Linux Driver");
 
 #ifdef OS_ABL_SUPPORT
 
