diff -ur bash-4.2-ylmf2/config-top.h bash-4.2-ylmf2.new//config-top.h
--- bash-4.2-ylmf2/config-top.h	2009-12-23 04:29:39.000000000 +0800
+++ bash-4.2-ylmf2.new//config-top.h	2011-06-28 09:39:49.599962278 +0800
@@ -54,14 +54,14 @@
 /* The default value of the PATH variable. */
 #ifndef DEFAULT_PATH_VALUE
 #define DEFAULT_PATH_VALUE \
-  "/usr/gnu/bin:/usr/local/bin:/bin:/usr/bin:."
+  "/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/sbin:/usr/local/bin"
 #endif
 
 /* The value for PATH when invoking `command -p'.  This is only used when
    the Posix.2 confstr () function, or CS_PATH define are not present. */
 #ifndef STANDARD_UTILS_PATH
 #define STANDARD_UTILS_PATH \
-  "/bin:/usr/bin:/sbin:/usr/sbin:/etc:/usr/etc"
+  "/bin:/usr/bin:/sbin:/usr/sbin"	
 #endif
 
 /* Default primary and secondary prompt strings. */
