diff -Naur DeviceKit-power-014.old//configure.ac DeviceKit-power-014/configure.ac
--- DeviceKit-power-014.old//configure.ac	2010-01-08 10:49:06.000000000 +0000
+++ DeviceKit-power-014/configure.ac	2010-05-03 15:34:30.000000000 +0100
@@ -52,6 +52,8 @@
   fi
 fi
 AM_CONDITIONAL(MAN_PAGES_ENABLED, test x$enable_man_pages = xyes)
+AM_CONDITIONAL(GTK_DOC_BUILD_HTML, test x$enable_html_pages = xyes)
+AM_CONDITIONAL(GTK_DOC_BUILD_PDF, test x$enable_pdf_pages = xyes)
 
 GTK_DOC_CHECK([1.3])
 
