diff -ur rtl8712_8188_8191_8192SU_usb_linux_v2.6.6.0.20110401/Makefile rtl8712_8188_8191_8192SU_usb_linux_v2.6.6.0.20110401.new//Makefile
--- rtl8712_8188_8191_8192SU_usb_linux_v2.6.6.0.20110401/Makefile	2011-04-01 15:25:21.000000000 +0800
+++ rtl8712_8188_8191_8192SU_usb_linux_v2.6.6.0.20110401.new//Makefile	2011-11-03 13:53:52.496456977 +0800
@@ -217,7 +217,8 @@
 	$(MAKE) ARCH=$(ARCH) CROSS_COMPILE=$(CROSS_COMPILE) -C $(KSRC) M=$(shell pwd)  modules
 
 install:
-	install -p -m 644 $(MODULE_NAME).ko  $(MODDESTDIR)
+	mkdir -p $(DESTDIR)/$(MODDESTDIR)
+	install -p -m 644 $(MODULE_NAME).ko $(DESTDIR)/$(MODDESTDIR)
 	/sbin/depmod -a ${KVER}
 
 uninstall:
