--- library/gnome-keyring.c	2011-03-01 17:11:44.534116993 +0800
+++ library/gnome-keyring.c.new	2011-03-01 17:11:59.920117001 +0800
@@ -2714,7 +2714,7 @@
 							    SERVICE_INTERFACE, "CreateCollection");
 			dbus_message_iter_init_append (req, &iter);
 			/* TRANSLATORS: This is the name of an automatically created default keyring. */
-			create_keyring_encode_properties (&iter, _("Default"));
+			create_keyring_encode_properties (&iter, _("default"));
 			gkr_operation_push (op, item_create_1_collection_reply, GKR_CALLBACK_OP_MSG, args, NULL);
 			gkr_operation_request (op, req);
 			dbus_message_unref (req);
