diff -ur gnome-system-tools-2.30.2-ylmf2/src/network/network.desktop.in.in gnome-system-tools-2.30.2-ylmf2.new/src/network/network.desktop.in.in
--- gnome-system-tools-2.30.2-ylmf2/src/network/network.desktop.in.in	2009-08-26 05:16:58.000000000 +0800
+++ gnome-system-tools-2.30.2-ylmf2.new/src/network/network.desktop.in.in	2011-05-26 09:50:41.451224898 +0800
@@ -3,7 +3,7 @@
 _Name=Network
 _Comment=Configure network devices and connections
 Exec=network-admin
-Icon=preferences-system-network
+Icon=network-setting
 Terminal=false
 Type=Application
 Categories=GNOME;GTK;System;Settings;X-GNOME-NetworkSettings;
