diff -ur qbittorrent-2.7.2/src/Icons/qBittorrent.desktop qbittorrent-2.7.2.new/src/Icons/qBittorrent.desktop
--- qbittorrent-2.7.2/src/Icons/qBittorrent.desktop	2011-03-20 21:04:59.000000000 +0800
+++ qbittorrent-2.7.2.new/src/Icons/qBittorrent.desktop	2011-08-09 15:40:36.511001038 +0800
@@ -33,7 +33,7 @@
 Comment[fr]=Télécharger et partager des fichiers avec BitTorrent
 Comment[pl]=Dzielenie się plikami przez BitTorrent
 Icon=qbittorrent
-MimeType=application/x-bittorrent;x-scheme-handler/magnet;
+MimeType=application/x-bittorrent;
 Name=qBittorrent
 Name[ko]=큐비토런트
 Terminal=false
