diff -ur xorg-server-1.10.4/glx/glxdri2.c xorg-server-1.10.4.new//glx/glxdri2.c
--- xorg-server-1.10.4/glx/glxdri2.c	2011-08-05 13:59:02.000000000 +0800
+++ xorg-server-1.10.4.new//glx/glxdri2.c	2012-03-07 15:34:01.420168130 +0800
@@ -709,6 +709,7 @@
     const __DRIextension **extensions;
     const __DRIconfig **driConfigs;
     int i;
+    int from = X_ERROR;
 
     screen = calloc(1, sizeof *screen);
     if (screen == NULL)
@@ -735,7 +736,9 @@
 
     screen->driver = dlopen(filename, RTLD_LAZY | RTLD_LOCAL);
     if (screen->driver == NULL) {
-	LogMessage(X_ERROR, "AIGLX error: dlopen of %s failed (%s)\n",
+       if (!strcmp(driverName, "nouveau"))
+          from = X_INFO;
+        LogMessage(from, "AIGLX error: dlopen of %s failed (%s)\n",
 		   filename, dlerror());
         goto handle_error;
     }
@@ -827,7 +830,7 @@
 
     free(screen);
 
-    LogMessage(X_ERROR, "AIGLX: reverting to software rendering\n");
+    LogMessage(from, "AIGLX: reverting to software rendering\n");
 
     return NULL;
 }
diff -ur xorg-server-1.10.4/glx/glxdri.c xorg-server-1.10.4.new//glx/glxdri.c
--- xorg-server-1.10.4/glx/glxdri.c	2011-08-05 13:59:02.000000000 +0800
+++ xorg-server-1.10.4.new//glx/glxdri.c	2012-03-07 15:32:43.058648938 +0800
@@ -978,6 +978,7 @@
     const __DRIconfig **driConfigs;
     const __DRIextension **extensions;
     int i;
+    int from = X_ERROR;
 
     if (!xf86LoaderCheckSymbol("DRIQueryDirectRenderingCapable") ||
 	!DRIQueryDirectRenderingCapable(pScreen, &isCapable) ||
@@ -1057,7 +1058,9 @@
 
     screen->driver = dlopen(filename, RTLD_LAZY | RTLD_LOCAL);
     if (screen->driver == NULL) {
-	LogMessage(X_ERROR, "AIGLX error: dlopen of %s failed (%s)\n",
+       if (!strcmp(driverName, "nouveau"))
+          from = X_INFO;
+       LogMessage(from, "AIGLX error: dlopen of %s failed (%s)\n",
 		   filename, dlerror());
         goto handle_error;
     }
@@ -1193,7 +1196,7 @@
 
     free(screen);
 
-    LogMessage(X_ERROR, "AIGLX: reverting to software rendering\n");
+    LogMessage(from, "AIGLX: reverting to software rendering\n");
 
     return NULL;
 }
diff -ur xorg-server-1.10.4/hw/xfree86/common/xf86pciBus.c xorg-server-1.10.4.new//hw/xfree86/common/xf86pciBus.c
--- xorg-server-1.10.4/hw/xfree86/common/xf86pciBus.c	2011-07-17 04:18:32.000000000 +0800
+++ xorg-server-1.10.4.new//hw/xfree86/common/xf86pciBus.c	2012-03-07 15:35:24.235107359 +0800
@@ -1123,7 +1123,24 @@
 	    break;
 	case 0x102b:		    driverList[0] = "mga";	break;
 	case 0x10c8:		    driverList[0] = "neomagic"; break;
-	case 0x10de: case 0x12d2:   driverList[0] = "nv";	break;
+        case 0x10de: case 0x12d2:
+           switch (dev->device_id) {
+           /* NV1 */
+           case 0x0008:
+           case 0x0009:
+               driverList[0] = "vesa";
+               break;
+           /* NV3 */
+           case 0x0018:
+           case 0x0019:
+               driverList[0] = "nv";
+               break;
+           default:
+               driverList[0] = "nouveau";
+               driverList[1] = "nv";
+               break;
+           }
+           break;
 	case 0x1106:		    driverList[0] = "openchrome"; break;
         case 0x1b36:		    driverList[0] = "qxl"; break;
 	case 0x1163:		    driverList[0] = "rendition"; break;
