diff -ur xorg-server-1.9.3-ylmf1/dix/globals.c xorg-server-1.9.3-ylmf1.new/dix/globals.c
--- xorg-server-1.9.3-ylmf1/dix/globals.c	2010-06-07 05:37:03.000000000 +0800
+++ xorg-server-1.9.3-ylmf1.new/dix/globals.c	2011-05-24 13:40:47.074053197 +0800
@@ -122,6 +122,7 @@
 CursorPtr rootCursor;
 Bool party_like_its_1989 = FALSE;
 Bool whiteRoot = FALSE;
+Bool bgNoneRoot = FALSE;
 
 TimeStamp currentTime;
 TimeStamp lastDeviceEventTime;
diff -ur xorg-server-1.9.3-ylmf1/dix/window.c xorg-server-1.9.3-ylmf1.new/dix/window.c
--- xorg-server-1.9.3-ylmf1/dix/window.c	2010-11-10 14:26:52.000000000 +0800
+++ xorg-server-1.9.3-ylmf1.new/dix/window.c	2011-05-24 13:43:13.287384594 +0800
@@ -459,22 +459,25 @@
     pWin->optional->cursor = rootCursor;
     rootCursor->refcnt++;
 
+    pWin->backingStore = defaultBackingStore;
+    pWin->forcedBS = (defaultBackingStore != NotUseful);
 
     if (party_like_its_1989) {
         MakeRootTile(pWin);
         backFlag |= CWBackPixmap;
+	pScreen->ChangeWindowAttributes(pWin, backFlag);
+	} else if (bgNoneRoot) {
+	/* nothing, handled in xf86CreateRootWindow */
+
     } else {
 	if (whiteRoot)
             pWin->background.pixel = pScreen->whitePixel;
         else
             pWin->background.pixel = pScreen->blackPixel;
         backFlag |= CWBackPixel;
-    } 
 
-    pWin->backingStore = defaultBackingStore;
-    pWin->forcedBS = (defaultBackingStore != NotUseful);
-    /* We SHOULD check for an error value here XXX */
-    (*pScreen->ChangeWindowAttributes)(pWin, backFlag);
+	pScreen->ChangeWindowAttributes(pWin, backFlag);
+    }
 
     MapWindow(pWin, serverClient);
 }
diff -ur xorg-server-1.9.3-ylmf1/hw/xfree86/common/xf86Init.c xorg-server-1.9.3-ylmf1.new/hw/xfree86/common/xf86Init.c
--- xorg-server-1.9.3-ylmf1/hw/xfree86/common/xf86Init.c	2010-09-29 09:41:33.000000000 +0800
+++ xorg-server-1.9.3-ylmf1.new/hw/xfree86/common/xf86Init.c	2011-05-24 13:45:14.790716316 +0800
@@ -76,6 +76,7 @@
 #include "xf86Xinput.h"
 #include "xf86InPriv.h"
 #include "picturestr.h"
+#include "xace.h"
 
 #include "xf86Bus.h"
 #include "xf86VGAarbiter.h"
@@ -242,10 +243,12 @@
   int ret = TRUE;
   int err = Success;
   ScreenPtr pScreen = pWin->drawable.pScreen;
+  ScrnInfoPtr pScrn = xf86Screens[pScreen->myNum];
   RootWinPropPtr pProp;
   CreateWindowProcPtr CreateWindow = (CreateWindowProcPtr)
       dixLookupPrivate(&pScreen->devPrivates, xf86CreateRootWindowKey);
 
+
   DebugF("xf86CreateRootWindow(%p)\n", pWin);
 
   if ( pScreen->CreateWindow != xf86CreateRootWindow ) {
@@ -293,6 +296,15 @@
     }
   }
 
+  if (bgNoneRoot && pScrn->canDoBGNoneRoot) {
+      pWin->backgroundState = XaceBackgroundNoneState(pWin);
+      pWin->background.pixel = pScreen->whitePixel;
+      pScreen->ChangeWindowAttributes(pWin, CWBackPixmap | CWBorderPixel | CWCursor | CWBackingStore);
+  } else {
+      pWin->background.pixel = pScreen->blackPixel;
+      pScreen->ChangeWindowAttributes(pWin, CWBackPixel | CWBorderPixel | CWCursor | CWBackingStore);
+  }
+
   DebugF("xf86CreateRootWindow() returns %d\n", ret);
   return ret;
 }
diff -ur xorg-server-1.9.3-ylmf1/hw/xfree86/common/xf86str.h xorg-server-1.9.3-ylmf1.new/hw/xfree86/common/xf86str.h
--- xorg-server-1.9.3-ylmf1/hw/xfree86/common/xf86str.h	2010-06-05 11:37:53.000000000 +0800
+++ xorg-server-1.9.3-ylmf1.new/hw/xfree86/common/xf86str.h	2011-05-24 13:46:38.694048539 +0800
@@ -502,7 +502,7 @@
 } confDRIRec, *confDRIPtr;
 
 /* These values should be adjusted when new fields are added to ScrnInfoRec */
-#define NUM_RESERVED_INTS		16
+#define NUM_RESERVED_INTS               15
 #define NUM_RESERVED_POINTERS		14
 #define NUM_RESERVED_FUNCS		11
 
@@ -774,6 +774,9 @@
     ClockRangePtr	clockRanges;
     int			adjustFlags;
 
+    /* -nr support */
+    int                 canDoBGNoneRoot;
+
     /*
      * These can be used when the minor ABI version is incremented.
      * The NUM_* parameters must be reduced appropriately to keep the
diff -ur xorg-server-1.9.3-ylmf1/include/opaque.h xorg-server-1.9.3-ylmf1.new/include/opaque.h
--- xorg-server-1.9.3-ylmf1/include/opaque.h	2010-09-12 05:20:51.000000000 +0800
+++ xorg-server-1.9.3-ylmf1.new/include/opaque.h	2011-05-24 13:47:10.334048119 +0800
@@ -71,6 +71,7 @@
 extern _X_EXPORT long maxBigRequestSize;
 extern _X_EXPORT Bool party_like_its_1989;
 extern _X_EXPORT Bool whiteRoot;
+extern _X_EXPORT Bool bgNoneRoot;
 
 extern _X_EXPORT Bool CoreDump;
 
diff -ur xorg-server-1.9.3-ylmf1/os/utils.c xorg-server-1.9.3-ylmf1.new/os/utils.c
--- xorg-server-1.9.3-ylmf1/os/utils.c	2010-09-12 05:20:51.000000000 +0800
+++ xorg-server-1.9.3-ylmf1.new/os/utils.c	2011-05-24 13:47:53.454047547 +0800
@@ -502,6 +502,7 @@
 #endif
     ErrorF("-nolisten string       don't listen on protocol\n");
     ErrorF("-noreset               don't reset after last client exists\n");
+    ErrorF("-nr                    create root window with no background\n");
     ErrorF("-reset                 reset after last client exists\n");
     ErrorF("-p #                   screen-saver pattern duration (minutes)\n");
     ErrorF("-pn                    accept failure to listen on all ports\n");
@@ -841,6 +842,8 @@
 	    defaultBackingStore = WhenMapped;
         else if ( strcmp( argv[i], "-wr") == 0)
             whiteRoot = TRUE;
+	else if ( strcmp( argv[i], "-nr") == 0)
+	    bgNoneRoot = TRUE;
         else if ( strcmp( argv[i], "-maxbigreqsize") == 0) {
              if(++i < argc) {
                  long reqSizeArg = atol(argv[i]);
