diff -ur gdm-2.30.2-ylmf1/gui/simple-greeter/gdm-greeter-panel.c gdm-2.30.2-ylmf1.new/gui/simple-greeter/gdm-greeter-panel.c
--- gdm-2.30.2-ylmf1/gui/simple-greeter/gdm-greeter-panel.c	2010-03-18 05:27:16.000000000 +0800
+++ gdm-2.30.2-ylmf1.new/gui/simple-greeter/gdm-greeter-panel.c	2011-05-09 17:02:59.265152376 +0800
@@ -854,7 +854,7 @@
 
         gtk_window_set_keep_above (GTK_WINDOW (panel), TRUE);
         gtk_window_set_type_hint (GTK_WINDOW (panel), GDK_WINDOW_TYPE_HINT_DOCK);
-        gtk_window_set_opacity (GTK_WINDOW (panel), 0.85);
+        //gtk_window_set_opacity (GTK_WINDOW (panel), 0.85);
 
         panel->priv->hbox = gtk_hbox_new (FALSE, 12);
         gtk_container_set_border_width (GTK_CONTAINER (panel->priv->hbox), 0);
