--- src/krubypluginfactory.cpp	2012-08-27 17:07:53.901699757 +0000
+++ src/krubypluginfactory.cpp.new	2012-08-27 17:09:30.581694526 +0000
@@ -164,7 +164,8 @@
 #if RUBY_VERSION >= 0x10900
     VALUE gem = rb_define_module("Gem");
     rb_const_set(gem, rb_intern("Enable"), Qtrue);
-    Init_prelude(); 
+    char *opts[] = { "ruby", "-e;" };
+    ruby_options(2, opts);
 #endif
 
     ruby_incpush(QFile::encodeName(program.path()));
@@ -341,7 +342,8 @@
 #if RUBY_VERSION >= 0x10900
     VALUE gem = rb_define_module("Gem");
     rb_const_set(gem, rb_intern("Enable"), Qtrue);
-    Init_prelude();                                             
+    char *opts[] = { "ruby", "-e;" };
+    ruby_options(2, opts);
 #endif
 
     ruby_incpush(QFile::encodeName(program.path()));
