diff -ur rtl8192se_linux_2.6.0019.1207.2010/HAL/rtl8192/Makefile rtl8192se_linux_2.6.0019.1207.2010.new/HAL/rtl8192/Makefile
--- rtl8192se_linux_2.6.0019.1207.2010/HAL/rtl8192/Makefile	2010-12-07 11:53:18.000000000 +0800
+++ rtl8192se_linux_2.6.0019.1207.2010.new/HAL/rtl8192/Makefile	2011-07-07 09:46:50.806084518 +0800
@@ -2,7 +2,7 @@
 
 CC = gcc
 KVER  := $(shell uname -r)
-MODDESTDIR := /lib/modules/$(KVER)/kernel/drivers/net/wireless/
+MODDESTDIR := $(DESTDIR)/lib/modules/$(KVER)/kernel/drivers/net/wireless/
 
 ifeq ($(shell uname -r|cut -d. -f1,2), 2.6)
 ###################### KERNEL 2.6 ################################
@@ -637,7 +637,7 @@
 ifeq ($(NIC_SELECT),RTL8190P)
 	find /lib/modules/$(shell uname -r) -name "r8190_*.ko" -exec ls -l {} \;
 	find /lib/modules/$(shell uname -r) -name "r8190_*.ko" -exec rm {} \;
-
+	install -d $(MODDESTDIR)
 	install -p -m 644 r8190_pci.ko $(MODDESTDIR)
 endif
 ifeq ($(NIC_SELECT),RTL8192E)
@@ -645,56 +645,63 @@
 	find /lib/modules/$(shell uname -r) -name "r8192e_*.ko" -exec rm {} \;
 	rm -fr /lib/modules/$(shell uname -r)/kernel/ubuntu/rtl8192se
 
+	install -d $(MODDESTDIR)
 	install -p -m 644 r8192e_pci.ko $(MODDESTDIR)
 endif
 ifeq ($(NIC_SELECT),RTL8192SE)
 	find /lib/modules/$(shell uname -r) -name "r8192se_*.ko" -exec ls -l {} \;
 	find /lib/modules/$(shell uname -r) -name "r8192se_*.ko" -exec rm {} \;
 
+	install -d $(MODDESTDIR)
 	install -p -m 644 r8192se_pci.ko $(MODDESTDIR)
 endif
 ifeq ($(NIC_SELECT),RTL8192CE)
 	find /lib/modules/$(shell uname -r) -name "r8192ce_*.ko" -exec ls -l {} \;
 	find /lib/modules/$(shell uname -r) -name "r8192ce_*.ko" -exec rm {} \;
 
+	install -d $(MODDESTDIR)
 	install -p -m 644 r8192ce_pci.ko $(MODDESTDIR)
 endif
 ifeq ($(NIC_SELECT),RTL8191SE_MESH)
 	find /lib/modules/$(shell uname -r) -name "r8192se_*.ko" -exec ls -l {} \;
 	find /lib/modules/$(shell uname -r) -name "r8192se_*.ko" -exec rm {} \;
 
-	mkdir /usr/share/enhanced-easy-network -p
-	cp ../../script/mesh-tools /usr/share/enhanced-easy-network/ -rf
-	cp ../../script/mesh.conf /etc/
-	cp ../../script/wpa_mesh_tkip.conf /etc/
-	cp ../../script/wpa_mesh_ccmp.conf /etc/
+	mkdir $(DESTDIR)/usr/share/enhanced-easy-network -p
+	cp ../../script/mesh-tools $(DESTDIR)/usr/share/enhanced-easy-network/ -rf
+	cp ../../script/mesh.conf $(DESTDIR)/etc/
+	cp ../../script/wpa_mesh_tkip.conf $(DESTDIR)/etc/
+	cp ../../script/wpa_mesh_ccmp.conf $(DESTDIR)/etc/
 	chmod 777 ../../script/do_install
 	../../script/do_install
+	install -d $(MODDESTDIR)
 	install -p -m 644 r8192se_pci_mesh.ko $(MODDESTDIR)
 endif
 ifeq ($(NIC_SELECT),RTL8192CE_MESH)
 	find /lib/modules/$(shell uname -r) -name "r8192ce_*.ko" -exec ls -l {} \;
 	find /lib/modules/$(shell uname -r) -name "r8192ce_*.ko" -exec rm {} \;
 
-	mkdir /usr/share/enhanced-easy-network -p
-	cp ../../script/mesh-tools /usr/share/enhanced-easy-network/ -rf
-	cp ../../script/mesh.conf /etc/
-	cp ../../script/wpa_mesh_tkip.conf /etc/
-	cp ../../script/wpa_mesh_ccmp.conf /etc/
+	mkdir $(DESTDIR)/usr/share/enhanced-easy-network -p
+	cp ../../script/mesh-tools $(DESTDIR)/usr/share/enhanced-easy-network/ -rf
+	cp ../../script/mesh.conf $(DESTDIR)/etc/
+	cp ../../script/wpa_mesh_tkip.conf $(DESTDIR)/etc/
+	cp ../../script/wpa_mesh_ccmp.conf $(DESTDIR)/etc/
 	chmod 777 ../../script/do_install
 	../../script/do_install
+	install -d $(MODDESTDIR)
 	install -p -m 644 r8192ce_pci_mesh.ko $(MODDESTDIR)
 endif
 ifeq ($(NIC_SELECT),RTL8192SE_SA)
 	find /lib/modules/$(shell uname -r) -name "r8192se_*.ko" -exec ls -l {} \;
 	find /lib/modules/$(shell uname -r) -name "r8192se_*.ko" -exec rm {} \;
 
+	install -d $(MODDESTDIR)
 	install -p -m 644 r8192se_pci_softap.ko $(MODDESTDIR)
 endif
 ifeq ($(NIC_SELECT),RTL8192CE_BT30)
 	find /lib/modules/$(shell uname -r) -name "r8192ce_*.ko" -exec ls -l {} \;
 	find /lib/modules/$(shell uname -r) -name "r8192ce_*.ko" -exec rm {} \;
 
+	install -d $(MODDESTDIR)
 	install -p -m 644 r8192ce_pci.ko $(MODDESTDIR)
 endif
 
diff -ur rtl8192se_linux_2.6.0019.1207.2010/Makefile rtl8192se_linux_2.6.0019.1207.2010.new/Makefile
--- rtl8192se_linux_2.6.0019.1207.2010/Makefile	2010-12-07 11:53:18.000000000 +0800
+++ rtl8192se_linux_2.6.0019.1207.2010.new/Makefile	2011-07-07 09:39:37.782756922 +0800
@@ -7,6 +7,7 @@
 HAL_SUB_DIR = rtl8192
 MODULE_FILE = $(RTL819x_DIR)/rtllib/Module.symvers
 SCRIPT_FOLDER = /etc/realtek
+DESTDIR=""
 
 export LINUX_KSRC_MODULE_DIR RTL819x_FIRM_DIR
 
@@ -27,7 +28,8 @@
 install: all
 	@rm -f /etc/acpi/wireless-rtl-ac-dc-power.sh
 	@rm -f /etc/acpi/events/RadioPower.sh
-	@cp -fr realtek/ /etc
+	install -d $(DESTDIR)/etc
+	@cp -fr realtek/ $(DESTDIR)/etc
 	@make -C HAL/$(HAL_SUB_DIR)/ install
 #	@cp $(RTL819x_DIR)/wapi_supplicant/wapi_supplicant /usr/bin/
 uninstall:
