diff -ur cmmusic-1.6-r1/src/commdef.h cmmusic-1.6-r1.new/src/commdef.h
--- cmmusic-1.6-r1/src/commdef.h	2009-06-21 20:35:14.000000000 +0800
+++ cmmusic-1.6-r1.new/src/commdef.h	2011-11-09 16:11:57.932652584 +0800
@@ -34,14 +34,14 @@
 //uimgr.h
 
 #ifdef HAVE_LIBNCURSESW
-#include <ncursesw/curses.h>
-#include <ncursesw/panel.h>
-#include <ncursesw/menu.h>
+#include <curses.h>
+#include <panel.h>
+#include <menu.h>
 #else
 #ifdef HAVE_LIBNCURSES
-#include <ncurses/curses.h>
-#include <ncurses/panel.h>
-#include <ncurses/menu.h>
+#include <curses.h>
+#include <panel.h>
+#include <menu.h>
 #else
 #include <curses.h>
 #include <panel.h>
