--- configure.old	2011-10-13 15:56:24.412068058 +0800
+++ configure	2011-10-13 15:57:42.560230434 +0800
@@ -36,8 +36,8 @@
 #############################################################################
 
   if test -e ffmpeg/.svn ; then
-    echo "You have an outdated FFmpeg SVN checkout in ffmpeg/, please (re)move or replace it"
-    exit 1
+    echo "You have an outdated FFmpeg SVN checkout in ffmpeg/, (re)move it"
+    	rm -rf ffmpeg/
   fi
 
   if test -e ffmpeg/mp_auto_pull ; then
@@ -48,8 +48,7 @@
   fi
 
   if ! test -e ffmpeg ; then
-    echo "No FFmpeg checkout, press enter to download one with git or CTRL+C to abort"
-    read tmp
+    echo "No FFmpeg checkout, download one with git"
     if ! git clone --depth 1 git://git.videolan.org/ffmpeg.git ffmpeg ; then
       rm -rf ffmpeg
       echo "Failed to get a FFmpeg checkout"
