diff -ur mcelog-1.0pre3/Makefile mcelog-1.0pre3.new//Makefile
--- mcelog-1.0pre3/Makefile	2010-01-21 10:36:52.000000000 +0800
+++ mcelog-1.0pre3.new//Makefile	2011-07-18 09:35:10.616424776 +0800
@@ -1,6 +1,6 @@
 CFLAGS := -g -Os
-prefix := /usr
-etcprefix :=
+prefix := $(DESTDIR)/usr
+etcprefix := $(prefix)
 # Define appropiately for your distribution
 # DOCDIR := /usr/share/doc/packages/mcelog
 
