diff -ur gtk+-2.24.6/gtk/gtksignal.h gtk+-2.24.6.new//gtk/gtksignal.h
--- gtk+-2.24.6/gtk/gtksignal.h	2011-08-16 10:30:52.000000000 +0800
+++ gtk+-2.24.6.new//gtk/gtksignal.h	2011-09-14 11:37:23.639968032 +0800
@@ -29,7 +29,9 @@
 #ifndef __GTK_SIGNAL_H__
 #define __GTK_SIGNAL_H__
 
-#include <gtk/gtk.h>
+#include <gtk/gtkenums.h>
+#include <gtk/gtktypeutils.h>
+#include <gtk/gtkobject.h>
 #include <gtk/gtkmarshal.h>
 
 G_BEGIN_DECLS
