diff -ur gtk+-2.24.6/gtk/Makefile.am gtk+-2.24.6.new//gtk/Makefile.am
--- gtk+-2.24.6/gtk/Makefile.am	2011-09-14 11:45:10.088962604 +0800
+++ gtk+-2.24.6.new//gtk/Makefile.am	2011-09-14 11:48:30.159526620 +0800
@@ -32,7 +32,7 @@
 endif
 endif
 
-INCLUDES =						\
+INCLUDES_COMMON =                                      \
 	-DG_LOG_DOMAIN=\"Gtk\"				\
 	-DGTK_LIBDIR=\"$(libdir)\"			\
 	-DGTK_DATADIR=\"$(datadir)\"			\
@@ -52,10 +52,17 @@
 	-DGTK_FILE_SYSTEM_ENABLE_UNSUPPORTED		\
 	-DGTK_PRINT_BACKEND_ENABLE_UNSUPPORTED		\
 	$(GTK_DEBUG_FLAGS)				\
-	$(GTK_DEP_CFLAGS)				\
+	$(GTK_DEP_CFLAGS)
+
+INCLUDES=                                              \
+	$(INCLUDES_COMMON)                              \
 	$(gtk_clipboard_dnd_c_sources_CFLAGS)		\
 	$(INCLUDED_IMMODULE_DEFINE)
 
+INCLUDES_GIR=                                          \
+	$(INCLUDES_COMMON)                              \
+	$(INCLUDED_IMMODULE_DEFINE)
+
 gtarget=$(gdktarget)
 
 if PLATFORM_WIN32
@@ -1019,7 +1026,7 @@
 endif
 Gtk_2_0_gir_INCLUDES = Atk-1.0 Gdk-2.0
 Gtk_2_0_gir_CFLAGS = \
-		$(INCLUDES) \
+		$(INCLUDES_GIR) \	
 		-UGDK_DISABLE_DEPRECATED \
 		-UGTK_DISABLE_DEPRECATED \
 		-DGTK_TEXT_USE_INTERNAL_UNSUPPORTED_API
