Index: dbus-1.2.16/tools/dbus-monitor.c
===================================================================
--- dbus-1.2.16.orig/tools/dbus-monitor.c	2009-07-14 21:42:25.000000000 +0200
+++ dbus-1.2.16/tools/dbus-monitor.c	2009-07-16 02:08:59.000000000 +0200
@@ -33,8 +33,6 @@
 
 #include <time.h>
 
-#include <signal.h>
-
 #include "dbus-print-message.h"
 
 #ifdef DBUS_WIN
@@ -195,14 +193,6 @@
   exit (ecode);
 }
 
-static dbus_bool_t sigint_received = FALSE;
-
-static void
-sigint_handler (int signum)
-{
-  sigint_received = TRUE;
-}
-
 int
 main (int argc, char *argv[])
 {
@@ -338,11 +328,7 @@
     fprintf (stderr, "Couldn't add filter!\n");
     exit (1);
   }
-
-  /* we handle SIGINT so exit() is reached and flushes stdout */
-  signal (SIGINT, sigint_handler);
-  while (dbus_connection_read_write_dispatch(connection, -1)
-          && !sigint_received)
+  while (dbus_connection_read_write_dispatch(connection, -1))
     ;
   exit (0);
  lose:
