Index: dbus-1.2.16/dbus/dbus-sysdeps-unix.c
===================================================================
--- dbus-1.2.16.orig/dbus/dbus-sysdeps-unix.c	2009-07-14 21:42:38.000000000 +0200
+++ dbus-1.2.16/dbus/dbus-sysdeps-unix.c	2009-07-16 02:09:20.000000000 +0200
@@ -1241,7 +1241,7 @@
 #elif defined(HAVE_CMSGCRED)
     struct cmsgcred *cred;
 
-    cred = (struct cmsgcred *) CMSG_DATA (&cmsg);
+    cred = (struct cmsgcred *) CMSG_DATA (&cmsg.hdr);
     pid_read = cred->cmcred_pid;
     uid_read = cred->cmcred_euid;
 #elif defined(LOCAL_CREDS)
