diff --git a/bus/Makefile.am b/bus/Makefile.am
index 3b4f69d..ad49e6d 100644
--- a/bus/Makefile.am
+++ b/bus/Makefile.am
@@ -78,8 +78,8 @@ dbus_daemon_SOURCES=				\
 
 dbus_daemon_LDADD=					\
 	$(EFENCE)					\
-	$(DBUS_BUS_LIBS)				\
-	$(top_builddir)/dbus/libdbus-convenience.la
+	$(top_builddir)/dbus/libdbus-convenience.la	\
+	$(DBUS_BUS_LIBS)
 
 dbus_daemon_LDFLAGS=@R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@ @PIE_LDFLAGS@
 
@@ -103,8 +103,8 @@ dbus_daemon_launch_helper_SOURCES=		\
 	$(LAUNCH_HELPER_SOURCES)
 
 dbus_daemon_launch_helper_LDADD=		\
-	$(DBUS_LAUNCHER_LIBS)			\
-	$(top_builddir)/dbus/libdbus-convenience.la
+	$(top_builddir)/dbus/libdbus-convenience.la \
+	$(DBUS_LAUNCHER_LIBS)
 
 dbus_daemon_launch_helper_LDFLAGS=@R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@
 
@@ -115,8 +115,8 @@ dbus_daemon_launch_helper_test_SOURCES=		\
 	$(LAUNCH_HELPER_SOURCES)
 
 dbus_daemon_launch_helper_test_LDADD=		\
-	$(DBUS_LAUNCHER_LIBS)			\
-	$(top_builddir)/dbus/libdbus-convenience.la
+	$(top_builddir)/dbus/libdbus-convenience.la \
+	$(DBUS_LAUNCHER_LIBS)
 
 dbus_daemon_launch_helper_test_LDFLAGS=@R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@
 dbus_daemon_launch_helper_test_CPPFLAGS=	\
@@ -129,8 +129,8 @@ bus_test_launch_helper_SOURCES=		\
 	$(LAUNCH_HELPER_SOURCES)
 
 bus_test_launch_helper_LDADD=		\
-	$(DBUS_LAUNCHER_LIBS)			\
-	$(top_builddir)/dbus/libdbus-convenience.la
+	$(top_builddir)/dbus/libdbus-convenience.la \
+	$(DBUS_LAUNCHER_LIBS)
 
 bus_test_launch_helper_LDFLAGS=@R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@
 bus_test_launch_helper_CPPFLAGS=	\
diff --git a/bus/Makefile.in b/bus/Makefile.in
index fde3829..02520b6 100644
--- a/bus/Makefile.in
+++ b/bus/Makefile.in
@@ -111,8 +111,9 @@ am_bus_test_launch_helper_OBJECTS =  \
 	bus_test_launch_helper-test-launch-helper.$(OBJEXT) \
 	$(am__objects_5)
 bus_test_launch_helper_OBJECTS = $(am_bus_test_launch_helper_OBJECTS)
-bus_test_launch_helper_DEPENDENCIES = $(am__DEPENDENCIES_1) \
-	$(top_builddir)/dbus/libdbus-convenience.la
+bus_test_launch_helper_DEPENDENCIES =  \
+	$(top_builddir)/dbus/libdbus-convenience.la \
+	$(am__DEPENDENCIES_1)
 bus_test_launch_helper_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(bus_test_launch_helper_LDFLAGS) $(LDFLAGS) -o $@
@@ -142,8 +143,9 @@ am__dbus_daemon_SOURCES_DIST = activation.c activation.h \
 	config-loader-libxml.c main.c
 am_dbus_daemon_OBJECTS = $(am__objects_3) main.$(OBJEXT)
 dbus_daemon_OBJECTS = $(am_dbus_daemon_OBJECTS)
-dbus_daemon_DEPENDENCIES = $(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
-	$(top_builddir)/dbus/libdbus-convenience.la
+dbus_daemon_DEPENDENCIES = $(am__DEPENDENCIES_1) \
+	$(top_builddir)/dbus/libdbus-convenience.la \
+	$(am__DEPENDENCIES_1)
 dbus_daemon_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(dbus_daemon_LDFLAGS) $(LDFLAGS) -o $@
@@ -160,8 +162,9 @@ am_dbus_daemon_launch_helper_OBJECTS =  \
 	activation-helper-bin.$(OBJEXT) $(am__objects_6)
 dbus_daemon_launch_helper_OBJECTS =  \
 	$(am_dbus_daemon_launch_helper_OBJECTS)
-dbus_daemon_launch_helper_DEPENDENCIES = $(am__DEPENDENCIES_1) \
-	$(top_builddir)/dbus/libdbus-convenience.la
+dbus_daemon_launch_helper_DEPENDENCIES =  \
+	$(top_builddir)/dbus/libdbus-convenience.la \
+	$(am__DEPENDENCIES_1)
 dbus_daemon_launch_helper_LINK = $(LIBTOOL) --tag=CC \
 	$(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=link $(CCLD) \
 	$(AM_CFLAGS) $(CFLAGS) $(dbus_daemon_launch_helper_LDFLAGS) \
@@ -185,8 +188,9 @@ am_dbus_daemon_launch_helper_test_OBJECTS = dbus_daemon_launch_helper_test-activ
 	$(am__objects_8)
 dbus_daemon_launch_helper_test_OBJECTS =  \
 	$(am_dbus_daemon_launch_helper_test_OBJECTS)
-dbus_daemon_launch_helper_test_DEPENDENCIES = $(am__DEPENDENCIES_1) \
-	$(top_builddir)/dbus/libdbus-convenience.la
+dbus_daemon_launch_helper_test_DEPENDENCIES =  \
+	$(top_builddir)/dbus/libdbus-convenience.la \
+	$(am__DEPENDENCIES_1)
 dbus_daemon_launch_helper_test_LINK = $(LIBTOOL) --tag=CC \
 	$(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=link $(CCLD) \
 	$(AM_CFLAGS) $(CFLAGS) \
@@ -497,8 +501,8 @@ dbus_daemon_SOURCES = \
 
 dbus_daemon_LDADD = \
 	$(EFENCE)					\
-	$(DBUS_BUS_LIBS)				\
-	$(top_builddir)/dbus/libdbus-convenience.la
+	$(top_builddir)/dbus/libdbus-convenience.la	\
+	$(DBUS_BUS_LIBS)
 
 dbus_daemon_LDFLAGS = @R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@ @PIE_LDFLAGS@
 LAUNCH_HELPER_SOURCES = \
@@ -520,8 +524,8 @@ dbus_daemon_launch_helper_SOURCES = \
 	$(LAUNCH_HELPER_SOURCES)
 
 dbus_daemon_launch_helper_LDADD = \
-	$(DBUS_LAUNCHER_LIBS)			\
-	$(top_builddir)/dbus/libdbus-convenience.la
+	$(top_builddir)/dbus/libdbus-convenience.la \
+	$(DBUS_LAUNCHER_LIBS)
 
 dbus_daemon_launch_helper_LDFLAGS = @R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@
 dbus_daemon_launch_helper_test_SOURCES = \
@@ -529,8 +533,8 @@ dbus_daemon_launch_helper_test_SOURCES = \
 	$(LAUNCH_HELPER_SOURCES)
 
 dbus_daemon_launch_helper_test_LDADD = \
-	$(DBUS_LAUNCHER_LIBS)			\
-	$(top_builddir)/dbus/libdbus-convenience.la
+	$(top_builddir)/dbus/libdbus-convenience.la \
+	$(DBUS_LAUNCHER_LIBS)
 
 dbus_daemon_launch_helper_test_LDFLAGS = @R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@
 dbus_daemon_launch_helper_test_CPPFLAGS = \
@@ -541,8 +545,8 @@ bus_test_launch_helper_SOURCES = \
 	$(LAUNCH_HELPER_SOURCES)
 
 bus_test_launch_helper_LDADD = \
-	$(DBUS_LAUNCHER_LIBS)			\
-	$(top_builddir)/dbus/libdbus-convenience.la
+	$(top_builddir)/dbus/libdbus-convenience.la \
+	$(DBUS_LAUNCHER_LIBS)
 
 bus_test_launch_helper_LDFLAGS = @R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@
 bus_test_launch_helper_CPPFLAGS = \
