diff -ur rtl8192_8188CU_linux_v3.0.1590.20110511/Makefile rtl8192_8188CU_linux_v3.0.1590.20110511.new//Makefile
--- rtl8192_8188CU_linux_v3.0.1590.20110511/Makefile	2011-05-11 11:34:42.000000000 +0800
+++ rtl8192_8188CU_linux_v3.0.1590.20110511.new//Makefile	2011-11-03 13:24:39.215870737 +0800
@@ -413,7 +413,8 @@
 	$(CROSS_COMPILE)strip $(MODULE_NAME).ko --strip-unneeded
 
 install:
-	install -p -m 644 $(MODULE_NAME).ko  $(MODDESTDIR)
+	mkdir -p $(DESTDIR)/$(MODDESTDIR)
+	install -p -m 644 $(MODULE_NAME).ko  $(DESTDIR)/$(MODDESTDIR)
 	/sbin/depmod -a ${KVER}
 
 uninstall:
