--- ./os-probes/mounted/common/90linux-distro	2013-01-07 10:57:50.321326343 +0800
+++ ./os-probes/mounted/common/90linux-distro.new	2013-01-07 11:07:56.634744112 +0800
@@ -125,6 +125,9 @@
 	elif [ -e "$dir/etc/meego-release" ]; then
 		short="MeeGo"
 		long="$(head -1 "$dir/etc/meego-release")"
+	elif [ -e "$dir/etc/startos-release" ]; then
+		short="StartOS"
+		long="$(head -1 "$dir/etc/startos-release")"
 	else
 		short="Linux"
 		long="unknown Linux distribution"
