diff -ur firefox-7.0.1-ylmf1/modules/libpref/src/init/all.js firefox-7.0.1-ylmf1.new/modules/libpref/src/init/all.js
--- firefox-7.0.1-ylmf1/modules/libpref/src/init/all.js	2011-09-29 04:25:00.000000000 +0800
+++ firefox-7.0.1-ylmf1.new/modules/libpref/src/init/all.js	2011-10-17 09:23:18.299701913 +0800
@@ -1090,7 +1090,7 @@
 pref("intl.charset.detector",               "chrome://global/locale/intl.properties");
 pref("intl.charset.default",                "chrome://global-platform/locale/intl.properties");
 pref("intl.ellipsis",                       "chrome://global-platform/locale/intl.properties");
-pref("intl.locale.matchOS",                 false);
+pref("intl.locale.matchOS",                 true);
 // fallback charset list for Unicode conversion (converting from Unicode)
 // currently used for mail send only to handle symbol characters (e.g Euro, trademark, smartquotes)
 // for ISO-8859-1
