diff -ur lightdm-1.1.7/data/lightdm.conf lightdm-1.1.7.new//data/lightdm.conf
--- lightdm-1.1.7/data/lightdm.conf	2012-03-08 12:53:30.000000000 +0800
+++ lightdm-1.1.7.new//data/lightdm.conf	2012-03-13 10:23:25.675812383 +0800
@@ -70,6 +70,7 @@
 #allow-guest=true
 #guest-session=UNIMPLEMENTED
 #session-wrapper=lightdm-session
+session-wrapper=/etc/lightdm/Xsession
 #display-setup-script=
 #greeter-setup-script=
 #session-setup-script=
