diff -ur rtl_92ce_92se_92de_linux_mac80211_0003.0401.2011/Makefile rtl_92ce_92se_92de_linux_mac80211_0003.0401.2011.new/Makefile
--- rtl_92ce_92se_92de_linux_mac80211_0003.0401.2011/Makefile	2011-03-31 17:07:26.000000000 +0800
+++ rtl_92ce_92se_92de_linux_mac80211_0003.0401.2011.new/Makefile	2011-07-07 11:26:40.106005160 +0800
@@ -2,8 +2,8 @@
 CC = gcc
 KVER  := $(shell uname -r)
 KSRC := /lib/modules/$(KVER)/build
-MODDESTDIR := /lib/modules/$(KVER)/kernel/drivers/net/wireless/rtlwifi
-FIRMWAREDIR := /lib/firmware/
+MODDESTDIR := $(DESTDIR)/lib/modules/$(KVER)/kernel/drivers/net/wireless/rtlwifi
+FIRMWAREDIR := $(DESTDIR)/lib/firmware/
 PWD := $(shell pwd)
 CLR_MODULE_FILES := *.mod.c *.mod *.o .*.cmd *.ko *~ .tmp_versions* modules.order Module.symvers
 SYMBOL_FILE := Module.symvers
@@ -35,10 +35,11 @@
 	find /lib/modules/$(shell uname -r) -name "r8192se_*.ko" -exec rm {} \;
 	find /lib/modules/$(shell uname -r) -name "r8192ce_*.ko" -exec rm {} \;
 	$(shell rm -fr $(MODDESTDIR))
-	$(shell mkdir $(MODDESTDIR))
-	$(shell mkdir $(MODDESTDIR)/rtl8192se)
-	$(shell mkdir $(MODDESTDIR)/rtl8192ce)
-	$(shell mkdir $(MODDESTDIR)/rtl8192de)
+	$(shell mkdir -p $(MODDESTDIR))
+	$(shell mkdir -p $(FIRMWAREDIR))
+	$(shell mkdir -p $(MODDESTDIR)/rtl8192se)
+	$(shell mkdir -p $(MODDESTDIR)/rtl8192ce)
+	$(shell mkdir -p $(MODDESTDIR)/rtl8192de)
 
 	@install -p -m 644 rtlwifi.ko $(MODDESTDIR)	
 	@install -p -m 644 ./rtl8192se/rtl8192se.ko $(MODDESTDIR)/rtl8192se
