--- systemd-29/src/gnome-ask-password-agent.c	2011-04-16 08:28:42.000000000 +0800
+++ systemd-29-ylmf1/src/gnome-ask-password-agent.c	2011-08-09 10:54:16.597734788 +0800
@@ -548,7 +548,7 @@
 	gtk_status_icon_set_from_icon_name ((GtkStatusIcon*) self, self->priv->icon);
 	gtk_status_icon_set_visible ((GtkStatusIcon*) self, TRUE);
 	_tmp17_ = gtk_status_icon_get_title ((GtkStatusIcon*) self);
-	_tmp18_ = notify_notification_new (_tmp17_, self->priv->message, self->priv->icon);
+	_tmp18_ = notify_notification_new (_tmp17_, self->priv->message, self->priv->icon, "NULL");
 	n = _tmp18_;
 	notify_notification_set_timeout (n, 5000);
 	notify_notification_show (n, &_inner_error_);
