--- wxglade	2011-12-26 20:20:50.000000000 +0800
+++ /usr/bin/wxglade	2012-04-25 09:25:21.026888002 +0800
@@ -22,6 +22,10 @@
   WXGLADE_MODULE_PATH="/usr/lib/python${PY_VERSION}/wxglade"
 fi
 
+if [ -e /etc/ylmfos-release ]; then
+	WXGLADE_MODULE_PATH="/usr/lib/python${PY_VERSION}/site-packages/wxglade"
+fi	
+
 # search wxglade.py
 if [ -e "$(dirname $0)/wxglade.py" ]; then
   WXG_PATH="$(dirname $0)/wxglade.py"
