diff -ur python-2.7.3-r2/Lib/platform.py python-2.7.3-r3//Lib/platform.py
--- python-2.7.3-r2/Lib/platform.py	2012-04-10 07:07:31.000000000 +0800
+++ python-2.7.3-r3//Lib/platform.py	2012-12-17 14:34:35.162574722 +0800
@@ -259,7 +259,7 @@
 _supported_dists = (
     'SuSE', 'debian', 'fedora', 'redhat', 'centos',
     'mandrake', 'mandriva', 'rocks', 'slackware', 'yellowdog', 'gentoo',
-    'UnitedLinux', 'turbolinux')
+    'UnitedLinux', 'turbolinux', 'startos')
 
 def _parse_release_file(firstline):
 
@@ -360,7 +360,7 @@
     """
     return linux_distribution(distname, version, id,
                               supported_dists=supported_dists,
-                              full_distribution_name=0)
+                              full_distribution_name=1)
 
 class _popen:
 
