diff -ur python-2.6.7-ylmf1/Lib/platform.py python-2.6.7-ylmf1.new//Lib/platform.py
--- python-2.6.7-ylmf1/Lib/platform.py	2010-08-03 16:09:15.000000000 +0800
+++ python-2.6.7-ylmf1.new//Lib/platform.py	2012-01-05 17:06:07.507810168 +0800
@@ -243,7 +243,7 @@
 _supported_dists = (
     'SuSE', 'debian', 'fedora', 'redhat', 'centos',
     'mandrake', 'mandriva', 'rocks', 'slackware', 'yellowdog', 'gentoo',
-    'UnitedLinux', 'turbolinux')
+    'UnitedLinux', 'turbolinux', 'ylmfos', 'Ylmf', 'YlmfOS')
 
 def _parse_release_file(firstline):
 
