=== modified file 'wine-1.1.18~winehq0~ubuntu~9.04/dlls/user32/sysparams.c'
--- wine-1.1.18~winehq0~ubuntu~9.04/dlls/user32/sysparams.c	2009-04-05 11:35:46 +0000
+++ wine-1.1.18~winehq0~ubuntu~9.04/dlls/user32/sysparams.c	2009-04-06 14:00:05 +0000
@@ -94,37 +94,37 @@
 
 static const char * const DefSysColors[] =
 {
-    "Scrollbar", "212 208 200",              /* COLOR_SCROLLBAR */
+    "Scrollbar", "200 200 200",              /* COLOR_SCROLLBAR */
     "Background", "58 110 165",              /* COLOR_BACKGROUND */
-    "ActiveTitle", "10 36 106",              /* COLOR_ACTIVECAPTION */
-    "InactiveTitle", "128 128 128",          /* COLOR_INACTIVECAPTION */
-    "Menu", "212 208 200",                   /* COLOR_MENU */
+    "ActiveTitle", "0 153 11",               /* COLOR_ACTIVECAPTION */
+    "InactiveTitle", "232 227 221",          /* COLOR_INACTIVECAPTION */
+    "Menu", "232 227 221",                   /* COLOR_MENU */
     "Window", "255 255 255",                 /* COLOR_WINDOW */
     "WindowFrame", "0 0 0",                  /* COLOR_WINDOWFRAME */
     "MenuText", "0 0 0",                     /* COLOR_MENUTEXT */
     "WindowText", "0 0 0",                   /* COLOR_WINDOWTEXT */
     "TitleText", "255 255 255",              /* COLOR_CAPTIONTEXT */
-    "ActiveBorder", "212 208 200",           /* COLOR_ACTIVEBORDER */
-    "InactiveBorder", "212 208 200",         /* COLOR_INACTIVEBORDER */
+    "ActiveBorder", "239 235 230",           /* COLOR_ACTIVEBORDER */
+    "InactiveBorder", "192 192 192",         /* COLOR_INACTIVEBORDER */
     "AppWorkSpace", "128 128 128",           /* COLOR_APPWORKSPACE */
-    "Hilight", "10 36 106",                  /* COLOR_HIGHLIGHT */
+    "Hilight", "0 153 11",                   /* COLOR_HIGHLIGHT */
     "HilightText", "255 255 255",            /* COLOR_HIGHLIGHTTEXT */
-    "ButtonFace", "212 208 200",             /* COLOR_BTNFACE */
+    "ButtonFace", "232 227 221",             /* COLOR_BTNFACE */
     "ButtonShadow", "128 128 128",           /* COLOR_BTNSHADOW */
     "GrayText", "128 128 128",               /* COLOR_GRAYTEXT */
     "ButtonText", "0 0 0",                   /* COLOR_BTNTEXT */
-    "InactiveTitleText", "212 208 200",      /* COLOR_INACTIVECAPTIONTEXT */
+    "InactiveTitleText", "0 0 0",            /* COLOR_INACTIVECAPTIONTEXT */
     "ButtonHilight", "255 255 255",          /* COLOR_BTNHIGHLIGHT */
-    "ButtonDkShadow", "64 64 64",            /* COLOR_3DDKSHADOW */
-    "ButtonLight", "212 208 200",            /* COLOR_3DLIGHT */
-    "InfoText", "0 0 0",                     /* COLOR_INFOTEXT */
+    "ButtonDkShadow", "128 128 128",         /* COLOR_3DDKSHADOW */
+    "ButtonLight", "239 235 230",            /* COLOR_3DLIGHT */
+    "InfoText", "0 0 0-0 0 0",               /* COLOR_INFOTEXT */
     "InfoWindow", "255 255 225",             /* COLOR_INFOBK */
-    "ButtonAlternateFace", "181 181 181",    /* COLOR_ALTERNATEBTNFACE */
-    "HotTrackingColor", "0 0 128",           /* COLOR_HOTLIGHT */
-    "GradientActiveTitle", "166 202 240",    /* COLOR_GRADIENTACTIVECAPTION */
-    "GradientInactiveTitle", "192 192 192",  /* COLOR_GRADIENTINACTIVECAPTION */
-    "MenuHilight", "10 36 106",              /* COLOR_MENUHILIGHT */
-    "MenuBar", "212 208 200"                 /* COLOR_MENUBAR */
+    "ButtonAlternateFace", "180 180 180",    /* COLOR_ALTERNATEBTNFACE */
+    "HotTrackingColor", "180 180 180",       /* COLOR_HOTLIGHT */
+    "GradientActiveTitle", "0 153 11",       /* COLOR_GRADIENTACTIVECAPTION */
+    "GradientInactiveTitle", "232 227 221",  /* COLOR_GRADIENTINACTIVECAPTION */
+    "MenuHilight", "0 0 0",                  /* COLOR_MENUHILIGHT */
+    "MenuBar", "232 227 221"                 /* COLOR_MENUBAR */
 };
 
 /**

