=== modified file 'wine-1.1.21~winehq0~ubuntu~9.04/dlls/gdi32/freetype.c'
--- wine-1.1.21~winehq0~ubuntu~9.04/dlls/gdi32/freetype.c	2009-05-21 09:27:10 +0000
+++ wine-1.1.21~winehq0~ubuntu~9.04/dlls/gdi32/freetype.c	2009-05-21 09:31:23 +0000
@@ -3398,6 +3398,14 @@ GdiFont *WineEngCreateFontInstance(DC *d
     if(!strcmpiW(lf.lfFaceName, SymbolW))
         lf.lfCharSet = SYMBOL_CHARSET;
 
+    if(!strncmp(getenv("LANG"), "zh_CN", 5) && (lf.lfCharSet == DEFAULT_CHARSET || lf.lfCharSet == ANSI_CHARSET)) { 
+        int codepage = 936;
+        if(!TranslateCharsetInfo((DWORD*)(INT_PTR)codepage, &csi, TCI_SRCCODEPAGE)) {
+           csi.fs.fsCsb[0] = 0;
+        }
+        lf.lfCharSet = csi.ciCharset;
+    }
+    else
     if(!TranslateCharsetInfo((DWORD*)(INT_PTR)lf.lfCharSet, &csi, TCI_SRCCHARSET)) {
         switch(lf.lfCharSet) {
 	case DEFAULT_CHARSET:
=== modified file 'wine-1.1.21~winehq0~ubuntu~9.04/dlls/winex11.drv/xrender.c'
--- wine-1.1.21~winehq0~ubuntu~9.04/dlls/winex11.drv/xrender.c	2009-05-21 09:27:10 +0000
+++ wine-1.1.21~winehq0~ubuntu~9.04/dlls/winex11.drv/xrender.c	2009-05-21 09:31:23 +0000
@@ -491,6 +491,7 @@ static int GetCacheEntry(X11DRV_PDEVICE 
     gsCacheEntry *entry;
     static int hinter = -1;
     static int subpixel = -1;
+    WORD flags;
     BOOL font_smoothing;
 
     if((ret = LookupEntry(plfsz)) != -1) return ret;
@@ -511,29 +512,13 @@ static int GetCacheEntry(X11DRV_PDEVICE 
             hinter = status.wFlags & WINE_TT_HINTER_ENABLED;
             subpixel = status.wFlags & WINE_TT_SUBPIXEL_RENDERING_ENABLED;
         }
-
-        switch (plfsz->lf.lfQuality)
-        {
-            case ANTIALIASED_QUALITY:
-                entry->aa_default = get_antialias_type( physDev, FALSE, hinter );
-                break;
-            case CLEARTYPE_QUALITY:
-            case CLEARTYPE_NATURAL_QUALITY:
-                entry->aa_default = get_antialias_type( physDev, subpixel, hinter );
-                break;
-            case DEFAULT_QUALITY:
-            case DRAFT_QUALITY:
-            case PROOF_QUALITY:
-            default:
-                if ( SystemParametersInfoW( SPI_GETFONTSMOOTHING, 0, &font_smoothing, 0) &&
-                     font_smoothing)
-                {
-                    entry->aa_default = get_antialias_type( physDev, subpixel, hinter );
-                }
-                else
-                    entry->aa_default = AA_None;
-                break;
-        }
+	// xxx the buggy antialias type decision code
+        if ( subpixel && X11DRV_XRender_Installed)
+            entry->aa_default = AA_RGB;
+        else if(!hinter || !get_gasp_flags(physDev, &flags) || flags & GASP_DOGRAY)
+            entry->aa_default = AA_Grey;
+        else
+            entry->aa_default = AA_None;
     }
     else
         entry->aa_default = AA_None;

